---
title: "Sheng and LaFleur 2018 RNA-seq reanalysis"
output: html_notebook
---

Sheng and LaGleur used RNA-seq. They aligned their data using STAR to hg19 (it appears ENSEMBL, judging from the gene names given), yielding Integer counts. They did differential expression analysis using edgeR. 

The RNA-seq methods description from their methods

```
Raw reads were aligned to hg19 or mm9 using STAR (v2.4.2a) with the parameter ‘‘quantMode’’ set as ‘‘GeneCounts’’ to produce count table for each gene. Differential gene analysis was performed on gene raw counts in R with edgeR package (v3.18.1) from bio- conductor. Read count table was filtered so that genes with at least one count across conditions were kept. The negative binomial generalized log-linear model was used in differential analysis. A FDR cut-off of 0.05 was used to determine significantly differentially expressed genes. The R package gProfileR (v0.6.4) was used to perform gene enrichment analysis on differential genes. Gene set enrichment analysis (GSEA) was performed with R package clusterProfiler (v3.4.4).
The function analyzeRepeats.pl from Homer (http://homer.ucsd.edu/homer/) software was used to get raw counts for repeats from RNA-seq data. Differential expression for repeats was performed with edgeR the same way as for genes.
```

This notebook depicts the re-analysis of human data

# Setup
```{r setup}
install.packages("readxl")
library(readxl)
source("https://bioconductor.org/biocLite.R")
biocLite("Biobase")
biocLite("GEOquery")
biocLite("limma")
biocLite("edgeR")
biocLite("gProfileR")
biocLite("GSVA")
biocLite("GO.db")
biocLite("org.Hs.eg.db")
biocLite("org.Mm.eg.db")
source("../../pipelines/gsva_script.R")
biocLite("EnsDb.Hsapiens.v86")
```

```{r}
library("Biobase")
library("GEOquery")
library("limma")
library("edgeR")
library("gProfileR")
library("GSVA")
library("GO.db")
library("org.Hs.eg.db")
library("org.Mm.eg.db")
library("EnsDb.Hsapiens.v86")
```


# Get data

## import datasets

```{r get_csv_counts}
human_rnaseq <- read.csv(file = "../../datasets/sheng_lafleur/GSE112230_2018-03-14_rnaseq_hs_counts.csv",header = TRUE,row.names = 1,sep = ",")
human_erv <- read.csv(file = "../../datasets/sheng_lafleur/GSE112230_2018-03-22_repeat_count_hs.csv",header = TRUE,row.names = 1,sep = ",")
```

## design

```{r}
#human_rnaseq_design <- data.frame()
human_rnaseq_design <- data.frame(cell = factor(rep("MCF7",ncol(human_rnaseq))))
rownames(human_rnaseq_design) <- colnames(human_rnaseq)
human_rnaseq_design$treatment <- factor(c(rep("shC",2),rep("shLSD1",2)))
human_rnaseq_design$rep <- factor(c(1,2,1,2))
print(human_rnaseq_design)

# model matrix
human_rnaseq_model_matrix <- model.matrix(~ treatment,human_rnaseq_design)
print(human_rnaseq_model_matrix)
# treat the sh-Control as the "baseline", so it gets an intercept term
```

# edgeR analysis

## dge list
set DGElist object
```{r dgelist}
human_rnaseq_dge <- DGEList(counts=human_rnaseq,
                            genes = data.frame(genes = rownames(human_rnaseq))) # set DGE list
human_rnaseq_dge <- calcNormFactors(object = human_rnaseq_dge) # set normalization factors
human_rnaseq_dge <- estimateDisp(y = human_rnaseq_dge,
                                 design = human_rnaseq_model_matrix) # set dispersion factors
plotBCV(y = human_rnaseq_dge)
# human_rnaseq_dge_exactTest <- exactTest(human_rnaseq_dge)
```

## glm
fit GLM
```{r}
human_rnaseq_dge_fit <- glmQLFit(human_rnaseq_dge, human_rnaseq_model_matrix)
```

## check conditions
Perform tests
```{r}
logFC_edgeR_lim <- 2
pval_cutoff <- 0.01
n_toptag <- 10000
# coef = 1 measures the baseline genes
human_rnaseq_dge_fit_coeff1_qlft <- glmQLFTest(glmfit = human_rnaseq_dge_fit,coef = 1)
# coef = 2 measures the fold-changes between the control and the fold change
human_rnaseq_dge_fit_coeff2_qlft <- glmQLFTest(glmfit = human_rnaseq_dge_fit,
                                               coef = 2,
                                               poisson.bound = TRUE)
# coef = 2, but we use a LRT. 
human_rnaseq_dge_fit_coeff2_lrt <- glmLRT(glmfit = human_rnaseq_dge_fit,
                                          coef = 2)
# coeff = 2 with "glmTreat". This option performs a fold-change cutoff
human_rnaseq_dge_fit_treat <- glmTreat(glmfit = human_rnaseq_dge_fit, 
                                       coef=2, 
                                       lfc=logFC_edgeR_lim)

# topTags; show top 10
human_rnaseq_dge_fit_coeff1_qlft_top <- topTags(human_rnaseq_dge_fit_coeff1_qlft)
human_rnaseq_dge_fit_coeff2_qlft_top <- topTags(human_rnaseq_dge_fit_coeff2_qlft,
                                                p.value = pval_cutoff,
                                                n = n_toptag,
                                                adjust.method = "BH",sort.by = "logFC")
human_rnaseq_dge_fit_coeff2_qlft_top_up <- human_rnaseq_dge_fit_coeff2_qlft_top
human_rnaseq_dge_fit_coeff2_qlft_top_up$table <- subset(human_rnaseq_dge_fit_coeff2_qlft_top$table,abs(logFC) > logFC_edgeR_lim)
human_rnaseq_dge_fit_coeff2_lrt_top <- topTags(object = human_rnaseq_dge_fit_coeff2_lrt,
                                               p.value = pval_cutoff,
                                               n = n_toptag,
                                               adjust.method = "BH",sort.by = "PValue")
human_rnaseq_dge_fit_coeff2_treat_top <- topTags(object = human_rnaseq_dge_fit_treat,
                                                 p.value = pval_cutoff,
                                                 adjust.method = "BH",
                                                 n = n_toptag,sort.by = "logFC")

# plot the fold-changes-vs-pvalue and the top genes by the LRT
plot(human_rnaseq_dge_fit_coeff2_qlft$table$logFC,
     -log10(human_rnaseq_dge_fit_coeff2_qlft$table$PValue),
     pch=20)
points(human_rnaseq_dge_fit_coeff2_qlft_top_up$table$logFC,
       -log10(human_rnaseq_dge_fit_coeff2_qlft_top_up$table$PValue),
       col="red",pch=20)

# plotMD
plotMD(object = human_rnaseq_dge_fit_coeff2_qlft,adjust.method = "BH",p.value = pval_cutoff)
```

# GO analysis

## preparation
We must convert the ENSEMBL names to gene symbols
```{r convert_ens_names}
# get ENSEMBL annotation
ensembl_annot <- read.table(file = "../../datasets/transcripts/Homo_sapiens.GRCh38.91.names.txt",header = FALSE,sep = "\t")
ensembl_annot$V1 <- sapply(ensembl_annot$V1,
                           function(x) {
                             a <- unlist(strsplit(x = as.character(x),split = " "))
                             a[2]
                           })
ensembl_annot$V2 <- sapply(ensembl_annot$V2,
                           function(x) {
                             a <- unlist(strsplit(x = as.character(x),split = " "))
                             a[2]
                           })
colnames(ensembl_annot) <- c("ensid","symbol")

# obtain the symbols of the most differentially-expressed genes
ens_names <- as.character(human_rnaseq_dge_fit_coeff2_lrt_top$table$genes)
ens_names <- as.character(human_rnaseq_dge_fit_coeff2_qlft_top_up$table$genes)

# split the given names
ens_names_split <- sapply(ens_names,
                          function(x) {
                            a <- unlist(strsplit(x = as.character(x),split = ".",fixed = TRUE))
                            a[1]
                          })

```


Match names
```{r}
# library(dplyr)
matched_names <- sapply(ens_names_split,
                        function(x) {
                          # get the gene name
                          symbol <- unlist(subset(ensembl_annot,ensid %in% x)[,2])
                          # symbol <- unlist(subset(ensembl_annot,ensid %in% x)[,2])
                          symbol
                        })
```

```{r}
# some of these genes may not have been identified in our annotation, so we will omit them for now
matched_names_is_annotated <- sapply(matched_names,
                                     function(x) {
                                       length(x) > 0})
matched_names_annotated <- matched_names[matched_names_is_annotated]
matched_names_annotated_df <- as.data.frame(do.call(rbind,matched_names_annotated))
colnames(matched_names_annotated_df) <- "symbol"
human_rnaseq_dge_fit_coeff2_qlft$dex_symbol <- matched_names_annotated_df #data.frame(symbols = matched_names)
# match the annotations to the top-most genes
# human_rnaseq_dge_fit_coeff2_lrt_top$table$symbol <- 

```

Isolate the counts for the genes of interest
```{r}
human_rnaseq_dge_fit_coeff2_qlft_top_counts <- human_rnaseq_dge_fit_coeff2_qlft$table
```


## goana

```{r}
# need to debug kegga
golrt <- kegga(de = human_rnaseq_dge_fit_coeff2_qlft,geneid = human_rnaseq_dge_fit_coeff2_qlft$genes$genes)
# de = human_rnaseq_dge_fit_coeff2_lrt,
#               species="Hs")
topGO(golrt)
# why do pathways not overlap with the universe?
```

New from edgeR
```{}
human_rnaseq_dge_fit_coeff2_go <- goana(de = human_rnaseq_dge_fit_coeff2, species="Hs")
```

Some helpful code notes
```
vvv3@login04:~/pillaifolder/rmarkdown/tfh_diffn$ grep -s "gsva" *.Rmd
ranking_enhancerCluster_genes.Rmd:                             "pipelines/transcriptomics_pipeline/gsea/gsva_script.R"),
tfhdiff.Rmd:source("pipelines/transcriptomics_pipeline/gsea/gsva_script.R")
tfhdiff.Rmd:prepare_gsva()
tfhdiff.Rmd:tcell.genesets.tfh.enrich <- gsva_enrich(counts = tcell.enrich.counts$counts.aggregate,geneset = tcell.genesets)
vvv3@login04:~/pillaifolder/rmarkdown/tfh_diffn$ grep -s "tcell.genesets" *.Rmd
tfhdiff.Rmd:tcell.genesets <- import_gmt(gmtfile = "referenceAnnotation/hg38/genesets/msigdb/custom/msigdb_tcell_genesets.gmt")
tfhdiff.Rmd:tcell.genesets.tfh.enrich <- gsva_enrich(counts = tcell.enrich.counts$counts.aggregate,geneset = tcell.genesets)
tfhdiff.Rmd:#colnames(tcell.genesets.tfh.enrich$es.obs) <- c("CD4naive","PD1loCXCR5pos","PD1intCXCR5pos","PD1hiCXCR5hi")
tfhdiff.Rmd:tcell.genesets.tfh.enrich.es.kmeans <- kmeans_intervals(matrix = tcell.genesets.tfh.enrich$es.obs,kmax = 5,B=50)
tfhdiff.Rmd:tcell.genesets.tfh.enrich.es <- as.data.frame(tcell.genesets.tfh.enrich$es.obs)
tfhdiff.Rmd:tcell.genesets.tfh.enrich.es$k.choice <- tcell.genesets.tfh.enrich.es.kmeans$matrix.kmeans$kmeans
tfhdiff.Rmd:length(tcell.genesets.tfh.enrich.es$k.choice)
tfhdiff.Rmd:dim(tcell.genesets.tfh.enrich.es)
tfhdiff.Rmd:tcell.genesets.tfh.enrich.es <- tcell.genesets.tfh.enrich.es[order(tcell.genesets.tfh.enrich.es$k.choice),]
tfhdiff.Rmd:#enrichments <- tcell.genesets.tfh.enrich$es.obs[sort(tcell.genesets.tfh.enrich.es.kmeans$matrix.kmeans$kmeans),]
tfhdiff.Rmd:rowannotation.ssgsea <- data.frame(kchoice = as.factor(sort(tcell.genesets.tfh.enrich.es.kmeans$matrix.kmeans$kmeans)))
tfhdiff.Rmd:#rowannotation.ssgsea <- data.frame(kchoice = as.factor(tcell.genesets.tfh.enrich.es.kmeans$matrix.kmeans$kmeans))
tfhdiff.Rmd:head(tcell.genesets.tfh.enrich.es)
tfhdiff.Rmd:pheatmap(mat = tcell.genesets.tfh.enrich.es[,c(1:(ncol(tcell.genesets.tfh.enrich.es)-1))],
vvv3@login04:~/pillaifolder/rmarkdown/tfh_diffn$
```
