---
title: "Sheng and LaFleur 2018 RNA-seq reanalysis"
output: html_notebook
---

Sheng and LaGleur used RNA-seq. They aligned their data using STAR to hg19 (it appears ENSEMBL, judging from the gene names given), yielding Integer counts. They did differential expression analysis using edgeR. 

The RNA-seq methods description from their methods

```
Raw reads were aligned to hg19 or mm9 using STAR (v2.4.2a) with the parameter ‘‘quantMode’’ set as ‘‘GeneCounts’’ to produce count table for each gene. Differential gene analysis was performed on gene raw counts in R with edgeR package (v3.18.1) from bio- conductor. Read count table was filtered so that genes with at least one count across conditions were kept. The negative binomial generalized log-linear model was used in differential analysis. A FDR cut-off of 0.05 was used to determine significantly differentially expressed genes. The R package gProfileR (v0.6.4) was used to perform gene enrichment analysis on differential genes. Gene set enrichment analysis (GSEA) was performed with R package clusterProfiler (v3.4.4).
The function analyzeRepeats.pl from Homer (http://homer.ucsd.edu/homer/) software was used to get raw counts for repeats from RNA-seq data. Differential expression for repeats was performed with edgeR the same way as for genes.
```

This notebook depicts the re-analysis of human data

# Setup
```{r}
install.packages("readxl")
library(readxl)
source("https://bioconductor.org/biocLite.R")
biocLite("Biobase")
biocLite("GEOquery")
biocLite("limma")
biocLite("edgeR")
biocLite("gProfileR")
biocLite("GSVA")
biocLite("GO.db")
biocLite("org.Hs.eg.db")
biocLite("org.Mm.eg.db")
source("../../pipelines/gsva_script.R")
biocLite("EnsDb.Hsapiens.v86")
```

```{r}
library("Biobase")
library("GEOquery")
library("limma")
library("edgeR")
library("gProfileR")
library("GSVA")
library("GO.db")
library("org.Hs.eg.db")
library("org.Mm.eg.db")
library("EnsDb.Hsapiens.v86")
```


# Get data

## import datasets

```{r get_csv_counts}
human_rnaseq <- read.csv(file = "../../datasets/sheng_lafleur/GSE112230_2018-03-14_rnaseq_hs_counts.csv",header = TRUE,row.names = 1,sep = ",")
human_erv <- read.csv(file = "../../datasets/sheng_lafleur/GSE112230_2018-03-22_repeat_count_hs.csv",header = TRUE,row.names = 1,sep = ",")
```

## design

```{r}
#human_rnaseq_design <- data.frame()
human_rnaseq_design <- data.frame(cell = factor(rep("MCF7",ncol(human_rnaseq))))
rownames(human_rnaseq_design) <- colnames(human_rnaseq)
human_rnaseq_design$treatment <- factor(c(rep("shC",2),rep("shLSD1",2)))
human_rnaseq_design$rep <- factor(c(1,2,1,2))
print(human_rnaseq_design)

# model matrix
human_rnaseq_model_matrix <- model.matrix(~ treatment,human_rnaseq_design)
print(human_rnaseq_model_matrix)
# treat the sh-Control as the "baseline", so it gets an intercept term
```

# edgeR analysis

## dge list
set DGElist object
```{r dgelist}
human_rnaseq_dge <- DGEList(counts=human_rnaseq,
                            genes = data.frame(genes = rownames(human_rnaseq))) # set DGE list
human_rnaseq_dge <- calcNormFactors(object = human_rnaseq_dge) # set normalization factors
human_rnaseq_dge <- estimateDisp(y = human_rnaseq_dge,
                                 design = human_rnaseq_model_matrix) # set dispersion factors
plotBCV(y = human_rnaseq_dge)
# human_rnaseq_dge_exactTest <- exactTest(human_rnaseq_dge)
```

## glm
fit GLM
NEed to redo with glmFit
```{r}
human_rnaseq_dge_fit <- glmFit(human_rnaseq_dge, human_rnaseq_model_matrix)
```

## check conditions
Perform tests
```{r}
logFC_edgeR_lim <- 1
pval_cutoff <- 0.05
n_toptag <- 10000
# coef = 1 measures the baseline genes
human_rnaseq_dge_fit_coeff1_qlft <- glmQLFTest(glmfit = human_rnaseq_dge_fit,
                                               coef = 1)
# coef = 2 measures the fold-changes between the control and the fold change
human_rnaseq_dge_fit_coeff2_qlft <- glmQLFTest(glmfit = human_rnaseq_dge_fit,
                                               coef = 2,
                                               poisson.bound = TRUE)

# coef = 2, but we use a LRT. 
human_rnaseq_dge_fit_coeff2_lrt <- glmLRT(glmfit = human_rnaseq_dge_fit,
                                          coef = 2)

# coeff = 2 with "glmTreat". This option performs a fold-change cutoff
human_rnaseq_dge_fit_treat <- glmTreat(glmfit = human_rnaseq_dge_fit, 
                                       coef=2, 
                                       lfc=logFC_edgeR_lim)

# topTags for LRT; show top 10
human_rnaseq_dge_fit_coeff2_lrt_top <- topTags(human_rnaseq_dge_fit_coeff2_lrt,
                                               p.value = pval_cutoff,
                                               n = n_toptag,
                                               sort.by = "logFC")

human_rnaseq_dge_fit_coeff2_lrt_top_up <- human_rnaseq_dge_fit_coeff2_lrt_top

human_rnaseq_dge_fit_coeff2_lrt_top_up$table <- subset(human_rnaseq_dge_fit_coeff2_lrt_top$table,
                                                       abs(logFC) > logFC_edgeR_lim)

# human_rnaseq_dge_fit_coeff2_lrt_top <- topTags(object = human_rnaseq_dge_fit_coeff2_lrt,
#                                                p.value = pval_cutoff,
#                                                n = n_toptag,
#                                                adjust.method = "BH",sort.by = "PValue")
# human_rnaseq_dge_fit_coeff2_treat_top <- topTags(object = human_rnaseq_dge_fit_treat,
#                                                  p.value = pval_cutoff,
#                                                  adjust.method = "BH",
#                                                  n = n_toptag,sort.by = "logFC")

# plot the fold-changes-vs-pvalue and the top genes by the LRT
plot(human_rnaseq_dge_fit_coeff2_lrt$table$logFC,
     -log10(human_rnaseq_dge_fit_coeff2_lrt$table$PValue),
     pch=20)
points(human_rnaseq_dge_fit_coeff2_lrt_top_up$table$logFC,
       -log10(human_rnaseq_dge_fit_coeff2_lrt_top_up$table$PValue),
       col="red",pch=20)

# plotMD
plotMD(object = human_rnaseq_dge_fit_coeff2_lrt,adjust.method = "BH",p.value = pval_cutoff)
```

# GO analysis

## preparation
We must convert the ENSEMBL names to gene symbols
```{r convert_ens_names}
# get ENSEMBL annotation
ensembl_annot <- read.table(file = "../../datasets/transcripts/Homo_sapiens.GRCh38.91.names.txt",header = FALSE,sep = "\t")
ensembl_annot$V1 <- sapply(ensembl_annot$V1,
                           function(x) {
                             a <- unlist(strsplit(x = as.character(x),split = " "))
                             a[2]
                           })
ensembl_annot$V2 <- sapply(ensembl_annot$V2,
                           function(x) {
                             a <- unlist(strsplit(x = as.character(x),split = " "))
                             a[2]
                           })
colnames(ensembl_annot) <- c("ensid","symbol")

# obtain the symbols of the most differentially-expressed genes
# ens_names <- as.character(human_rnaseq_dge_fit_coeff2_lrt_top$table$genes)
ens_names <- as.character(human_rnaseq_dge_fit_coeff2_lrt_top_up$table$genes)

# split the given names
ens_names_split <- sapply(ens_names,
                          function(x) {
                            a <- unlist(strsplit(x = as.character(x),split = ".",fixed = TRUE))
                            a[1]
                          })

```


Match names
```{r}
# library(dplyr)
matched_names <- sapply(ens_names_split,
                        function(x) {
                          # get the gene name
                          symbol <- unlist(subset(ensembl_annot,ensid %in% x)[,2])
                          # symbol <- unlist(subset(ensembl_annot,ensid %in% x)[,2])
                          symbol
                        })
```

```{r}
# some of these genes may not have been identified in our annotation, so we will omit them for now
matched_names_is_annotated <- sapply(matched_names,
                                     function(x) {
                                       length(x) > 0})
matched_names_annotated <- matched_names[matched_names_is_annotated]
matched_names_annotated_df <- as.data.frame(do.call(rbind,matched_names_annotated))
colnames(matched_names_annotated_df) <- "symbol"
human_rnaseq_dge_fit_coeff2_lrt_top_up$dex_symbol <- matched_names_annotated_df #data.frame(symbols = matched_names)
human_rnaseq_dge_fit_coeff2_lrt_top_up$dex_ensg_symbol <- paste(rownames(matched_names_annotated_df),matched_names_annotated_df$symbol,sep = "_")

```

Isolate the counts for the genes of interest
```{r}
temp_counts <- human_rnaseq_dge_fit$counts
temp_counts <- temp_counts[rownames(temp_counts) %in% rownames(human_rnaseq_dge_fit_coeff2_lrt_top_up$dex_symbol),]
temp_counts_names <- sapply(rownames(temp_counts),
                            function (x) {
                              a.ind <- grep(pattern = x,
                                            x = human_rnaseq_dge_fit_coeff2_lrt_top_up$dex_ensg_symbol)
                              a <- human_rnaseq_dge_fit_coeff2_lrt_top_up$dex_ensg_symbol[a.ind]
                              a
                            })
rownames(temp_counts) <- unlist(temp_counts_names)
human_rnaseq_dge_fit_coeff2_lrt_top_up$counts <- temp_counts
pheatmap(log2(temp_counts+1),
         show_rownames = FALSE,
         annotation_col = human_rnaseq_design,
         show_colnames = FALSE)
```

## GSVA

Some helpful code notes
```
vvv3@login04:~/pillaifolder/rmarkdown/tfh_diffn$ grep -s "gsva" *.Rmd
ranking_enhancerCluster_genes.Rmd:                             "pipelines/transcriptomics_pipeline/gsea/gsva_script.R"),
tfhdiff.Rmd:source("pipelines/transcriptomics_pipeline/gsea/gsva_script.R")
tfhdiff.Rmd:prepare_gsva()
tfhdiff.Rmd:tcell.genesets.tfh.enrich <- gsva_enrich(counts = tcell.enrich.counts$counts.aggregate,geneset = tcell.genesets)
vvv3@login04:~/pillaifolder/rmarkdown/tfh_diffn$ grep -s "tcell.genesets" *.Rmd
tfhdiff.Rmd:tcell.genesets <- import_gmt(gmtfile = "referenceAnnotation/hg38/genesets/msigdb/custom/msigdb_tcell_genesets.gmt")
tfhdiff.Rmd:tcell.genesets.tfh.enrich <- gsva_enrich(counts = tcell.enrich.counts$counts.aggregate,geneset = tcell.genesets)
tfhdiff.Rmd:#colnames(tcell.genesets.tfh.enrich$es.obs) <- c("CD4naive","PD1loCXCR5pos","PD1intCXCR5pos","PD1hiCXCR5hi")
tfhdiff.Rmd:tcell.genesets.tfh.enrich.es.kmeans <- kmeans_intervals(matrix = tcell.genesets.tfh.enrich$es.obs,kmax = 5,B=50)
tfhdiff.Rmd:tcell.genesets.tfh.enrich.es <- as.data.frame(tcell.genesets.tfh.enrich$es.obs)
tfhdiff.Rmd:tcell.genesets.tfh.enrich.es$k.choice <- tcell.genesets.tfh.enrich.es.kmeans$matrix.kmeans$kmeans
tfhdiff.Rmd:length(tcell.genesets.tfh.enrich.es$k.choice)
tfhdiff.Rmd:dim(tcell.genesets.tfh.enrich.es)
tfhdiff.Rmd:tcell.genesets.tfh.enrich.es <- tcell.genesets.tfh.enrich.es[order(tcell.genesets.tfh.enrich.es$k.choice),]
tfhdiff.Rmd:#enrichments <- tcell.genesets.tfh.enrich$es.obs[sort(tcell.genesets.tfh.enrich.es.kmeans$matrix.kmeans$kmeans),]
tfhdiff.Rmd:rowannotation.ssgsea <- data.frame(kchoice = as.factor(sort(tcell.genesets.tfh.enrich.es.kmeans$matrix.kmeans$kmeans)))
tfhdiff.Rmd:#rowannotation.ssgsea <- data.frame(kchoice = as.factor(tcell.genesets.tfh.enrich.es.kmeans$matrix.kmeans$kmeans))
tfhdiff.Rmd:head(tcell.genesets.tfh.enrich.es)
tfhdiff.Rmd:pheatmap(mat = tcell.genesets.tfh.enrich.es[,c(1:(ncol(tcell.genesets.tfh.enrich.es)-1))],
vvv3@login04:~/pillaifolder/rmarkdown/tfh_diffn$
```

```{r}
# prepare for GSVA
prepare_gsva()
```

```{r}
# import some gene sets for GSVA
kegg_genesets <- import_gmt(gmtfile = "../../referenceAnnot/genesets/hg38/msigdb_c2.cp.kegg_hsapiens.gmt")
bp_genesets <- import_gmt(gmtfile = "../../referenceAnnot/genesets/hg38/msigdb_c5.bp_hsapiens.gmt")
cancer_genesets <- import_gmt(gmtfile = "../../referenceAnnot/genesets/hg38/msigdb_c6_hsapiens.gmt")
immune_genesets <- import_gmt(gmtfile = "../../referenceAnnot/genesets/hg38/msigdb_c7_hsapiens.gmt")
# from the Li et al oncotarget 2014 paper
interferon_genesets <- import_gmt(gmtfile = "../../referenceAnnot/genesets/hg38/custom/msigdb_interferon.gmt")
cytochemokine_genesets <- import_gmt(gmtfile = "../../referenceAnnot/genesets/hg38/custom/msigdb_cytochemokine.gmt")
inflammation_genesets <- import_gmt(gmtfile = "../../referenceAnnot/genesets/hg38/custom/msigdb_inflammation.gmt")
antigen_genesets <- import_gmt(gmtfile = "../../referenceAnnot/genesets/hg38/custom/msigdb_antigen.gmt")
# virus-related terms
virus_genesets <- import_gmt(gmtfile = "../../referenceAnnot/genesets/hg38/custom/msigdb_virus.gmt")

# hallmark sets--verified in microarrays
hallmark_genesets <- import_gmt(gmtfile = "../../referenceAnnot/genesets/hg38/msigdb_hallmark_hsapiens.gmt")


# import counts into gsva script
human_rnaseq_dge_fit_coeff2_lrt_top_up_gsva_counts <- import_counts(countsfile = human_rnaseq_dge_fit_coeff2_lrt_top_up$counts,
                                                                    imported_counts = TRUE)
# perform GSVA with KEGG
human_rnaseq_dge_fit_coeff2_lrt_top_up_kegg_enrich <- gsva_enrich(counts = human_rnaseq_dge_fit_coeff2_lrt_top_up_gsva_counts$counts.aggregate,
                                                                  geneset = kegg_genesets)
# perform GSVA with Cancer pathways
human_rnaseq_dge_fit_coeff2_lrt_top_up_cancer_enrich <- gsva_enrich(counts = human_rnaseq_dge_fit_coeff2_lrt_top_up_gsva_counts$counts.aggregate,
                                                                    geneset = cancer_genesets)
# perform GSVA with Immune pathways
human_rnaseq_dge_fit_coeff2_lrt_top_up_immune_enrich <- gsva_enrich(counts = human_rnaseq_dge_fit_coeff2_lrt_top_up_gsva_counts$counts.aggregate,
                                                                    geneset = immune_genesets)
# perform GSVA with Immune pathways
human_rnaseq_dge_fit_coeff2_lrt_top_up_bp_enrich <- gsva_enrich(counts = human_rnaseq_dge_fit_coeff2_lrt_top_up_gsva_counts$counts.aggregate,
                                                                geneset = bp_genesets)

# perform GSVA with GO terms from Li et al Oncotarget
interferon_enrich <- gsva_enrich(counts = human_rnaseq_dge_fit_coeff2_lrt_top_up_gsva_counts$counts.aggregate,
                                 geneset = interferon_genesets)
cytochemokine_enrich <- gsva_enrich(counts = human_rnaseq_dge_fit_coeff2_lrt_top_up_gsva_counts$counts.aggregate,
                                 geneset = cytochemokine_genesets)
inflammation_enrich <- gsva_enrich(counts = human_rnaseq_dge_fit_coeff2_lrt_top_up_gsva_counts$counts.aggregate,
                                 geneset = inflammation_genesets)
antigen_enrich <- gsva_enrich(counts = human_rnaseq_dge_fit_coeff2_lrt_top_up_gsva_counts$counts.aggregate,
                                 geneset = antigen_genesets)

# perform GSVA with viral terms
virus_enrich <- gsva_enrich(counts = human_rnaseq_dge_fit_coeff2_lrt_top_up_gsva_counts$counts.aggregate,
                                 geneset = virus_genesets)

# perform GSVA with hallmark terms
hallmark_enrich <- gsva_enrich(counts = human_rnaseq_dge_fit_coeff2_lrt_top_up_gsva_counts$counts.aggregate,
                                 geneset = hallmark_genesets)



```

Compact naming of pathways
```{r}
name_pathway <- function(x) {
  pathway_name <- sapply(x,
                         function(y) {
                           a <- unlist(strsplit(x = as.character(y),split = ":",fixed = TRUE))
                           a[2]
                         })
  return(pathway_name)
}
```

Set the terms of the signatures that we want
```{r}
# set li oncotarget terms
li_oncotarget_terms <- paste("interferon",
                             "cytokine",
                             "inflammation",
                             "influenza",
                             "testes",
                             "presentation",
                             sep = "|")
```

Plot enrichment
```{r}
pheatmap(human_rnaseq_dge_fit_coeff2_lrt_top_up_kegg_enrich,
         annotation_col = human_rnaseq_design,
         show_colnames = FALSE,
         fontsize = 5.0,
         main = "KEGG")
pheatmap(human_rnaseq_dge_fit_coeff2_lrt_top_up_cancer_enrich,
         annotation_col = human_rnaseq_design,
         show_colnames = FALSE,
         fontsize = 5.0,
         main = "Cancer")
set.seed(0)
kmeans_immune <- kmeans(x = human_rnaseq_dge_fit_coeff2_lrt_top_up_immune_enrich,
                        centers = 4)
kmeans_immune_annot <- as.data.frame(sort(kmeans_immune$cluster))
colnames(kmeans_immune_annot) <- c("cluster")

human_rnaseq_dge_fit_coeff2_lrt_top_up_immune_enrich <- human_rnaseq_dge_fit_coeff2_lrt_top_up_immune_enrich[rownames(kmeans_immune_annot),]

pheatmap(human_rnaseq_dge_fit_coeff2_lrt_top_up_immune_enrich,
         annotation_col = human_rnaseq_design,
         annotation_row = kmeans_immune_annot,
         cluster_rows = FALSE,
         show_rownames = FALSE,
         show_colnames = FALSE,
         fontsize = 5.0,
         main = "Immune")

```

```{r}
# highlight the immunological GO_BP terms
c5_bp_li_oncotarget <- grep(pattern = li_oncotarget_terms,
                            x = rownames(human_rnaseq_dge_fit_coeff2_lrt_top_up_bp_enrich),
                            ignore.case = TRUE)
c5_bp_not_li_oncotarget <- grep(pattern = li_oncotarget_terms,
                                x = rownames(human_rnaseq_dge_fit_coeff2_lrt_top_up_bp_enrich),
                                ignore.case = TRUE,invert = TRUE)
c5_bp_annot <- data.frame(li_oncotarget = c(rep("li_onco",length(c5_bp_li_oncotarget)),
                                            rep("non_li_onco",
                                                length(c5_bp_not_li_oncotarget))))
rownames(c5_bp_annot) <- c(rownames(human_rnaseq_dge_fit_coeff2_lrt_top_up_bp_enrich)[c5_bp_li_oncotarget],
                           rownames(human_rnaseq_dge_fit_coeff2_lrt_top_up_bp_enrich)[c5_bp_not_li_oncotarget])
#c5_bp_annot <- c5_bp_annot[order(c5_bp_annot$),]
c5_bp_sorted <- human_rnaseq_dge_fit_coeff2_lrt_top_up_bp_enrich[rownames(c5_bp_annot),]
pheatmap(c5_bp_sorted,
         annotation_col = human_rnaseq_design,
         annotation_row = c5_bp_annot,
         show_rownames = FALSE,
         show_colnames = FALSE,
         fontsize = 5.0,
         main = "BP")

```


Enrichments for terms from Li et al 2014 and Chiappinelli et al 2015.
```{r}

pheatmap(interferon_enrich,
         annotation_col = human_rnaseq_design,
         labels_row = rownames(interferon_enrich),
         show_colnames = FALSE,
         cluster_cols = TRUE,
         fontsize = 2.5,
         main = "Interferon GO Terms")
pheatmap(cytochemokine_enrich,
         annotation_col = human_rnaseq_design,
         labels_row = rownames(cytochemokine_enrich),
         show_colnames = FALSE,
         fontsize = 5.0,
         cluster_cols = FALSE,
         main = "Cytokine/Chemokine GO Terms")
pheatmap(inflammation_enrich,
         annotation_col = human_rnaseq_design,
         labels_row = rownames(inflammation_enrich),
         cluster_rows = FALSE,
         cluster_cols = FALSE,
         show_colnames = FALSE,
         fontsize = 5.0,
         main = "Inflammation GO terms")
pheatmap(antigen_enrich,
         annotation_col = human_rnaseq_design,
         labels_row = rownames(antigen_enrich),
         show_rownames = TRUE,
         fontsize = 4.0,
         show_colnames = FALSE,
         cluster_cols = FALSE,
         main = "Antigen Presentation and Receptor GO Terms")
pheatmap(virus_enrich,
         annotation_col = human_rnaseq_design,
         labels_row = rownames(virus_enrich),
         show_rownames = TRUE,
         show_colnames = FALSE,
         cluster_cols = FALSE,
         fontsize_row = 2.5,
         main = "Virus GO Terms")

```


Hallmarks
```{r}
pheatmap(hallmark_enrich,
         annotation_col = human_rnaseq_design,
         labels_row = rownames(hallmark_enrich),
         show_rownames = TRUE,
         show_colnames = FALSE,
         cluster_cols = FALSE,
         fontsize_row = 5.0,
         main = "Hallmark GO Terms")

```


We see some pathway enrichments.

# Replication of Sheng and LaFleur's gene signature

In the Sheng and LaFleur paper, the following procedure was used to get the ERV signature:

```
For analyzing LSD1 expression level versus IFN signature, genes were extracted from ‘virus’ and ‘interferon’ related terms of the biological pathways enriched in upregulated genes upon LSD1 KD in MCF7 cells. These genes were further filtered for those that are upregulated in KD condition to obtain a refined IFN signature gene list. Gene-wise z-score was calculated in each tumor type and the sum of genes in the refined IFN signature gene list was used to characterize IFN signature. Pearson’s correlation was used to evaluate the correlation of LSD1 level (z-score) and IFN signature
```

## Extract the 'virus' and 'interferon' signatures with strongest upregulation in shLSD1

Let's divide the "interferon" GSVA analysis further. 

```{r}

set.seed(0)
# should do a gap-statistic calculation
kmeans_interferon <- kmeans(x = interferon_enrich,
                            centers = 4)

kmeans_interferon_annot <- as.data.frame(sort(kmeans_interferon$cluster))
colnames(kmeans_interferon_annot) <- c("cluster")

interferon_enrich <- interferon_enrich[rownames(kmeans_interferon_annot),]

pheatmap(interferon_enrich,
         annotation_col = human_rnaseq_design,
         annotation_row = kmeans_interferon_annot,
         cluster_rows = FALSE,
         cluster_cols = FALSE,
         show_colnames = FALSE,
         fontsize = 3.0,
         main = "Interferon")

```


Specifically, let's look at the cluster with largely uniform increased expression of the pathways in the shLSD1 cases
```{r split_ifn}
chosen_ifn_k <- 2
interferon_enrich_subset <- interferon_enrich[rownames(subset(kmeans_interferon_annot,cluster %in% c(chosen_ifn_k))),]

pheatmap(interferon_enrich_subset,
         annotation_col = human_rnaseq_design,
         annotation_row = kmeans_interferon_annot,
         cluster_rows = FALSE,
         show_colnames = FALSE,
         fontsize = 5.0,
         main = "Interferon, enriched")

```


Let's do the same with the viral signatures
```{r split_viral}

set.seed(0)
# should do a gap-statistic calculation
kmeans_viral <- kmeans(x = virus_enrich,
                       centers = 4)

kmeans_viral_annot <- as.data.frame(sort(kmeans_viral$cluster))
colnames(kmeans_viral_annot) <- c("cluster")

virus_enrich <- virus_enrich[rownames(kmeans_viral_annot),]

pheatmap(virus_enrich,
         annotation_col = human_rnaseq_design,
         annotation_row = kmeans_viral_annot,
         cluster_rows = FALSE,
         show_colnames = FALSE,
         fontsize = 3.0,
         main = "Interferon")
```

```{r}
chosen_viral_k <- 2
viral_enrich_subset <- virus_enrich[rownames(subset(kmeans_viral_annot,cluster %in% c(chosen_viral_k))),]

pheatmap(viral_enrich_subset,
         annotation_col = human_rnaseq_design,
         annotation_row = kmeans_viral_annot,
         cluster_rows = FALSE,
         show_colnames = FALSE,
         fontsize = 5.0,
         main = "Viral")

```


Let's test if the enrichment differences between one set of samples and another for this cluster are significant. For this, we will bootstrap a set of enrichment values from the samples for each signature and construct a distribution of the differences in mean enrichment. 
```{r}
# virus
random_sig <- apply(X = virus_enrich,
                    MARGIN = 1,
                    function(x) {
                      set.seed(0)
                      a <- sample(x = x,size = length(x),replace = TRUE)
                      mean(a[3:4]) - mean(a[1:2])
                    })
hist(random_sig)
q_cutoffs <- c(mean(random_sig) - sd(random_sig)*qnorm(p = 0.025),
               mean(random_sig) + sd(random_sig)*qnorm(p = 0.025))
abline(v = q_cutoffs,
       lty=2,
       col="red")
# plot each of the other
test_sig <- apply(X = virus_enrich,
                  MARGIN = 1,
                  function(x) {
                    a <- mean(x[3:4]) - mean(x[1:2])
                    a_col <- "grey"
                    if (a >= q_cutoffs[1] | a <= q_cutoffs[2]) {
                      a_col <- "pink"
                    }
                    abline(v = a,col = a_col,lty=3)
                    a >= q_cutoffs[1] | a <= q_cutoffs[2]
                    })

kmeans_viral_annot$sig_boot <- as.factor(test_sig)
pheatmap(virus_enrich,
         annotation_col = human_rnaseq_design,
         annotation_row = kmeans_viral_annot,
         cluster_rows = FALSE,
         show_colnames = FALSE,
         fontsize = 3.0,
         main = "Virus")

# interferon
random_sig <- apply(X = interferon_enrich,
                    MARGIN = 1,
                    function(x) {
                      set.seed(0)
                      a <- sample(x = x,size = length(x),replace = TRUE)
                      mean(a[3:4]) - mean(a[1:2])
                    })
hist(random_sig)
q_cutoffs <- c(mean(random_sig) - sd(random_sig)*qnorm(p = 0.025),
               mean(random_sig) + sd(random_sig)*qnorm(p = 0.025))
abline(v = q_cutoffs,
       lty=2,
       col="red")
# plot each of the other
test_sig <- apply(X = interferon_enrich,
                  MARGIN = 1,
                  function(x) {
                    a <- mean(x[3:4]) - mean(x[1:2])
                    a_col <- "grey"
                    if (a >= q_cutoffs[1] | a <= q_cutoffs[2]) {
                      a_col <- "pink"
                    }
                    abline(v = a,col = a_col,lty=3)
                    a >= q_cutoffs[1] | a <= q_cutoffs[2]
                    })

kmeans_interferon_annot$sig_boot <- as.factor(test_sig)
pheatmap(interferon_enrich,
         annotation_col = human_rnaseq_design,
         annotation_row = kmeans_interferon_annot,
         cluster_rows = FALSE,
         show_colnames = FALSE,
         fontsize = 3.0,
         main = "Interferon")



```

Good; all of the signatures in the desired clusters have significant mean enrichment changes. 

## Extract the DEX genes in the upregulated 'virus' and 'interferon'

Combine the gene sets in the desired clusters
```{r}
viral_enriched_gos <- subset(kmeans_viral_annot,cluster %in% chosen_viral_k)
ifn_enriched_gos <- subset(kmeans_interferon_annot,cluster %in% chosen_ifn_k)
enriched_sig_gos <- unique(c(rownames(viral_enriched_gos),
                          rownames(ifn_enriched_gos))) # since there is some overlap
```

Identify the genes in the enriched, significantly-expressed GO terms
```{r}
# template dataset
all_viral_ifn_gos <- unique(do.call(c,list(virus_genesets,interferon_genesets)))
names(all_viral_ifn_gos) <- unique(c(names(virus_genesets),names(interferon_genesets)))
all_viral_ifn_gos <- all_viral_ifn_gos[names(all_viral_ifn_gos) %in% enriched_sig_gos]
all_viral_ifn_genes <- unique(unlist(all_viral_ifn_gos))
```

Obtain the differentially-expressed genes (from the sequencing experiment) that are present in these genes
```{r get_signature}
# get ifn and viral genes

dex_genes <- human_rnaseq_dge_fit_coeff2_lrt_top_up$dex_symbol
ifn_viral_genes <- subset(dex_genes,symbol %in% all_viral_ifn_genes)

# get those genes that are upregulated
# also impose a logCPM count limit of at least 0
upreg <- subset(human_rnaseq_dge_fit_coeff2_lrt_top_up$table,
                logFC > logFC_edgeR_lim & logCPM >= 0)
upreg <- subset(human_rnaseq_dge_fit_coeff2_lrt_top_up$table,
                logFC > logFC_edgeR_lim)

upreg_genes <- human_rnaseq_dge_fit_coeff2_lrt_top_up$dex_symbol[rownames(human_rnaseq_dge_fit_coeff2_lrt_top_up$dex_symbol) %in% upreg$genes,]
upreg_genes <- unlist(upreg_genes)

ifn_viral_signature <- subset(ifn_viral_genes,symbol %in% upreg_genes)

temp_aggregate_counts <- human_rnaseq_dge_fit_coeff2_lrt_top_up_gsva_counts$counts.aggregate
ifn_viral_signature_counts <- temp_aggregate_counts[rownames(temp_aggregate_counts) %in% ifn_viral_signature$symbol,]
pheatmap(log2(ifn_viral_signature_counts+1),
         show_rownames = FALSE,
         annotation_col = human_rnaseq_design,
         show_colnames = FALSE)

# plot distribution of signature fold changes
ifn_viral_signature_table <- subset(human_rnaseq_dge_fit_coeff2_lrt_top_up$table,genes %in% rownames(ifn_viral_signature))

hist(ifn_viral_signature_table$logCPM)
hist(ifn_viral_signature_table$logFC)

```

Save this signature

```{r}
write.table(x = ifn_viral_signature,
            file = "ifn_viral_signature_shenglafleur_hs.txt",
            quote = FALSE,sep = "\t",row.names = TRUE,col.names = TRUE)
write.table(x = ifn_viral_signature_table,
            file = "ifn_viral_signature_table_shenglafleur_hs.txt",
            quote = FALSE,sep = "\t",row.names = TRUE,col.names = TRUE)

```


# ERV analysis

