---
title: "Unified analysis of signatures"
output: html_notebook
---

We derived three signatures from each of the human datasets--Chiappinelli, Canadas, and Sheng/LaFleur. We want to know how similar these signatures, and specifically the IFN-ERV signature, are. 

# Import signatures

These are up-regulated signatures
```{r get_chiappinelli_sig}
# breast table
print(getwd())
chiappinelli_up_breast <- read.table(file = "./li_chiappinelli_oncotarget_2014/breast_up_genes_in_signature_table.txt",
                                  header = TRUE,
                                  sep = "\t",
                                  row.names = 1)
chiappinelli_down_breast <- read.table(file = "./li_chiappinelli_oncotarget_2014/breast_down_genes_in_signature_table.txt",
                                  header = TRUE,sep = "\t",row.names = 1)

# colon table
chiappinelli_up_colon <- read.table(file = "./li_chiappinelli_oncotarget_2014/colon_up_genes_in_signature_table.txt",
                                 header = TRUE,sep = "\t",row.names = 1)
chiappinelli_down_colon <- read.table(file = "./li_chiappinelli_oncotarget_2014/colon_down_genes_in_signature_table.txt",
                                  header = TRUE,sep = "\t",row.names = 1)

# ovarian table
chiappinelli_up_ovarian <- read.table(file = "li_chiappinelli_oncotarget_2014/ovarian_up_genes_in_signature_table.txt",
                                      header = TRUE,sep = "\t",row.names = 1)
chiappinelli_down_ovarian <- read.table(file = "li_chiappinelli_oncotarget_2014/ovarian_down_genes_in_signature_table.txt",
                                  header = TRUE,sep = "\t",row.names = 1)


# get the genes that go up and their average expressions
breast_ave_expr <- data.frame(avexpr=chiappinelli_up_breast$AveExpr)
breast_ave_expr.aggr <- aggregate.data.frame(x = breast_ave_expr,
                                             by = list(chiappinelli_up_breast$GENE_SYMBOL),
                                             FUN = mean)
colon_ave_expr <- data.frame(avexpr=chiappinelli_up_colon$AveExpr)
colon_ave_expr.aggr <- aggregate.data.frame(x = colon_ave_expr,
                                            by = list(chiappinelli_up_colon$GENE_SYMBOL),
                                            FUN = mean)
ovarian_ave_expr <- data.frame(avexpr=chiappinelli_up_ovarian$AveExpr)
ovarian_ave_expr.aggr <- aggregate.data.frame(x = ovarian_ave_expr,
                                              by = list(chiappinelli_up_ovarian$GENE_SYMBOL),
                                              FUN = mean)

# up signature
ifn_signature_up_genes_list <- read.table(file = "li_chiappinelli_oncotarget_2014/test_ifn_signature_probeInd_genes.txt",sep = "\t")
# down signature
ifn_signature_down_genes_list <- read.table(file = "li_chiappinelli_oncotarget_2014/test_down_ifn_signature_probeInd_genes.txt",sep = "\t")
# get unique signatures
ifn_signature_genes_list <- do.call(rbind,
                                    list(ifn_signature_up_genes_list,
                                         ifn_signature_down_genes_list))
chiappinelli_ifn_signature_unique_genes <- unique(ifn_signature_genes_list$gene)
# annotate these signatures
chiappinelli_ifn_signature_unique_genes_annot<- data.frame(is_up = chiappinelli_ifn_signature_unique_genes %in% ifn_signature_up_genes_list$gene,
                                                           is_down = chiappinelli_ifn_signature_unique_genes %in% ifn_signature_down_genes_list$gene)
chiappinelli_ifn_signature_unique_genes_annot$is_up <- factor(x = chiappinelli_ifn_signature_unique_genes_annot$is_up,
                                                              levels = c(FALSE,TRUE),
                                                              labels = c(0,1))
chiappinelli_ifn_signature_unique_genes_annot$is_down <- factor(x = chiappinelli_ifn_signature_unique_genes_annot$is_down,
                                                              levels = c(FALSE,TRUE),
                                                              labels = c(0,1))

rownames(chiappinelli_ifn_signature_unique_genes_annot) <- chiappinelli_ifn_signature_unique_genes
```

```{r get_canadas_sig}
# get all up-regulated genes that we designated
# write.table(x = tT_canadas_sig_up_in_sem,file = "./tT_canadas_sig_up_in_sem.txt",row.names = TRUE,col.names = TRUE,sep = "\t",quote = FALSE)
# write.table(x = tT_canadas_sig_up_h69m,file = "./tT_canadas_sig_up_h69m.txt",row.names = TRUE,col.names = TRUE,sep = "\t",quote = FALSE)
# write.table(x = tT_canadas_sig_up_h69m_in_sem,file = "./tT_canadas_sig_up_h69m_in_sem.txt",row.names = TRUE,col.names = TRUE,sep = "\t",quote = FALSE)
canadas_sig_nsclc <- read.table(file = "canadas_2018/tT_canadas_sig_up_h69m.txt",
                            header = TRUE,
                            row.names = 1,
                            sep = "\t")

# get the up-regulated genes that Canadas designated
canadas_sig_sem_nsclc <- read.table(file = "canadas_2018/tT_canadas_sig_up_in_sem.txt",
                            header = TRUE,
                            row.names = 1,
                            sep = "\t")

# get their SPARCs genes
canadas_sparcs_nsclc <- read.table(file = "canadas_2018/tT_canadas_sparcs.txt",
                            header = TRUE,
                            row.names = 1,
                            sep = "\t")

# get all genes with a 3' UTR
canadas_3putr_nsclc <- read.table(file = "canadas_2018/tT_canadas_erv_sig_all.txt",
                            header = TRUE,
                            row.names = 1,
                            sep = "\t")

# complete, SPARCS or non-SPARCS signature from canadas
canadas_ifn_signature_unique_genes <- unique(c(as.character(canadas_sig_nsclc$GENE_SYMBOL),
                                               as.character(canadas_sig_sem_nsclc$GENE_SYMBOL)))
canadas_ifn_signature_unique_genes <- canadas_ifn_signature_unique_genes[!is.na(canadas_ifn_signature_unique_genes)]
```

```{r get_shenglafleur_sig}
# get the human upregulated signatures
sheng_lafleur_signature_symbols <- read.table(file = "sheng_lafleur_2018/ifn_viral_signature_shenglafleur_hs.txt",
                                              header = TRUE,
                                              row.names = 1,
                                              sep = "\t")
sheng_lafleur_signature <- as.character(sheng_lafleur_signature_symbols$symbol)
```

# Measure intersections

```{r vennDiag}
# install.package("VennDiagram")
# draw a Venn Diagram
library(VennDiagram)
chiap_canad_venn <- venn.diagram(x = list(chiappinelli = as.character(chiappinelli_ifn_signature_unique_genes),
                                          canadas = canadas_ifn_signature_unique_genes,
                                          sheng_lafleur = sheng_lafleur_signature),
                                 filename = "./canadas_chiappinelli_intersect.jpg")

```

```{r}
unified_sig <- unique(c(as.character(chiappinelli_ifn_signature_unique_genes),
                        canadas_ifn_signature_unique_genes,
                        sheng_lafleur_signature))
# assemble the table of intersections
unified_sig_intersection <- data.frame(chiappinelli = (unified_sig %in% as.character(chiappinelli_ifn_signature_unique_genes)))
unified_sig_intersection$canadas <- (unified_sig %in% canadas_ifn_signature_unique_genes)
unified_sig_intersection$shenglafleur <- (unified_sig %in% sheng_lafleur_signature)
rownames(unified_sig_intersection) <- unified_sig
# pheatmap(unified_sig_intersection)

unified_sig_intersection_all <- subset(unified_sig_intersection,
                                   chiappinelli & canadas & shenglafleur)
unified_sig_intersection_chia_sheng <- subset(unified_sig_intersection,
                                              chiappinelli & shenglafleur & !canadas)
unified_sig_intersection_canad_sheng <- subset(unified_sig_intersection,
                                              canadas & shenglafleur & !chiappinelli)
unified_sig_intersection_chia_canad <- subset(unified_sig_intersection,
                                              chiappinelli & canadas & !shenglafleur)

# save the table
write.table(x = unified_sig_intersection_all,
            file = "./unified_sig_intersection_all.txt",
            sep = "\t",
            col.names = TRUE,
            row.names = TRUE)
print(unified_sig_intersection_all)
print(unified_sig_intersection_chia_sheng)
print(unified_sig_intersection_canad_sheng)
print(unified_sig_intersection_chia_canad)

```

The studies have very little overlap with each other. The four genes conserved across all the conditions are involved in endogenous peptide presentation (TAP2 loads peptides onto MHCI complex). The individual pairwise matches show different components of the interferon response and innate immunity up-regulated.


# TCGA re-analysis

## Get GDAC data
Using the different intersections of signatures, we want to re-cluster TCGA data and find if we can group tumors just as well. Data was downloaded from GDAC firehose
```{r gdac}

brca_gdac <- read.table(file = "../datasets/gdac_firehose/gdac.broadinstitute.org_BRCA.Merge_rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.Level_3.2016012800.0.0/BRCA.rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.data.txt",header = TRUE,sep = "\t",row.names = 1)
# save(brca_gdac, file = "./data/RNAseq/brcaRNAseq.RData")

# Colon
coad_gdac <- read.table(file = "../datasets/gdac_firehose/gdac.broadinstitute.org_COAD.Merge_rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.Level_3.2016012800.0.0/COAD.rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.data.txt",header = TRUE,sep = "\t",row.names = 1)
# save(coad_gdac, file = "./data/RNAseq/coadRNAseq.RData")

# Lung
lusc_gdac <- read.table(file = "../datasets/gdac_firehose/gdac.broadinstitute.org_LUSC.Merge_rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.Level_3.2016012800.0.0/LUSC.rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.data.txt",header = TRUE,sep = "\t",row.names = 1)
# save(lusc_gdac, file = "./data/RNAseq/luscRNAseq.RData")

# Ovarian
ov_gdac <- read.table(file = "../datasets/gdac_firehose/gdac.broadinstitute.org_OV.Merge_rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.Level_3.2016012800.0.0/OV.rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.data.txt",header = TRUE,sep = "\t",row.names = 1)
# save(ov_gdac, file = "./data/RNAseq/ovRNAseq.RData")

# save cancer files
gdac_cancer_files <- list(brca = brca_gdac,
                          coad = coad_gdac,
                          lusc = lusc_gdac,
                          ov = ov_gdac)

```

Compress into a single data frame
```{r}
gdac_selected_cancers <- as.data.frame(do.call(what = cbind,args = gdac_cancer_files))
gdac_selected_cancers_annot <- data.frame(TCGA = c(rep("BRCA",length(brca_gdac)),
                                                   rep("COAD",length(coad_gdac)),
                                                   rep("LUSC",length(lusc_gdac)),
                                                   rep("OV",length(ov_gdac))
                                                   ))
rownames(gdac_selected_cancers_annot) <- colnames(gdac_selected_cancers)
```


## Identify expression of each IFN-viral gene signature

### Chiappinelli 
```{r chiappinelli_tcga}

chiappinelli_ifn_signature_unique_genes <- as.character(chiappinelli_ifn_signature_unique_genes)

chiappinelli_ifn_signature_tcga <- sapply(chiappinelli_ifn_signature_unique_genes,
                                          function(g) {
                                            a <- grepl(pattern = as.character(paste0("^",g,"\\|")),
                                                       x = rownames(gdac_selected_cancers),
                                                       ignore.case = FALSE)
                                            rownames(gdac_selected_cancers)[a]
                                          })
gdac_selected_cancers.chiappinelli_tcga <- gdac_selected_cancers[rownames(gdac_selected_cancers) %in% 
                                                                   unlist(chiappinelli_ifn_signature_tcga),]
rownames(gdac_selected_cancers.chiappinelli_tcga) <- sapply(rownames(gdac_selected_cancers.chiappinelli_tcga),
                                                            function(x) {
                                                              a <- unlist(strsplit(x = x,split = "|",fixed = TRUE))
                                                              a[1]
                                                            })
#rownames(gdac_selected_cancers.chiappinelli_tcga) <- names(unlist(chiappinelli_ifn_signature_tcga))

gdac_selected_cancers.chiappinelli_tcga_mat <- apply(X = gdac_selected_cancers.chiappinelli_tcga,
                                                 MARGIN = 2,
                                                 FUN = function(x) {
                                                   as.numeric(x)
                                                 })
rownames(gdac_selected_cancers.chiappinelli_tcga_mat) <- rownames(gdac_selected_cancers.chiappinelli_tcga)
  #grep(pattern = paste0(chiappinelli_ifn_signature_unique_genes,"|"),x = rownames(gdac_selected_cancers),ignore.case = TRUE)

# gdac_selected_cancers.chiappinelli <- gdac_selected_cancers[]

```

```{r chiappinelli_tcga_plot}
chiappinelli_tcga_clust <- pheatmap(mat = log10(gdac_selected_cancers.chiappinelli_tcga_mat+1),
         show_colnames = FALSE,
         show_rownames = TRUE,
         fontsize_row = 2.5,
         annotation_col = gdac_selected_cancers_annot,
         annotation_row = chiappinelli_ifn_signature_unique_genes_annot,
         legend_labels = c("log10(RSEM-scaled counts+1)"),
         cutree_rows = 4,
         main = c(paste("DEX genes from Li et al 2014 and Chiappinelli et al 2015\nin",ncol(gdac_selected_cancers.chiappinelli_tcga_mat),"TCGA samples")))
# plot(chiappinelli_tcga_clust)
```

the specific ERV signature?

### Canadas

```{r chiappinelli_tcga}

canadas_ifn_signature_unique_genes <- as.character(canadas_ifn_signature_unique_genes)

canadas_ifn_signature_tcga <- sapply(canadas_ifn_signature_unique_genes,
                                          function(g) {
                                            a <- grepl(pattern = as.character(paste0("^",g,"\\|")),
                                                       x = rownames(gdac_selected_cancers),
                                                       ignore.case = FALSE)
                                            rownames(gdac_selected_cancers)[a]
                                          })
gdac_selected_cancers.canadas_tcga <- gdac_selected_cancers[rownames(gdac_selected_cancers) %in% 
                                                                   unlist(canadas_ifn_signature_tcga),]
canadas_split_names <- sapply(rownames(gdac_selected_cancers.canadas_tcga),
                                                            function(x) {
                                                              a <- unlist(strsplit(x = x,
                                                                                   split = "|",fixed = TRUE))
                                                              a[1]
                                                            })

gdac_selected_cancers.canadas_tcga_mat <- apply(X = gdac_selected_cancers.canadas_tcga,
                                                 MARGIN = 2,
                                                 FUN = function(x) {
                                                   as.numeric(x)
                                                 })

rownames(gdac_selected_cancers.canadas_tcga_mat) <- as.character(canadas_split_names)

rownames(gdac_selected_cancers.canadas_tcga_mat) <- as.character(canadas_split_names) # rownames(gdac_selected_cancers.canadas_tcga)
  #grep(pattern = paste0(chiappinelli_ifn_signature_unique_genes,"|"),x = rownames(gdac_selected_cancers),ignore.case = TRUE)

# gdac_selected_cancers.chiappinelli <- gdac_selected_cancers[]

```

```{r chiappinelli_tcga_plot}
canadas_tcga_clust <- pheatmap(mat = log10(gdac_selected_cancers.canadas_tcga_mat+1),
         show_colnames = FALSE,
         show_rownames = TRUE,
         fontsize_row = 2.5,
         annotation_col = gdac_selected_cancers_annot,
         legend_labels = c("log10(RSEM-scaled counts+1)"),
         cutree_rows = 4,
         main = c(paste("DEX genes from Canadas et al 2014 and Canadas et al 2018\nin",ncol(gdac_selected_cancers.chiappinelli_tcga_mat),"TCGA samples")))
# plot(chiappinelli_tcga_clust)
```

Cluster 2 seems to be a "lung inflammatory" sub-signature

```{r}
gdac_selected_cancers.canadas_tcga_mat_3putr <- gdac_selected_cancers.canadas_tcga_mat[rownames(gdac_selected_cancers.canadas_tcga_mat) %in% 
                                                                                         canadas_3putr_nsclc$GENE_SYMBOL,] 
canadas_3putrerv_tcga_clust <- pheatmap(mat = log10(gdac_selected_cancers.canadas_tcga_mat_3putr+1),
         show_colnames = FALSE,
         show_rownames = TRUE,
         fontsize_row = 5.0,
         annotation_col = gdac_selected_cancers_annot,
         legend_labels = c("log10(RSEM-scaled counts+1)"),
         main = c(paste("3' UTR ERV-containing DEX genes from Canadas et al 2014 and\nCanadas et al 2018 in",ncol(gdac_selected_cancers.chiappinelli_tcga_mat),"TCGA samples")))
# plot(chiappinelli_tcga_clust)
```

Most of the 3' UTR ERV-containing DEX genes are uniformly increased in expression across the tumors


### Sheng and LaFleur

All Sheng and LaFleur DEX?

Sheng and LaFleur signature
```{r chiappinelli_tcga}

sheng_lafleur_signature_unique_genes <- as.character(sheng_lafleur_signature)

sheng_lafleur_signature_tcga <- sapply(sheng_lafleur_signature_unique_genes,
                                          function(g) {
                                            a <- grepl(pattern = as.character(paste0("^",g,"\\|")),
                                                       x = rownames(gdac_selected_cancers),
                                                       ignore.case = FALSE)
                                            rownames(gdac_selected_cancers)[a]
                                          })
gdac_selected_cancers.sheng_lafleur_tcga <- gdac_selected_cancers[rownames(gdac_selected_cancers) %in% 
                                                                   unlist(sheng_lafleur_signature_tcga),]

gdac_selected_cancers.sheng_lafleur_tcga_mat <- apply(X = gdac_selected_cancers.sheng_lafleur_tcga,
                                                 MARGIN = 2,
                                                 FUN = function(x) {
                                                   as.numeric(x)
                                                 })

rownames(gdac_selected_cancers.sheng_lafleur_tcga_mat) <- sapply(rownames(gdac_selected_cancers.sheng_lafleur_tcga),
                                                            function(x) {
                                                              a <- unlist(strsplit(x = x,split = "|",fixed = TRUE))
                                                              a[1]
                                                            })

# rownames(gdac_selected_cancers.sheng_lafleur_tcga_mat) <- rownames(gdac_selected_cancers.sheng_lafleur_tcga)
  #grep(pattern = paste0(chiappinelli_ifn_signature_unique_genes,"|"),x = rownames(gdac_selected_cancers),ignore.case = TRUE)

# gdac_selected_cancers.chiappinelli <- gdac_selected_cancers[]

```

```{r sheng_lafleur_plot}
sheng_lafleur_tcga_clust <- pheatmap(mat = log10(gdac_selected_cancers.sheng_lafleur_tcga_mat+1),
         show_colnames = FALSE,
         show_rownames = TRUE,
         fontsize_row = 2.5,
         annotation_col = gdac_selected_cancers_annot,
         legend_labels = c("log10(RSEM-scaled counts+1)"),
         cutree_rows = 5,
         main = c(paste("DEX genes from Sheng and LaFleur et al 2018\nin",ncol(gdac_selected_cancers.chiappinelli_tcga_mat),"TCGA samples")))
# plot(chiappinelli_tcga_clust)
```


The "up-regulation" and "down-regulation" subsets from the cell lines don't exactly predict the "high" and "low" subsets in the patients. How consistent is the classifier?


### Consolidated signature

```{r unified_tcga}


unified_signature_tcga <- sapply(rownames(unified_sig_intersection),
                                          function(g) {
                                            a <- grepl(pattern = as.character(paste0("^",g,"\\|")),
                                                       x = rownames(gdac_selected_cancers),
                                                       ignore.case = FALSE)
                                            rownames(gdac_selected_cancers)[a]
                                          })
gdac_selected_cancers.unified_signature_tcga <- gdac_selected_cancers[rownames(gdac_selected_cancers) %in% 
                                                                   unlist(unified_signature_tcga),]

gdac_selected_cancers.unified_signature_tcga_mat <- apply(X = gdac_selected_cancers.unified_signature_tcga,
                                                 MARGIN = 2,
                                                 FUN = function(x) {
                                                   as.numeric(x)
                                                 })

rownames(gdac_selected_cancers.unified_signature_tcga_mat) <- unlist(sapply(rownames(gdac_selected_cancers.unified_signature_tcga),
                                                            function(x) {
                                                              a <- unlist(strsplit(x = x,split = "|",fixed = TRUE))
                                                              a[1]
                                                            }))

# rownames(gdac_selected_cancers.sheng_lafleur_tcga_mat) <- rownames(gdac_selected_cancers.sheng_lafleur_tcga)
  #grep(pattern = paste0(chiappinelli_ifn_signature_unique_genes,"|"),x = rownames(gdac_selected_cancers),ignore.case = TRUE)

# gdac_selected_cancers.chiappinelli <- gdac_selected_cancers[]

```

```{r unified_row_annot}

unified_sig_intersection$chiappinelli <- factor(x = unified_sig_intersection$chiappinelli,levels = c(FALSE,TRUE),labels = c("no","yes"))
unified_sig_intersection$canadas <- factor(x = unified_sig_intersection$canadas,levels = c(FALSE,TRUE),labels = c("no","yes"))
unified_sig_intersection$shenglafleur <- factor(x = unified_sig_intersection$shenglafleur,levels = c(FALSE,TRUE),labels = c("no","yes"))
```

```{r unified_row_plot}
unified_sig_intersection <- unified_sig_intersection[order(unified_sig_intersection$chiappinelli,
                                                           unified_sig_intersection$canadas,
                                                           unified_sig_intersection$shenglafleur),]

set.seed(0)
nhclust <- 4
unified_tcga_clust <- pheatmap(mat = log10(gdac_selected_cancers.unified_signature_tcga_mat+1),
         show_colnames = FALSE,
         show_rownames = FALSE,
         fontsize_row = 2.5,
         annotation_col = gdac_selected_cancers_annot,
         annotation_row = unified_sig_intersection,
         cutree_rows = 4,
         legend_labels = c("log10(RSEM-scaled counts+1)"),
         main = c(paste("DEX genes from Sheng and LaFleur et al 2018\nin",ncol(gdac_selected_cancers.chiappinelli_tcga_mat),"TCGA samples")))

unified_tcga_clust_hclusters <- cutree(tree = unified_tcga_clust$tree_row,k = nhclust)

# plot(chiappinelli_tcga_clust)
```

We get four different "TCGA-IFN-viral" clusters. Cluster 4 (smallest) is uniformly increased across all of the samples, while cluster 3 (mostly blue) is uniformly decreased. 

Top-most cluster
```{r top_cluster}
highest_cluster_unified_signature <- unified_tcga_clust_hclusters[unified_tcga_clust_hclusters %in% 4]
lowest_cluster_unified_signature <- unified_tcga_clust_hclusters[unified_tcga_clust_hclusters %in% 3]

gdac_selected_cancers.unified_signature_tcga_mat.top_vs_bottom <- gdac_selected_cancers.unified_signature_tcga_mat[rownames(gdac_selected_cancers.unified_signature_tcga_mat) %in% 
                                                                                         names(c(highest_cluster_unified_signature,lowest_cluster_unified_signature)),] 
rowannot_top_bottom_clusters <- data.frame(cluster = as.factor(c(highest_cluster_unified_signature,
                                                                 lowest_cluster_unified_signature)))
rownames(rowannot_top_bottom_clusters) <- names(c(highest_cluster_unified_signature,
                                                     lowest_cluster_unified_signature))

unified_signature_tcga_mat.top_vs_bottom_clust <- pheatmap(mat = log10(gdac_selected_cancers.unified_signature_tcga_mat.top_vs_bottom+1),
         show_colnames = FALSE,
         show_rownames = TRUE,
         cluster_cols = FALSE,
         fontsize_row = 5.0,
         annotation_col = gdac_selected_cancers_annot,
         annotation_row = rowannot_top_bottom_clusters,
         cutree_rows = 3,
         legend_labels = c("log10(RSEM-scaled counts+1)"),
         main = c(paste("Highest- and lowest-expressed gene sets of the",nrow(unified_sig_intersection),
                        "in\n",ncol(gdac_selected_cancers.chiappinelli_tcga_mat),"TCGA samples")))
# plot(chiappinelli_tcga_clust)


var1_cluster_unified_signature <- unified_tcga_clust_hclusters[unified_tcga_clust_hclusters %in% 1]
var2_cluster_unified_signature <- unified_tcga_clust_hclusters[unified_tcga_clust_hclusters %in% 2]

gdac_selected_cancers.unified_signature_tcga_mat.variable_exp <- gdac_selected_cancers.unified_signature_tcga_mat[rownames(gdac_selected_cancers.unified_signature_tcga_mat) %in% 
                                                                                         names(c(var1_cluster_unified_signature,
                                                                                                 var2_cluster_unified_signature)),] 
rowannot_variable_clusters <- data.frame(cluster = as.factor(c(var1_cluster_unified_signature,
                                                               var2_cluster_unified_signature)))
rownames(rowannot_variable_clusters) <- names(c(var1_cluster_unified_signature,
                                                var2_cluster_unified_signature))
unified_signature_tcga_mat.top_vs_bottom_clust <- pheatmap(mat = log10(gdac_selected_cancers.unified_signature_tcga_mat.variable_exp+1),
         show_colnames = FALSE,
         show_rownames = TRUE,
         cluster_cols = FALSE,
         fontsize_row = 2.5,
         annotation_col = gdac_selected_cancers_annot,
         annotation_row = rowannot_variable_clusters,
         cutree_rows = 3,
         legend_labels = c("log10(RSEM-scaled counts+1)"),
         main = c(paste("Middle-expressed gene sets of the",nrow(unified_sig_intersection),
                        "in\n",ncol(gdac_selected_cancers.chiappinelli_tcga_mat),"TCGA samples")))
# plot(chiappinelli_tcga_clust)

```

```{r}
print("highest")
print(names(highest_cluster_unified_signature)) # mostly housekeeping stuff... 
print("lowest")
print(names(lowest_cluster_unified_signature))
```

### GSVA of the Consolidated signatures' four clusters

How enriched are the four main TCGA-IFN-viral signatures for different gene sets? 
```{r get_genesets}
prepare_gsva()
# from the Li et al oncotarget 2014 paper
interferon_genesets <- import_gmt(gmtfile = "../referenceAnnot/genesets/hg38/custom/msigdb_interferon.gmt")
cytochemokine_genesets <- import_gmt(gmtfile = "../referenceAnnot/genesets/hg38/custom/msigdb_cytochemokine.gmt")
inflammation_genesets <- import_gmt(gmtfile = "../referenceAnnot/genesets/hg38/custom/msigdb_inflammation.gmt")
antigen_genesets <- import_gmt(gmtfile = "../referenceAnnot/genesets/hg38/custom/msigdb_antigen.gmt")
# virus-related terms
virus_genesets <- import_gmt(gmtfile = "../referenceAnnot/genesets/hg38/custom/msigdb_virus.gmt")

```

Prepare counts
```{r}
med1_unified_sig <- unified_tcga_clust_hclusters[unified_tcga_clust_hclusters %in% 1]
med2_unified_sig <- unified_tcga_clust_hclusters[unified_tcga_clust_hclusters %in% 2]
lowest_unified_sig <- unified_tcga_clust_hclusters[unified_tcga_clust_hclusters %in% 3]
highest_unified_sig <- unified_tcga_clust_hclusters[unified_tcga_clust_hclusters %in% 4]

med1_unified_counts <- gdac_selected_cancers.unified_signature_tcga_mat[rownames(gdac_selected_cancers.unified_signature_tcga_mat) %in% 
                                                                          names(med1_unified_sig),]
med2_unified_counts <- gdac_selected_cancers.unified_signature_tcga_mat[rownames(gdac_selected_cancers.unified_signature_tcga_mat) %in% 
                                                                          names(med2_unified_sig),]
lowest_unified_counts <- gdac_selected_cancers.unified_signature_tcga_mat[rownames(gdac_selected_cancers.unified_signature_tcga_mat) %in% 
                                                                          names(lowest_unified_sig),]
highest_unified_counts <- gdac_selected_cancers.unified_signature_tcga_mat[rownames(gdac_selected_cancers.unified_signature_tcga_mat) %in% 
                                                                          names(highest_unified_sig),]

tcga_cluster_counts <- list(highest=highest_unified_counts,
                            med2=med2_unified_counts,
                            med1=med1_unified_counts,
                            lowest=highest_unified_counts)
```


```{}

tcga_gsva_enrich_by_clus <- lapply(seq_along(along.with = tcga_cluster_counts),
                                   function(x) {
                                     X <- tcga_cluster_counts[[x]]
                                     # IFN-related signatures
                                     ifn <- gsva_enrich(counts = X,
                                                        geneset = interferon_genesets)
                                     pheatmap(mat = ifn,
                                              annotation_col = gdac_selected_cancers_annot,
                                              show_colnames = FALSE,
                                              fontsize = 3.0,
                                              main=paste("IFN, cluster",names(tcga_cluster_counts)[x]))
                                     # # cytokine and chemokine-related signatures
                                     # cytochemo <- gsva_enrich(counts = X,
                                     #                    geneset = cytochemokine_genesets)
                                     # pheatmap(mat = cytochemo,
                                     #          annotation_col = gdac_selected_cancers_annot,
                                     #          show_colnames = FALSE,
                                     #          fontsize = 3.0,
                                     #          main=paste("Cytokine/Chemokine, cluster",names(tcga_cluster_counts)[x]))
                                     
                                     # inflammation signatures
                                     # inflamm <- gsva_enrich(counts = X,
                                     #                    geneset = inflammation_genesets)
                                     # pheatmap(mat = inflamm,
                                     #          annotation_col = gdac_selected_cancers_annot,
                                     #          show_colnames = FALSE,
                                     #          fontsize = 3.0,
                                     #          main=paste("Inflammation, cluster",names(tcga_cluster_counts)[x]))
                                     # # antigen presentation signatures
                                     # agpres <- gsva_enrich(counts = X,
                                     #                    geneset = antigen_genesets)
                                     # pheatmap(mat = agpres,
                                     #          annotation_col = gdac_selected_cancers_annot,
                                     #          show_colnames = FALSE,
                                     #          fontsize = 3.0,
                                     #          main=paste("Antigen presentation, cluster",names(tcga_cluster_counts)[x]))
                                     # 
                                     
                                     # viral <- gsva_enrich(counts = X,
                                     #                    geneset = virus_genesets)
                                     # pheatmap(mat = viral,
                                     #          annotation_col = gdac_selected_cancers_annot,
                                     #          show_colnames = FALSE,
                                     #          fontsize = 3.0,
                                     #          main=paste("Virus-related GOs, cluster",names(tcga_cluster_counts)[x]))
                                     # list(ifn,
                                     #      cytochemo,
                                     #      inflamm,
                                     #      agpres,
                                     #      viral)
                                   })

# # perform GSVA with GO terms from Li et al Oncotarget
# tcga_interferon_enrich <- gsva_enrich(counts = human_rnaseq_dge_fit_coeff2_lrt_top_up_gsva_counts$counts.aggregate,
#                                  geneset = interferon_genesets)
# tcga_cytochemokine_enrich <- gsva_enrich(counts = human_rnaseq_dge_fit_coeff2_lrt_top_up_gsva_counts$counts.aggregate,
#                                  geneset = cytochemokine_genesets)
# tcga_inflammation_enrich <- gsva_enrich(counts = human_rnaseq_dge_fit_coeff2_lrt_top_up_gsva_counts$counts.aggregate,
#                                  geneset = inflammation_genesets)
# tcga_antigen_enrich <- gsva_enrich(counts = human_rnaseq_dge_fit_coeff2_lrt_top_up_gsva_counts$counts.aggregate,
#                                  geneset = antigen_genesets)
# 
# # perform GSVA with viral terms
# tcga_virus_enrich <- gsva_enrich(counts = human_rnaseq_dge_fit_coeff2_lrt_top_up_gsva_counts$counts.aggregate,
#                                  geneset = virus_genesets)

```

Perform the GSVA enrichment for IFN-related signatures analysis
```{}
ifn_highest <- gsva_enrich(counts = highest_unified_counts,
                           geneset = interferon_genesets) 
pheatmap(mat = ifn_highest,
         annotation_col = gdac_selected_cancers_annot,
         show_colnames = FALSE,
         fontsize = 3.0,
         main=paste("IFN, highest-expression cluster"))

ifn_med1 <- gsva_enrich(counts = med2_unified_counts,
                           geneset = interferon_genesets) # error is here
pheatmap(mat = ifn_med1,
         annotation_col = gdac_selected_cancers_annot,
         show_colnames = FALSE,
         fontsize = 3.0,
         main=paste("IFN, medium-expression cluster #1"))

ifn_med1 <- gsva_enrich(counts = med1_unified_counts,
                           geneset = interferon_genesets)
pheatmap(mat = ifn_med1,
         annotation_col = gdac_selected_cancers_annot,
         show_colnames = FALSE,
         fontsize = 3.0,
         main=paste("IFN, medium-expression cluster #2"))

ifn_lowest <- gsva_enrich(counts = lowest_unified_counts,
                           geneset = interferon_genesets)
pheatmap(mat = ifn_lowest,
         annotation_col = gdac_selected_cancers_annot,
         show_colnames = FALSE,
         fontsize = 3.0,
         main=paste("IFN, lowest-expression cluster"))


```



# How mutually informative are the signatures

Finally, we would like to know how our three paper-specific IFN-viral signatures and our four TCGA-IFN-viral clusters. We can determine how correlated are the estimates of every pair of signatures (21 different pairs) for a particular tumor type. We use a bootstrapping procedure. 

Let $x_i = x_{i1} ... x_{in}$ and $x_j = x_{j1} ... x_{jm}$ represent the expression values of two signatures $i$ and $j$. We want to test how well-correlated would these signatures' mean and median values be for a tumor type. We draw $R$ bootstrap samples (say that $R = 100$) for each tumor sample. In each bootstrap sample, we draw $K$ pairs of medians (say $K=30$) from $n$ randomly-sampled gene expression values from one signature and $m$ randomly-sampled gene expression values from the other signature. We obtain the correlation coefficient across these $K$ pairs, and ultimately we obtain $R$ correlation coefficients (for which we obtain the mean correlation and the values at the 2.5th and 97.5th percentiles). We repeat this bootstrapping procedure for each tumor sample. 

The proposed elaborate bootstrapping procedure is intended to prevent batch-level effects within a particular tumor type, but we can modify it for a more simplified sampling procedure. Instead of performing $R \times K \times (n + m)$ samples for $N \approx 3000$ tumors (!), we will instead sample our $n$ and $m$ gene signature values across all tumor samples of a particular type so that we only have to repeat $R \times K \times (n + m)$ for 4 different tumor types (TCGA has 27 types in total, but we are just focused on 4 for now). We will have a faster procedure, but we will obtain estimates of signature correlations for a broad tumor type (and not for particular samples).

```{r signature_bootstrapping_fn}

signature_median <- function(sig1,sig2) {
  # draw the random sample from signature 1
  n_sig1 <- length(sig1)
  n_sig2 <- length(sig2)
  # boot_sig1 <- median(sample(sig1,n_sig1,replace=TRUE))
  boot_sig1 <- median(apply(X = sig1,
                            MARGIN=1,
                            FUN = function(x) {
                              sample(x,1,replace=TRUE)
                              }
                            ))
  # randomly sample n_sig1 entries from boot_sig1
  boot_sig1_random <- sample(boot_sig1,n_sig1,replace=TRUE)
  boot_sig2 <- median(apply(sig2,
                            MARGIN=1,
                            function(x) {
                              sample(x,1,replace=TRUE)
                              }
                            ))
  boot_sig2_random <- sample(boot_sig2,n_sig2,replace=TRUE)
  # boot_sig2 <- median(sample(sig2,n_sig2,replace=TRUE))
  
  boot_x <- c(boot_sig1,boot_sig2)
  return(boot_x)
}

estimate_corr_median_signatures <- function(sig1,sig2,K=30) {
  median_boot <- lapply(seq_len(K),function(x) {signature_median(sig1,sig2)})
  median_boot <- do.call(rbind,median_boot)
  boot_corr <- cor(x = median_boot[,1],y = median_boot[,2])
  return(boot_corr)
}

```

## For the human signatures from the three papers

```{r}
# chiappinelli vs canadas
R <- 100

# get OV samples
gdac_selected_cancers_annot_ov <- subset(gdac_selected_cancers_annot,TCGA %in% c("OV"))
sig1_counts <- gdac_selected_cancers.chiappinelli_tcga_mat[,colnames(gdac_selected_cancers.chiappinelli_tcga_mat) %in% rownames(gdac_selected_cancers_annot_ov)]
sig2_counts <- gdac_selected_cancers.canadas_tcga_mat[,colnames(gdac_selected_cancers.canadas_tcga_mat) %in% rownames(gdac_selected_cancers_annot_ov)]
bootstrap_corr_medians_ov <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
hist(bootstrap_corr_medians_ov)
mean_corr_ov.chiappinelli_canadas <- mean(bootstrap_corr_medians_ov)
ci_corr_ov.chiappinelli_canadas <- quantile(bootstrap_corr_medians_ov,c(0.025,0.975))

# get BRCA samples
gdac_selected_cancers_annot_brca <- subset(gdac_selected_cancers_annot,TCGA %in% c("BRCA"))
sig1_counts <- gdac_selected_cancers.chiappinelli_tcga_mat[,colnames(gdac_selected_cancers.chiappinelli_tcga_mat) %in% rownames(gdac_selected_cancers_annot_brca)]
sig2_counts <- gdac_selected_cancers.canadas_tcga_mat[,colnames(gdac_selected_cancers.canadas_tcga_mat) %in% rownames(gdac_selected_cancers_annot_brca)]
bootstrap_corr_medians_brca <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
hist(bootstrap_corr_medians_brca)
mean_corr_brca.chiappinelli_canadas <- mean(bootstrap_corr_medians_brca)
ci_corr_brca.chiappinelli_canadas <- quantile(bootstrap_corr_medians_brca,c(0.025,0.975))

# get COAD samples
gdac_selected_cancers_annot_coad <- subset(gdac_selected_cancers_annot,TCGA %in% c("COAD"))
sig1_counts <- gdac_selected_cancers.chiappinelli_tcga_mat[,colnames(gdac_selected_cancers.chiappinelli_tcga_mat) %in% rownames(gdac_selected_cancers_annot_coad)]
sig2_counts <- gdac_selected_cancers.canadas_tcga_mat[,colnames(gdac_selected_cancers.canadas_tcga_mat) %in% rownames(gdac_selected_cancers_annot_coad)]
bootstrap_corr_medians_coad <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
hist(bootstrap_corr_medians_coad)
mean_corr_coad.chiappinelli_canadas <- mean(bootstrap_corr_medians_coad)
ci_corr_coad.chiappinelli_canadas <- quantile(bootstrap_corr_medians_coad,c(0.025,0.975))

# get LUSC samples
gdac_selected_cancers_annot_lusc <- subset(gdac_selected_cancers_annot,TCGA %in% c("LUSC"))
sig1_counts <- gdac_selected_cancers.chiappinelli_tcga_mat[,colnames(gdac_selected_cancers.chiappinelli_tcga_mat) %in% rownames(gdac_selected_cancers_annot_lusc)]
sig2_counts <- gdac_selected_cancers.canadas_tcga_mat[,colnames(gdac_selected_cancers.canadas_tcga_mat) %in% rownames(gdac_selected_cancers_annot_lusc)]
bootstrap_corr_medians_lusc <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
hist(bootstrap_corr_medians_lusc)
mean_corr_lusc.chiappinelli_canadas <- mean(bootstrap_corr_medians_lusc)
ci_corr_lusc.chiappinelli_canadas <- quantile(bootstrap_corr_medians_lusc,c(0.025,0.975))

```


```{r}
# chiappinelli vs sheng/lafleur
# get OV samples
gdac_selected_cancers_annot_ov <- subset(gdac_selected_cancers_annot,TCGA %in% c("OV"))
sig1_counts <- gdac_selected_cancers.chiappinelli_tcga_mat[,colnames(gdac_selected_cancers.chiappinelli_tcga_mat) %in% rownames(gdac_selected_cancers_annot_ov)]
sig2_counts <- gdac_selected_cancers.sheng_lafleur_tcga_mat[,colnames(gdac_selected_cancers.sheng_lafleur_tcga_mat) %in% rownames(gdac_selected_cancers_annot_ov)]
bootstrap_corr_medians_ov.chiappinelli_sheng_lafleur <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
hist(bootstrap_corr_medians_ov.chiappinelli_sheng_lafleur)
mean_corr_ov.chiappinelli_sheng_lafleur <- mean(bootstrap_corr_medians_ov.chiappinelli_sheng_lafleur)
ci_corr_ov.chiappinelli_sheng_lafleur <- quantile(bootstrap_corr_medians_ov.chiappinelli_sheng_lafleur,c(0.025,0.975))

# get BRCA samples
gdac_selected_cancers_annot_brca <- subset(gdac_selected_cancers_annot,TCGA %in% c("BRCA"))
sig1_counts <- gdac_selected_cancers.chiappinelli_tcga_mat[,colnames(gdac_selected_cancers.chiappinelli_tcga_mat) %in% rownames(gdac_selected_cancers_annot_brca)]
sig2_counts <- gdac_selected_cancers.sheng_lafleur_tcga_mat[,colnames(gdac_selected_cancers.sheng_lafleur_tcga_mat) %in% rownames(gdac_selected_cancers_annot_brca)]
bootstrap_corr_medians_brca.chiappinelli_sheng_lafleur <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
hist(bootstrap_corr_medians_brca.chiappinelli_sheng_lafleur)
mean_corr_brca.chiappinelli_sheng_lafleur <- mean(bootstrap_corr_medians_brca.chiappinelli_sheng_lafleur)
ci_corr_brca.chiappinelli_sheng_lafleur <- quantile(bootstrap_corr_medians_brca.chiappinelli_sheng_lafleur,c(0.025,0.975))

# get COAD samples
gdac_selected_cancers_annot_coad. <- subset(gdac_selected_cancers_annot,TCGA %in% c("COAD"))
sig1_counts <- gdac_selected_cancers.chiappinelli_tcga_mat[,colnames(gdac_selected_cancers.chiappinelli_tcga_mat) %in% rownames(gdac_selected_cancers_annot_coad)]
sig2_counts <- gdac_selected_cancers.sheng_lafleur_tcga_mat[,colnames(gdac_selected_cancers.sheng_lafleur_tcga_mat) %in% rownames(gdac_selected_cancers_annot_coad)]
bootstrap_corr_medians_coad.chiappinelli_sheng_lafleur <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
mean_corr_coad.chiappinelli_sheng_lafleur <- mean(bootstrap_corr_medians_coad.chiappinelli_sheng_lafleur)
ci_corr_coad.chiappinelli_sheng_lafleur <- quantile(bootstrap_corr_medians_coad.chiappinelli_sheng_lafleur,c(0.025,0.975))

# get LUSC samples
gdac_selected_cancers_annot_lusc <- subset(gdac_selected_cancers_annot,TCGA %in% c("LUSC"))
sig1_counts <- gdac_selected_cancers.chiappinelli_tcga_mat[,colnames(gdac_selected_cancers.chiappinelli_tcga_mat) %in% rownames(gdac_selected_cancers_annot_lusc)]
sig2_counts <- gdac_selected_cancers.sheng_lafleur_tcga_mat[,colnames(gdac_selected_cancers.sheng_lafleur_tcga_mat) %in% rownames(gdac_selected_cancers_annot_lusc)]
bootstrap_corr_medians_lusc.chiappinelli_sheng_lafleur <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
mean_corr_lusc.chiappinelli_sheng_lafleur <- mean(bootstrap_corr_medians_lusc.chiappinelli_sheng_lafleur)
ci_corr_lusc.chiappinelli_sheng_lafleur <- quantile(bootstrap_corr_medians_lusc.chiappinelli_sheng_lafleur,c(0.025,0.975))

hist(bootstrap_corr_medians_ov.chiappinelli_sheng_lafleur)
hist(bootstrap_corr_medians_brca.chiappinelli_sheng_lafleur)
hist(bootstrap_corr_medians_coad.chiappinelli_sheng_lafleur)
hist(bootstrap_corr_medians_lusc.chiappinelli_sheng_lafleur)

```

```{r}
# canadas vs sheng/lafleur
# get OV samples
gdac_selected_cancers_annot_ov <- subset(gdac_selected_cancers_annot,TCGA %in% c("OV"))
sig1_counts <- gdac_selected_cancers.canadas_tcga_mat[,colnames(gdac_selected_cancers.canadas_tcga_mat) %in% rownames(gdac_selected_cancers_annot_ov)]
sig2_counts <- gdac_selected_cancers.sheng_lafleur_tcga_mat[,colnames(gdac_selected_cancers.sheng_lafleur_tcga_mat) %in% rownames(gdac_selected_cancers_annot_ov)]
bootstrap_corr_medians_ov.canadas_sheng_lafleur <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
mean_corr_ov.canadas_sheng_lafleur <- mean(bootstrap_corr_medians_ov.canadas_sheng_lafleur)
ci_corr_ov.canadas_sheng_lafleur <- quantile(bootstrap_corr_medians_ov.canadas_sheng_lafleur,c(0.025,0.975))

# get BRCA samples
gdac_selected_cancers_annot_brca <- subset(gdac_selected_cancers_annot,TCGA %in% c("BRCA"))
sig1_counts <- gdac_selected_cancers.canadas_tcga_mat[,colnames(gdac_selected_cancers.canadas_tcga_mat) %in% rownames(gdac_selected_cancers_annot_brca)]
sig2_counts <- gdac_selected_cancers.sheng_lafleur_tcga_mat[,colnames(gdac_selected_cancers.sheng_lafleur_tcga_mat) %in% rownames(gdac_selected_cancers_annot_brca)]
bootstrap_corr_medians_brca.canadas_sheng_lafleur <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
mean_corr_brca.canadas_sheng_lafleur <- mean(bootstrap_corr_medians_brca.canadas_sheng_lafleur)
ci_corr_brca.canadas_sheng_lafleur <- quantile(bootstrap_corr_medians_brca.canadas_sheng_lafleur,c(0.025,0.975))

# get COAD samples
gdac_selected_cancers_annot_coad. <- subset(gdac_selected_cancers_annot,TCGA %in% c("COAD"))
sig1_counts <- gdac_selected_cancers.canadas_tcga_mat[,colnames(gdac_selected_cancers.canadas_tcga_mat) %in% rownames(gdac_selected_cancers_annot_coad)]
sig2_counts <- gdac_selected_cancers.sheng_lafleur_tcga_mat[,colnames(gdac_selected_cancers.sheng_lafleur_tcga_mat) %in% rownames(gdac_selected_cancers_annot_coad)]
bootstrap_corr_medians_coad.canadas_sheng_lafleur <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
mean_corr_coad.canadas_sheng_lafleur <- mean(bootstrap_corr_medians_coad.canadas_sheng_lafleur)
ci_corr_coad.canadas_sheng_lafleur <- quantile(bootstrap_corr_medians_coad.canadas_sheng_lafleur,c(0.025,0.975))

# get LUSC samples
gdac_selected_cancers_annot_lusc <- subset(gdac_selected_cancers_annot,TCGA %in% c("LUSC"))
sig1_counts <- gdac_selected_cancers.canadas_tcga_mat[,colnames(gdac_selected_cancers.canadas_tcga_mat) %in% rownames(gdac_selected_cancers_annot_lusc)]
sig2_counts <- gdac_selected_cancers.sheng_lafleur_tcga_mat[,colnames(gdac_selected_cancers.sheng_lafleur_tcga_mat) %in% rownames(gdac_selected_cancers_annot_lusc)]
bootstrap_corr_medians_lusc.canadas_sheng_lafleur <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
mean_corr_lusc.canadas_sheng_lafleur <- mean(bootstrap_corr_medians_lusc.canadas_sheng_lafleur)
ci_corr_lusc.canadas_sheng_lafleur <- quantile(bootstrap_corr_medians_lusc.canadas_sheng_lafleur,c(0.025,0.975))


# plots
hist(bootstrap_corr_medians_ov.canadas_sheng_lafleur)
hist(bootstrap_corr_medians_brca.canadas_sheng_lafleur)
hist(bootstrap_corr_medians_coad.canadas_sheng_lafleur)
hist(bootstrap_corr_medians_lusc.canadas_sheng_lafleur)

# use the CI's as interval estimates for the actual signature correlations. 
```


The Chiappinelli and Canadas signatures are not particularly well-correlated with each other (mean correlation is between 0.1 and -0.1 for the tumor types studied). However, what if we limited the chiappinelli signature to genes that go up with AZA treatment (AIM genes include the up- and down-regulated genes, but the genes that increase in expression with AZA-treatment may be informative).

## For the four clusters from TCGA-wide analysis

Just show the highest-expressed and lowest expressed for now
```{r}
# highest vs lowest
# get OV samples
gdac_selected_cancers_annot_ov <- subset(gdac_selected_cancers_annot,TCGA %in% c("OV"))
sig1_counts <- highest_unified_counts[,colnames(highest_unified_counts) %in% rownames(gdac_selected_cancers_annot_ov)]
sig2_counts <- lowest_unified_counts[,colnames(lowest_unified_counts) %in% rownames(gdac_selected_cancers_annot_ov)]
bootstrap_corr_medians_ov.highest_vs_lowest <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
mean_corr_ov.highest_vs_lowest <- mean(bootstrap_corr_medians_ov.highest_vs_lowest)
ci_corr_ov.highest_vs_lowest <- quantile(bootstrap_corr_medians_ov.highest_vs_lowest,c(0.025,0.975))

# get BRCA samples
gdac_selected_cancers_annot_brca <- subset(gdac_selected_cancers_annot,TCGA %in% c("BRCA"))
sig1_counts <- gdac_selected_cancers.canadas_tcga_mat[,colnames(gdac_selected_cancers.canadas_tcga_mat) %in% rownames(gdac_selected_cancers_annot_brca)]
sig2_counts <- gdac_selected_cancers.sheng_lafleur_tcga_mat[,colnames(gdac_selected_cancers.sheng_lafleur_tcga_mat) %in% rownames(gdac_selected_cancers_annot_brca)]
bootstrap_corr_medians_brca.highest_vs_lowest <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
mean_corr_brca.highest_vs_lowest <- mean(bootstrap_corr_medians_brca.highest_vs_lowest)
ci_corr_brca.highest_vs_lowest <- quantile(bootstrap_corr_medians_brca.highest_vs_lowest,c(0.025,0.975))

# get COAD samples
gdac_selected_cancers_annot_coad. <- subset(gdac_selected_cancers_annot,TCGA %in% c("COAD"))
sig1_counts <- gdac_selected_cancers.canadas_tcga_mat[,colnames(gdac_selected_cancers.canadas_tcga_mat) %in% rownames(gdac_selected_cancers_annot_coad)]
sig2_counts <- gdac_selected_cancers.sheng_lafleur_tcga_mat[,colnames(gdac_selected_cancers.sheng_lafleur_tcga_mat) %in% rownames(gdac_selected_cancers_annot_coad)]
bootstrap_corr_medians_coad.highest_vs_lowest <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
mean_corr_coad.highest_vs_lowest <- mean(bootstrap_corr_medians_coad.highest_vs_lowest)
ci_corr_coad.highest_vs_lowest <- quantile(bootstrap_corr_medians_coad.highest_vs_lowest,c(0.025,0.975))

# get LUSC samples
gdac_selected_cancers_annot_lusc <- subset(gdac_selected_cancers_annot,TCGA %in% c("LUSC"))
sig1_counts <- gdac_selected_cancers.canadas_tcga_mat[,colnames(gdac_selected_cancers.canadas_tcga_mat) %in% rownames(gdac_selected_cancers_annot_lusc)]
sig2_counts <- gdac_selected_cancers.sheng_lafleur_tcga_mat[,colnames(gdac_selected_cancers.sheng_lafleur_tcga_mat) %in% rownames(gdac_selected_cancers_annot_lusc)]
bootstrap_corr_medians_lusc.highest_vs_lowest <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
mean_corr_lusc.highest_vs_lowest <- mean(bootstrap_corr_medians_lusc.highest_vs_lowest)
ci_corr_lusc.highest_vs_lowest <- quantile(bootstrap_corr_medians_lusc.highest_vs_lowest,c(0.025,0.975))


# plots
hist(bootstrap_corr_medians_ov.highest_vs_lowest)
hist(bootstrap_corr_medians_brca.highest_vs_lowest)
hist(bootstrap_corr_medians_coad.highest_vs_lowest)
hist(bootstrap_corr_medians_lusc.highest_vs_lowest)
```

```{r}
hist(bootstrap_corr_medians_ov.highest_vs_lowest)
hist(bootstrap_corr_medians_brca.highest_vs_lowest)
hist(bootstrap_corr_medians_coad.highest_vs_lowest)
hist(bootstrap_corr_medians_lusc.highest_vs_lowest)

```

## Conclusions

1. The three signatures from the paper all seem to include different components of tumor innate immunity and the antiviral response. However, the signatures are very inconsistent with each other
  a. Why would ERV activation lead to such different signatures?
  b. What happens if we consider only up-regulated components of each signature?
2. When we pool the signatures, we find four groups that stand out which could constitute a "unified" signature: a highly-expressed subset consisting of housekeeping genes and components of antigen presentation, a more sparsely-expressed subset in tumors, and two more "medium-expressed" subset of genes. One of the two "medium-subsets" is enriched for components of the interferon response pathway.
3. 
