---
title: "Unified analysis of signatures"
output: html_notebook
---

We derived three signatures from each of the human datasets--Chiappinelli, Canadas, and Sheng/LaFleur. We want to know how similar these signatures, and specifically the IFN-ERV signature, are. 

# Import signatures

These are up-regulated signatures
```{r get_chiappinelli_sig}
# breast table
print(getwd())
chiappinelli_up_breast <- read.table(file = "./li_chiappinelli_oncotarget_2014/breast_up_genes_in_signature_table.txt",
                                  header = TRUE,
                                  sep = "\t",
                                  row.names = 1)
chiappinelli_down_breast <- read.table(file = "./li_chiappinelli_oncotarget_2014/breast_down_genes_in_signature_table.txt",
                                  header = TRUE,sep = "\t",row.names = 1)

# colon table
chiappinelli_up_colon <- read.table(file = "./li_chiappinelli_oncotarget_2014/colon_up_genes_in_signature_table.txt",
                                 header = TRUE,sep = "\t",row.names = 1)
chiappinelli_down_colon <- read.table(file = "./li_chiappinelli_oncotarget_2014/colon_down_genes_in_signature_table.txt",
                                  header = TRUE,sep = "\t",row.names = 1)

# ovarian table
chiappinelli_up_ovarian <- read.table(file = "li_chiappinelli_oncotarget_2014/ovarian_up_genes_in_signature_table.txt",
                                      header = TRUE,sep = "\t",row.names = 1)
chiappinelli_down_ovarian <- read.table(file = "li_chiappinelli_oncotarget_2014/ovarian_down_genes_in_signature_table.txt",
                                  header = TRUE,sep = "\t",row.names = 1)


# get the genes that go up and their average expressions
breast_ave_expr <- data.frame(avexpr=chiappinelli_up_breast$AveExpr)
breast_ave_expr.aggr <- aggregate.data.frame(x = breast_ave_expr,
                                             by = list(chiappinelli_up_breast$GENE_SYMBOL),
                                             FUN = mean)
colon_ave_expr <- data.frame(avexpr=chiappinelli_up_colon$AveExpr)
colon_ave_expr.aggr <- aggregate.data.frame(x = colon_ave_expr,
                                            by = list(chiappinelli_up_colon$GENE_SYMBOL),
                                            FUN = mean)
ovarian_ave_expr <- data.frame(avexpr=chiappinelli_up_ovarian$AveExpr)
ovarian_ave_expr.aggr <- aggregate.data.frame(x = ovarian_ave_expr,
                                              by = list(chiappinelli_up_ovarian$GENE_SYMBOL),
                                              FUN = mean)

# up signature
ifn_signature_up_genes_list <- read.table(file = "li_chiappinelli_oncotarget_2014/test_ifn_signature_probeInd_genes.txt",sep = "\t")
# down signature
ifn_signature_down_genes_list <- read.table(file = "li_chiappinelli_oncotarget_2014/test_down_ifn_signature_probeInd_genes.txt",sep = "\t")
# get unique signatures
ifn_signature_genes_list <- do.call(rbind,
                                    list(ifn_signature_up_genes_list,
                                         ifn_signature_down_genes_list))
chiappinelli_ifn_signature_unique_genes <- unique(ifn_signature_genes_list$gene)

# experimentally-annotated signatures
chiappinelli_ifn_signature_experimental <- read.table(file = "li_chiappinelli_oncotarget_2014/manually_written_signature.true_unique.txt",
                                                      sep = "\t")

# annotate these signatures
chiappinelli_ifn_signature_unique_genes_annot<- data.frame(is_up = chiappinelli_ifn_signature_unique_genes %in% ifn_signature_up_genes_list$gene,
                                                           is_down = chiappinelli_ifn_signature_unique_genes %in% ifn_signature_down_genes_list$gene)
chiappinelli_ifn_signature_unique_genes_annot$is_up <- factor(x = chiappinelli_ifn_signature_unique_genes_annot$is_up,
                                                              levels = c(FALSE,TRUE),
                                                              labels = c(0,1))
chiappinelli_ifn_signature_unique_genes_annot$is_down <- factor(x = chiappinelli_ifn_signature_unique_genes_annot$is_down,
                                                              levels = c(FALSE,TRUE),
                                                              labels = c(0,1))

rownames(chiappinelli_ifn_signature_unique_genes_annot) <- chiappinelli_ifn_signature_unique_genes

```

```{r get_canadas_sig}
# get all up-regulated genes that we designated
# write.table(x = tT_canadas_sig_up_in_sem,file = "./tT_canadas_sig_up_in_sem.txt",row.names = TRUE,col.names = TRUE,sep = "\t",quote = FALSE)
# write.table(x = tT_canadas_sig_up_h69m,file = "./tT_canadas_sig_up_h69m.txt",row.names = TRUE,col.names = TRUE,sep = "\t",quote = FALSE)
# write.table(x = tT_canadas_sig_up_h69m_in_sem,file = "./tT_canadas_sig_up_h69m_in_sem.txt",row.names = TRUE,col.names = TRUE,sep = "\t",quote = FALSE)
canadas_sig_nsclc <- read.table(file = "canadas_2018/tT_canadas_sig_up_h69m.txt",
                            header = TRUE,
                            row.names = 1,
                            sep = "\t")

# get the up-regulated genes that Canadas designated
canadas_sig_sem_nsclc <- read.table(file = "canadas_2018/tT_canadas_sig_up_in_sem.txt",
                            header = TRUE,
                            row.names = 1,
                            sep = "\t")

# get their SPARCs genes
canadas_sparcs_nsclc <- read.table(file = "canadas_2018/tT_canadas_sparcs.txt",
                            header = TRUE,
                            row.names = 1,
                            sep = "\t")

# get all genes with a 3' UTR
canadas_3putr_nsclc <- read.table(file = "canadas_2018/tT_canadas_erv_sig_all.txt",
                            header = TRUE,
                            row.names = 1,
                            sep = "\t")

# complete, SPARCS or non-SPARCS signature from canadas
canadas_ifn_signature_unique_genes <- unique(c(as.character(canadas_sig_nsclc$GENE_SYMBOL),
                                               as.character(canadas_sig_sem_nsclc$GENE_SYMBOL)))
canadas_ifn_signature_unique_genes <- canadas_ifn_signature_unique_genes[!is.na(canadas_ifn_signature_unique_genes)]
```

```{r get_shenglafleur_sig}
# get the human upregulated signatures
sheng_lafleur_signature_symbols <- read.table(file = "sheng_lafleur_2018/ifn_viral_signature_shenglafleur_hs.txt",
                                              header = TRUE,
                                              row.names = 1,
                                              sep = "\t")
sheng_lafleur_signature <- as.character(sheng_lafleur_signature_symbols$symbol)
```

# Measure intersections

```{r vennDiag}
# install.package("VennDiagram")
# draw a Venn Diagram
library(VennDiagram)
chiap_canad_venn <- venn.diagram(x = list(chiappinelli = as.character(chiappinelli_ifn_signature_unique_genes),
                                          canadas = canadas_ifn_signature_unique_genes,
                                          sheng_lafleur = sheng_lafleur_signature),
                                 filename = "./canadas_chiappinelli_intersect.png",
                                 main = "Intersection between all DEX genes",
                                 imagetype = "png",
                                 main.fontfamily = "Arial",sub.fontfamily = "Arial")
```

```{r}
unified_sig <- unique(c(as.character(chiappinelli_ifn_signature_unique_genes),
                        canadas_ifn_signature_unique_genes,
                        sheng_lafleur_signature))
# assemble the table of intersections
unified_sig_intersection <- data.frame(chiappinelli = (unified_sig %in% as.character(chiappinelli_ifn_signature_unique_genes)))
unified_sig_intersection$canadas <- (unified_sig %in% canadas_ifn_signature_unique_genes)
unified_sig_intersection$shenglafleur <- (unified_sig %in% sheng_lafleur_signature)
rownames(unified_sig_intersection) <- unified_sig
# pheatmap(unified_sig_intersection)

unified_sig_intersection_all <- subset(unified_sig_intersection,
                                   chiappinelli & canadas & shenglafleur)
unified_sig_intersection_chia_sheng <- subset(unified_sig_intersection,
                                              chiappinelli & shenglafleur & !canadas)
unified_sig_intersection_canad_sheng <- subset(unified_sig_intersection,
                                              canadas & shenglafleur & !chiappinelli)
unified_sig_intersection_chia_canad <- subset(unified_sig_intersection,
                                              chiappinelli & canadas & !shenglafleur)

# save the table
write.table(x = unified_sig_intersection_all,
            file = "./unified_sig_intersection_all.txt",
            sep = "\t",
            col.names = TRUE,
            row.names = TRUE)
print(unified_sig_intersection_all)
print(unified_sig_intersection_chia_sheng)
print(unified_sig_intersection_canad_sheng)
print(unified_sig_intersection_chia_canad)

```

The studies have very little overlap with each other. The four genes conserved across all the conditions are involved in endogenous peptide presentation (TAP2 loads peptides onto MHCI complex). The individual pairwise matches show different components of the interferon response and innate immunity up-regulated.


# TCGA re-analysis

## Get GDAC data
Using the different intersections of signatures, we want to re-cluster TCGA data and find if we can group tumors just as well. Data was downloaded from GDAC firehose
```{r gdac}

brca_gdac <- read.table(file = "../datasets/gdac_firehose/gdac.broadinstitute.org_BRCA.Merge_rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.Level_3.2016012800.0.0/BRCA.rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.data.txt",header = TRUE,sep = "\t",row.names = 1)
# save(brca_gdac, file = "./data/RNAseq/brcaRNAseq.RData")

# Colon
coad_gdac <- read.table(file = "../datasets/gdac_firehose/gdac.broadinstitute.org_COAD.Merge_rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.Level_3.2016012800.0.0/COAD.rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.data.txt",header = TRUE,sep = "\t",row.names = 1)
# save(coad_gdac, file = "./data/RNAseq/coadRNAseq.RData")

# Lung
lusc_gdac <- read.table(file = "../datasets/gdac_firehose/gdac.broadinstitute.org_LUSC.Merge_rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.Level_3.2016012800.0.0/LUSC.rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.data.txt",header = TRUE,sep = "\t",row.names = 1)
# save(lusc_gdac, file = "./data/RNAseq/luscRNAseq.RData")

# Ovarian
ov_gdac <- read.table(file = "../datasets/gdac_firehose/gdac.broadinstitute.org_OV.Merge_rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.Level_3.2016012800.0.0/OV.rnaseqv2__illuminahiseq_rnaseqv2__unc_edu__Level_3__RSEM_genes_normalized__data.data.txt",header = TRUE,sep = "\t",row.names = 1)
# save(ov_gdac, file = "./data/RNAseq/ovRNAseq.RData")

# save cancer files
gdac_cancer_files <- list(brca = brca_gdac,
                          coad = coad_gdac,
                          lusc = lusc_gdac,
                          ov = ov_gdac)

```

Compress into a single data frame
```{r}
gdac_selected_cancers <- as.data.frame(do.call(what = cbind,args = gdac_cancer_files))
gdac_selected_cancers_annot <- data.frame(TCGA = c(rep("BRCA",length(brca_gdac)),
                                                   rep("COAD",length(coad_gdac)),
                                                   rep("LUSC",length(lusc_gdac)),
                                                   rep("OV",length(ov_gdac))
                                                   ))
rownames(gdac_selected_cancers_annot) <- colnames(gdac_selected_cancers)
```


## Identify expression of each IFN-viral gene signature

### Chiappinelli 
```{r chiappinelli_tcga}

chiappinelli_ifn_signature_unique_genes <- as.character(chiappinelli_ifn_signature_unique_genes)

chiappinelli_ifn_signature_tcga <- sapply(chiappinelli_ifn_signature_unique_genes,
                                          function(g) {
                                            a <- grepl(pattern = as.character(paste0("^",g,"\\|")),
                                                       x = rownames(gdac_selected_cancers),
                                                       ignore.case = FALSE)
                                            rownames(gdac_selected_cancers)[a]
                                          })

chiappinelli_ifn_signature_experimental <- sapply(chiappinelli_ifn_signature_experimental$V1,
                                          function(g) {
                                            a <- grepl(pattern = as.character(paste0("^",g,"\\|")),
                                                       x = rownames(gdac_selected_cancers),
                                                       ignore.case = FALSE)
                                            rownames(gdac_selected_cancers)[a]
                                          })
  
  
# gdac_selected_cancers.chiappinelli_tcga_mat[rownames(gdac_selected_cancers) %in% chiappinelli_ifn_signature_experimental$V1,]

gdac_selected_cancers.chiappinelli_tcga <- gdac_selected_cancers[rownames(gdac_selected_cancers) %in% 
                                                                   unlist(chiappinelli_ifn_signature_tcga),]
gdac_selected_cancers.chiappinelli_tcga_experimental <- gdac_selected_cancers[rownames(gdac_selected_cancers) %in% 
                                                                   unlist(chiappinelli_ifn_signature_experimental),]

rownames(gdac_selected_cancers.chiappinelli_tcga) <- sapply(rownames(gdac_selected_cancers.chiappinelli_tcga),
                                                            function(x) {
                                                              a <- unlist(strsplit(x = x,split = "|",fixed = TRUE))
                                                              a[1]
                                                            })
rownames(gdac_selected_cancers.chiappinelli_tcga_experimental) <- sapply(rownames(gdac_selected_cancers.chiappinelli_tcga_experimental),
                                                            function(x) {
                                                              a <- unlist(strsplit(x = x,split = "|",fixed = TRUE))
                                                              a[1]
                                                            })

#rownames(gdac_selected_cancers.chiappinelli_tcga) <- names(unlist(chiappinelli_ifn_signature_tcga))

gdac_selected_cancers.chiappinelli_tcga_mat <- apply(X = gdac_selected_cancers.chiappinelli_tcga,
                                                 MARGIN = 2,
                                                 FUN = function(x) {
                                                   as.numeric(x)
                                                 })
gdac_selected_cancers.chiappinelli_experimental_tcga_mat <- apply(X = gdac_selected_cancers.chiappinelli_tcga_experimental,
                                                 MARGIN = 2,
                                                 FUN = function(x) {
                                                   as.numeric(x)
                                                 })

rownames(gdac_selected_cancers.chiappinelli_tcga_mat) <- rownames(gdac_selected_cancers.chiappinelli_tcga)
rownames(gdac_selected_cancers.chiappinelli_experimental_tcga_mat) <- rownames(gdac_selected_cancers.chiappinelli_tcga_experimental)

  #grep(pattern = paste0(chiappinelli_ifn_signature_unique_genes,"|"),x = rownames(gdac_selected_cancers),ignore.case = TRUE)

# gdac_selected_cancers.chiappinelli <- gdac_selected_cancers[]


```

```{r chiappinelli_tcga_plot}
pdf(file = "chiappinelli_DEX_all.pdf")
chiappinelli_tcga_clust <- pheatmap(mat = log10(gdac_selected_cancers.chiappinelli_tcga_mat+1),
         show_colnames = FALSE,
         show_rownames = TRUE,
         fontsize_row = 2.5,
         annotation_col = gdac_selected_cancers_annot,
         annotation_row = chiappinelli_ifn_signature_unique_genes_annot,
         legend_labels = c("log10(RSEM-scaled counts+1)"),
         cutree_rows = 4,
         main = c(paste("DEX genes from Li et al 2014 and Chiappinelli et al 2015\nin",ncol(gdac_selected_cancers.chiappinelli_tcga_mat),"TCGA samples")))
dev.off()
# plot(chiappinelli_tcga_clust)
# experimental verification
chiappinelli_experimental_tcga_clust <- pheatmap(mat = log10(gdac_selected_cancers.chiappinelli_experimental_tcga_mat+1),
         show_colnames = FALSE,
         show_rownames = TRUE,
         fontsize_row = 5.0,
         annotation_col = gdac_selected_cancers_annot,
         legend_labels = c("log10(RSEM-scaled counts+1)"),
         cutree_rows = 4,
         main = c(paste("Experimentally-validated signature from Chiappinelli et al 2015\nin",ncol(gdac_selected_cancers.chiappinelli_tcga_mat),"TCGA samples")))
venn.diagram(x = list(AIM=rownames(gdac_selected_cancers.chiappinelli_tcga_mat),
                      Experimental=rownames(gdac_selected_cancers.chiappinelli_experimental_tcga_mat)),
             filename = "./chiappinell_intersection_AIM_Experimental.png",imagetype = "png",sub.fontfamily = "Arial",main.fontfamily = "Arial")
```

the specific ERV signature?

### Canadas

```{r chiappinelli_tcga}

canadas_ifn_signature_unique_genes <- as.character(canadas_ifn_signature_unique_genes)

canadas_ifn_signature_tcga <- sapply(canadas_ifn_signature_unique_genes,
                                          function(g) {
                                            a <- grepl(pattern = as.character(paste0("^",g,"\\|")),
                                                       x = rownames(gdac_selected_cancers),
                                                       ignore.case = FALSE)
                                            rownames(gdac_selected_cancers)[a]
                                          })
gdac_selected_cancers.canadas_tcga <- gdac_selected_cancers[rownames(gdac_selected_cancers) %in% 
                                                                   unlist(canadas_ifn_signature_tcga),]
canadas_split_names <- sapply(rownames(gdac_selected_cancers.canadas_tcga),
                                                            function(x) {
                                                              a <- unlist(strsplit(x = x,
                                                                                   split = "|",fixed = TRUE))
                                                              a[1]
                                                            })

gdac_selected_cancers.canadas_tcga_mat <- apply(X = gdac_selected_cancers.canadas_tcga,
                                                 MARGIN = 2,
                                                 FUN = function(x) {
                                                   as.numeric(x)
                                                 })

rownames(gdac_selected_cancers.canadas_tcga_mat) <- as.character(canadas_split_names)

rownames(gdac_selected_cancers.canadas_tcga_mat) <- as.character(canadas_split_names) # rownames(gdac_selected_cancers.canadas_tcga)
  #grep(pattern = paste0(chiappinelli_ifn_signature_unique_genes,"|"),x = rownames(gdac_selected_cancers),ignore.case = TRUE)

# gdac_selected_cancers.chiappinelli <- gdac_selected_cancers[]

```

```{r chiappinelli_tcga_plot}
pdf(file = "canadas_DEX_all.pdf")
canadas_tcga_clust <- pheatmap(mat = log10(gdac_selected_cancers.canadas_tcga_mat+1),
         show_colnames = FALSE,
         show_rownames = TRUE,
         fontsize_row = 2.5,
         annotation_col = gdac_selected_cancers_annot,
         legend_labels = c("log10(RSEM-scaled counts+1)"),
         cutree_rows = 4,
         main = c(paste("DEX genes from Canadas et al 2014 and Canadas et al 2018\nin",ncol(gdac_selected_cancers.chiappinelli_tcga_mat),"TCGA samples")))
dev.off()
# plot(chiappinelli_tcga_clust)
```

Cluster 2 seems to be a "lung inflammatory" sub-signature

```{r}
gdac_selected_cancers.canadas_tcga_mat_3putr <- gdac_selected_cancers.canadas_tcga_mat[rownames(gdac_selected_cancers.canadas_tcga_mat) %in% 
                                                                                         canadas_3putr_nsclc$GENE_SYMBOL,]
pdf(file = "canadas_3pUTR_DEX_all.pdf")
canadas_3putrerv_tcga_clust <- pheatmap(mat = log10(gdac_selected_cancers.canadas_tcga_mat_3putr+1),
         show_colnames = FALSE,
         show_rownames = TRUE,
         fontsize_row = 5.0,
         annotation_col = gdac_selected_cancers_annot,
         legend_labels = c("log10(RSEM-scaled counts+1)"),
         main = c(paste("3' UTR ERV-containing DEX genes from Canadas et al 2014 and\nCanadas et al 2018 in",ncol(gdac_selected_cancers.chiappinelli_tcga_mat),"TCGA samples")))
# plot(chiappinelli_tcga_clust)
dev.off()
```

Most of the 3' UTR ERV-containing DEX genes are uniformly increased in expression across the tumors


### Sheng and LaFleur

All Sheng and LaFleur DEX?

Sheng and LaFleur signature
```{r chiappinelli_tcga}

sheng_lafleur_signature_unique_genes <- as.character(sheng_lafleur_signature)

sheng_lafleur_signature_tcga <- sapply(sheng_lafleur_signature_unique_genes,
                                          function(g) {
                                            a <- grepl(pattern = as.character(paste0("^",g,"\\|")),
                                                       x = rownames(gdac_selected_cancers),
                                                       ignore.case = FALSE)
                                            rownames(gdac_selected_cancers)[a]
                                          })
gdac_selected_cancers.sheng_lafleur_tcga <- gdac_selected_cancers[rownames(gdac_selected_cancers) %in% 
                                                                   unlist(sheng_lafleur_signature_tcga),]

gdac_selected_cancers.sheng_lafleur_tcga_mat <- apply(X = gdac_selected_cancers.sheng_lafleur_tcga,
                                                 MARGIN = 2,
                                                 FUN = function(x) {
                                                   as.numeric(x)
                                                 })

rownames(gdac_selected_cancers.sheng_lafleur_tcga_mat) <- sapply(rownames(gdac_selected_cancers.sheng_lafleur_tcga),
                                                            function(x) {
                                                              a <- unlist(strsplit(x = x,split = "|",fixed = TRUE))
                                                              a[1]
                                                            })

# rownames(gdac_selected_cancers.sheng_lafleur_tcga_mat) <- rownames(gdac_selected_cancers.sheng_lafleur_tcga)
  #grep(pattern = paste0(chiappinelli_ifn_signature_unique_genes,"|"),x = rownames(gdac_selected_cancers),ignore.case = TRUE)

# gdac_selected_cancers.chiappinelli <- gdac_selected_cancers[]

```

```{r sheng_lafleur_plot}
pdf(file = "sheng_lafleur_DEX_all.pdf")
sheng_lafleur_tcga_clust <- pheatmap(mat = log10(gdac_selected_cancers.sheng_lafleur_tcga_mat+1),
         show_colnames = FALSE,
         show_rownames = TRUE,
         fontsize_row = 2.5,
         annotation_col = gdac_selected_cancers_annot,
         legend_labels = c("log10(RSEM-scaled counts+1)"),
         cutree_rows = 5,
         main = c(paste("DEX genes from Sheng and LaFleur et al 2018\nin",ncol(gdac_selected_cancers.chiappinelli_tcga_mat),"TCGA samples")))
dev.off()

png(file = "sheng_lafleur_DEX_all.png")
sheng_lafleur_tcga_clust <- pheatmap(mat = log10(gdac_selected_cancers.sheng_lafleur_tcga_mat+1),
         show_colnames = FALSE,
         show_rownames = TRUE,
         fontsize_row = 2.5,
         annotation_col = gdac_selected_cancers_annot,
         legend_labels = c("log10(RSEM-scaled counts+1)"),
         cutree_rows = 5,
         main = c(paste("DEX genes from Sheng and LaFleur et al 2018\nin",ncol(gdac_selected_cancers.chiappinelli_tcga_mat),"TCGA samples")))
dev.off()

# plot(chiappinelli_tcga_clust)
```


The "up-regulation" and "down-regulation" subsets from the cell lines don't exactly predict the "high" and "low" subsets in the patients. How consistent is the classifier?


### Consolidated signature

```{r unified_tcga}


unified_signature_tcga <- sapply(rownames(unified_sig_intersection),
                                          function(g) {
                                            a <- grepl(pattern = as.character(paste0("^",g,"\\|")),
                                                       x = rownames(gdac_selected_cancers),
                                                       ignore.case = FALSE)
                                            rownames(gdac_selected_cancers)[a]
                                          })
gdac_selected_cancers.unified_signature_tcga <- gdac_selected_cancers[rownames(gdac_selected_cancers) %in% 
                                                                   unlist(unified_signature_tcga),]

gdac_selected_cancers.unified_signature_tcga_mat <- apply(X = gdac_selected_cancers.unified_signature_tcga,
                                                 MARGIN = 2,
                                                 FUN = function(x) {
                                                   as.numeric(x)
                                                 })

rownames(gdac_selected_cancers.unified_signature_tcga_mat) <- unlist(sapply(rownames(gdac_selected_cancers.unified_signature_tcga),
                                                            function(x) {
                                                              a <- unlist(strsplit(x = x,split = "|",fixed = TRUE))
                                                              a[1]
                                                            }))

# rownames(gdac_selected_cancers.sheng_lafleur_tcga_mat) <- rownames(gdac_selected_cancers.sheng_lafleur_tcga)
  #grep(pattern = paste0(chiappinelli_ifn_signature_unique_genes,"|"),x = rownames(gdac_selected_cancers),ignore.case = TRUE)

# gdac_selected_cancers.chiappinelli <- gdac_selected_cancers[]

```

```{r unified_row_annot}

unified_sig_intersection$chiappinelli <- factor(x = unified_sig_intersection$chiappinelli,levels = c(FALSE,TRUE),labels = c("no","yes"))
unified_sig_intersection$canadas <- factor(x = unified_sig_intersection$canadas,levels = c(FALSE,TRUE),labels = c("no","yes"))
unified_sig_intersection$shenglafleur <- factor(x = unified_sig_intersection$shenglafleur,levels = c(FALSE,TRUE),labels = c("no","yes"))
```

```{r unified_row_plot}
unified_sig_intersection <- unified_sig_intersection[order(unified_sig_intersection$chiappinelli,
                                                           unified_sig_intersection$canadas,
                                                           unified_sig_intersection$shenglafleur),]

set.seed(0)
nhclust <- 4
pdf("unified_DEX_genes.pdf")
unified_tcga_clust <- pheatmap(mat = log10(gdac_selected_cancers.unified_signature_tcga_mat+1),
         show_colnames = FALSE,
         show_rownames = FALSE,
         fontsize_row = 2.5,
         annotation_col = gdac_selected_cancers_annot,
         annotation_row = unified_sig_intersection,
         cutree_rows = 4,
         legend_labels = c("log10(RSEM-scaled counts+1)"),
         main = c(paste("All DEX genes found in at least one dataset\nin",ncol(gdac_selected_cancers.chiappinelli_tcga_mat),"TCGA samples")))
dev.off()

# png("unified_DEX_genes.png")
# unified_tcga_clust <- pheatmap(mat = log10(gdac_selected_cancers.unified_signature_tcga_mat+1),
#          show_colnames = FALSE,
#          show_rownames = FALSE,
#          fontsize_row = 2.5,
#          annotation_col = gdac_selected_cancers_annot,
#          annotation_row = unified_sig_intersection,
#          cutree_rows = 4,
#          legend_labels = c("log10(RSEM-scaled counts+1)"),
#          main = c(paste("All DEX genes found in at least one dataset\nin",ncol(gdac_selected_cancers.chiappinelli_tcga_mat),"TCGA samples")))
# dev.off()
unified_tcga_clust_hclusters <- cutree(tree = unified_tcga_clust$tree_row,k = nhclust)

# plot(chiappinelli_tcga_clust)
```

We get four different "TCGA-IFN-viral" clusters. Cluster 4 (smallest) is uniformly increased across all of the samples, while cluster 3 (mostly blue) is uniformly decreased. 

Top-most cluster
```{r top_cluster}
highest_cluster_unified_signature <- unified_tcga_clust_hclusters[unified_tcga_clust_hclusters %in% 4]
lowest_cluster_unified_signature <- unified_tcga_clust_hclusters[unified_tcga_clust_hclusters %in% 3]

gdac_selected_cancers.unified_signature_tcga_mat.top_vs_bottom <- gdac_selected_cancers.unified_signature_tcga_mat[rownames(gdac_selected_cancers.unified_signature_tcga_mat) %in% 
                                                                                         names(c(highest_cluster_unified_signature,lowest_cluster_unified_signature)),] 
rowannot_top_bottom_clusters <- data.frame(cluster = as.factor(c(highest_cluster_unified_signature,
                                                                 lowest_cluster_unified_signature)))
rownames(rowannot_top_bottom_clusters) <- names(c(highest_cluster_unified_signature,
                                                     lowest_cluster_unified_signature))

pdf("unified_DEX_genes.top_vs_bottom_clusters.pdf")
unified_signature_tcga_mat.top_vs_bottom_clust <- pheatmap(mat = log10(gdac_selected_cancers.unified_signature_tcga_mat.top_vs_bottom+1),
         show_colnames = FALSE,
         show_rownames = FALSE,
         cluster_cols = FALSE,
         fontsize_row = 5.0,
         annotation_col = gdac_selected_cancers_annot,
         annotation_row = rowannot_top_bottom_clusters,
         cutree_rows = 2,
         legend_labels = c("log10(RSEM-scaled counts+1)"),
         main = c(paste("Highest- and lowest-expressed gene sets of the",nrow(unified_sig_intersection),
                        "\nunique DEX genes in ",ncol(gdac_selected_cancers.chiappinelli_tcga_mat),"TCGA samples")))
# plot(chiappinelli_tcga_clust)
dev.off()


var1_cluster_unified_signature <- unified_tcga_clust_hclusters[unified_tcga_clust_hclusters %in% 1]
var2_cluster_unified_signature <- unified_tcga_clust_hclusters[unified_tcga_clust_hclusters %in% 2]

gdac_selected_cancers.unified_signature_tcga_mat.variable_exp <- gdac_selected_cancers.unified_signature_tcga_mat[rownames(gdac_selected_cancers.unified_signature_tcga_mat) %in% 
                                                                                         names(c(var1_cluster_unified_signature,
                                                                                                 var2_cluster_unified_signature)),] 
rowannot_variable_clusters <- data.frame(cluster = as.factor(c(var1_cluster_unified_signature,
                                                               var2_cluster_unified_signature)))
rownames(rowannot_variable_clusters) <- names(c(var1_cluster_unified_signature,
                                                var2_cluster_unified_signature))
pdf("unified_DEX_genes.middle_clusters.pdf")
unified_signature_tcga_mat.top_vs_bottom_clust <- pheatmap(mat = log10(gdac_selected_cancers.unified_signature_tcga_mat.variable_exp+1),
         show_colnames = FALSE,
         show_rownames = FALSE,
         cluster_cols = FALSE,
         fontsize_row = 2.5,
         annotation_col = gdac_selected_cancers_annot,
         annotation_row = rowannot_variable_clusters,
         cutree_rows = 2,
         legend_labels = c("log10(RSEM-scaled counts+1)"),
         main = c(paste("Middle-expressed gene sets of the",nrow(unified_sig_intersection),
                        "\nunique DEX genes in",ncol(gdac_selected_cancers.chiappinelli_tcga_mat),"TCGA samples")))
dev.off()
# plot(chiappinelli_tcga_clust)

```

```{r}
print("highest")
print(names(highest_cluster_unified_signature)) # mostly housekeeping stuff... 
print("lowest")
print(names(lowest_cluster_unified_signature))
```

### GSVA of the Consolidated signatures' four clusters

How enriched are the four main TCGA-IFN-viral signatures for different gene sets? 
```{r get_genesets}
prepare_gsva()
# from the Li et al oncotarget 2014 paper
interferon_genesets <- import_gmt(gmtfile = "../referenceAnnot/genesets/hg38/custom/msigdb_interferon.gmt")
cytochemokine_genesets <- import_gmt(gmtfile = "../referenceAnnot/genesets/hg38/custom/msigdb_cytochemokine.gmt")
inflammation_genesets <- import_gmt(gmtfile = "../referenceAnnot/genesets/hg38/custom/msigdb_inflammation.gmt")
antigen_genesets <- import_gmt(gmtfile = "../referenceAnnot/genesets/hg38/custom/msigdb_antigen.gmt")
# virus-related terms
virus_genesets <- import_gmt(gmtfile = "../referenceAnnot/genesets/hg38/custom/msigdb_virus.gmt")

```

Prepare counts
```{r}
med1_unified_sig <- unified_tcga_clust_hclusters[unified_tcga_clust_hclusters %in% 1]
med2_unified_sig <- unified_tcga_clust_hclusters[unified_tcga_clust_hclusters %in% 2]
lowest_unified_sig <- unified_tcga_clust_hclusters[unified_tcga_clust_hclusters %in% 3]
highest_unified_sig <- unified_tcga_clust_hclusters[unified_tcga_clust_hclusters %in% 4]

med1_unified_counts <- gdac_selected_cancers.unified_signature_tcga_mat[rownames(gdac_selected_cancers.unified_signature_tcga_mat) %in% 
                                                                          names(med1_unified_sig),]
med2_unified_counts <- gdac_selected_cancers.unified_signature_tcga_mat[rownames(gdac_selected_cancers.unified_signature_tcga_mat) %in% 
                                                                          names(med2_unified_sig),]
lowest_unified_counts <- gdac_selected_cancers.unified_signature_tcga_mat[rownames(gdac_selected_cancers.unified_signature_tcga_mat) %in% 
                                                                          names(lowest_unified_sig),]
highest_unified_counts <- gdac_selected_cancers.unified_signature_tcga_mat[rownames(gdac_selected_cancers.unified_signature_tcga_mat) %in% 
                                                                          names(highest_unified_sig),]

tcga_cluster_counts <- list(highest=highest_unified_counts,
                            med2=med2_unified_counts,
                            med1=med1_unified_counts,
                            lowest=highest_unified_counts)
```


```{}

tcga_gsva_enrich_by_clus <- lapply(seq_along(along.with = tcga_cluster_counts),
                                   function(x) {
                                     X <- tcga_cluster_counts[[x]]
                                     # IFN-related signatures
                                     ifn <- gsva_enrich(counts = X,
                                                        geneset = interferon_genesets)
                                     pheatmap(mat = ifn,
                                              annotation_col = gdac_selected_cancers_annot,
                                              show_colnames = FALSE,
                                              fontsize = 3.0,
                                              main=paste("IFN, cluster",names(tcga_cluster_counts)[x]))
                                     # # cytokine and chemokine-related signatures
                                     # cytochemo <- gsva_enrich(counts = X,
                                     #                    geneset = cytochemokine_genesets)
                                     # pheatmap(mat = cytochemo,
                                     #          annotation_col = gdac_selected_cancers_annot,
                                     #          show_colnames = FALSE,
                                     #          fontsize = 3.0,
                                     #          main=paste("Cytokine/Chemokine, cluster",names(tcga_cluster_counts)[x]))
                                     
                                     # inflammation signatures
                                     # inflamm <- gsva_enrich(counts = X,
                                     #                    geneset = inflammation_genesets)
                                     # pheatmap(mat = inflamm,
                                     #          annotation_col = gdac_selected_cancers_annot,
                                     #          show_colnames = FALSE,
                                     #          fontsize = 3.0,
                                     #          main=paste("Inflammation, cluster",names(tcga_cluster_counts)[x]))
                                     # # antigen presentation signatures
                                     # agpres <- gsva_enrich(counts = X,
                                     #                    geneset = antigen_genesets)
                                     # pheatmap(mat = agpres,
                                     #          annotation_col = gdac_selected_cancers_annot,
                                     #          show_colnames = FALSE,
                                     #          fontsize = 3.0,
                                     #          main=paste("Antigen presentation, cluster",names(tcga_cluster_counts)[x]))
                                     # 
                                     
                                     # viral <- gsva_enrich(counts = X,
                                     #                    geneset = virus_genesets)
                                     # pheatmap(mat = viral,
                                     #          annotation_col = gdac_selected_cancers_annot,
                                     #          show_colnames = FALSE,
                                     #          fontsize = 3.0,
                                     #          main=paste("Virus-related GOs, cluster",names(tcga_cluster_counts)[x]))
                                     # list(ifn,
                                     #      cytochemo,
                                     #      inflamm,
                                     #      agpres,
                                     #      viral)
                                   })

# # perform GSVA with GO terms from Li et al Oncotarget
# tcga_interferon_enrich <- gsva_enrich(counts = human_rnaseq_dge_fit_coeff2_lrt_top_up_gsva_counts$counts.aggregate,
#                                  geneset = interferon_genesets)
# tcga_cytochemokine_enrich <- gsva_enrich(counts = human_rnaseq_dge_fit_coeff2_lrt_top_up_gsva_counts$counts.aggregate,
#                                  geneset = cytochemokine_genesets)
# tcga_inflammation_enrich <- gsva_enrich(counts = human_rnaseq_dge_fit_coeff2_lrt_top_up_gsva_counts$counts.aggregate,
#                                  geneset = inflammation_genesets)
# tcga_antigen_enrich <- gsva_enrich(counts = human_rnaseq_dge_fit_coeff2_lrt_top_up_gsva_counts$counts.aggregate,
#                                  geneset = antigen_genesets)
# 
# # perform GSVA with viral terms
# tcga_virus_enrich <- gsva_enrich(counts = human_rnaseq_dge_fit_coeff2_lrt_top_up_gsva_counts$counts.aggregate,
#                                  geneset = virus_genesets)

```

Perform the GSVA enrichment for IFN-related signatures analysis
```{r}
ifn_highest <- gsva_enrich(counts = highest_unified_counts,
                           geneset = interferon_genesets) 
pheatmap(mat = ifn_highest,
         annotation_col = gdac_selected_cancers_annot,
         show_colnames = FALSE,
         fontsize = 5.0,
         main=paste("IFN, highest-expression cluster"))

# ifn_med2 <- gsva_enrich(counts = med2_unified_counts,geneset = interferon_genesets) # error is here
# pheatmap(mat = ifn_med2,
#          annotation_col = gdac_selected_cancers_annot,
#          show_colnames = FALSE,
#          fontsize = 3.0,
#          main=paste("IFN, medium-expression cluster #1"))

ifn_lowest <- gsva_enrich(counts = lowest_unified_counts,
                           geneset = interferon_genesets)
pheatmap(mat = ifn_lowest,
         annotation_col = gdac_selected_cancers_annot,
         show_colnames = FALSE,
         fontsize = 3.0,
         main=paste("IFN, lowest-expression cluster"))


```


```{r}
ifn_med1 <- gsva_enrich(counts = med1_unified_counts,
                           geneset = interferon_genesets)
pheatmap(mat = ifn_med1,
         annotation_col = gdac_selected_cancers_annot,
         show_colnames = FALSE,
         fontsize = 3.0,
         main=paste("IFN, medium-expression cluster #2"))

```

Consistently, a chunk of BRCA patients exhibit low and weak interferon responses for the high, low, and medium-expression unified signature clusters

# How mutually informative are the signatures

Finally, we would like to know how our three paper-specific IFN-viral signatures and our four TCGA-IFN-viral clusters. We can determine how correlated are the estimates of every pair of signatures (21 different pairs) for a particular tumor type. We use a bootstrapping procedure. 

Let $x_i = x_{i1} ... x_{in}$ and $x_j = x_{j1} ... x_{jm}$ represent the expression values of two signatures $i$ and $j$. We want to test how well-correlated would these signatures' mean and median values be for a tumor type. We draw $R$ bootstrap samples (say that $R = 100$) for each tumor sample. In each bootstrap sample, we draw $K$ pairs of medians (say $K=30$) from $n$ randomly-sampled gene expression values from one signature and $m$ randomly-sampled gene expression values from the other signature. We obtain the correlation coefficient across these $K$ pairs, and ultimately we obtain $R$ correlation coefficients (for which we obtain the mean correlation and the values at the 2.5th and 97.5th percentiles). We repeat this bootstrapping procedure for each tumor sample. 

The proposed elaborate bootstrapping procedure is intended to prevent batch-level effects within a particular tumor type, but we can modify it for a more simplified sampling procedure. Instead of performing $R \times K \times (n + m)$ samples for $N \approx 3000$ tumors (!), we will instead sample our $n$ and $m$ gene signature values across all tumor samples of a particular type so that we only have to repeat $R \times K \times (n + m)$ for 4 different tumor types (TCGA has 27 types in total, but we are just focused on 4 for now). We will have a faster procedure, but we will obtain estimates of signature correlations for a broad tumor type (and not for particular samples).

```{r signature_bootstrapping_fn}

signature_median <- function(sig1,sig2) {
  # draw the random sample from signature 1
  n_sig1 <- length(sig1)
  n_sig2 <- length(sig2)
  # boot_sig1 <- median(sample(sig1,n_sig1,replace=TRUE))
  boot_sig1 <- median(apply(X = sig1,
                            MARGIN=1,
                            FUN = function(x) {
                              sample(x,1,replace=TRUE)
                              }
                            ))
  # randomly sample n_sig1 entries from boot_sig1
  boot_sig1_random <- sample(boot_sig1,n_sig1,replace=TRUE)
  boot_sig2 <- median(apply(sig2,
                            MARGIN=1,
                            function(x) {
                              sample(x,1,replace=TRUE)
                              }
                            ))
  boot_sig2_random <- sample(boot_sig2,n_sig2,replace=TRUE)
  # boot_sig2 <- median(sample(sig2,n_sig2,replace=TRUE))
  
  boot_x <- c(boot_sig1,boot_sig2)
  return(boot_x)
}

estimate_corr_median_signatures <- function(sig1,sig2,K=30) {
  median_boot <- lapply(seq_len(K),function(x) {signature_median(sig1,sig2)})
  median_boot <- do.call(rbind,median_boot)
  boot_corr <- cor(x = median_boot[,1],y = median_boot[,2])
  return(boot_corr)
}

```

## For the human signatures from the three papers

```{r}
# chiappinelli vs canadas
R <- 100

# get OV samples
gdac_selected_cancers_annot_ov <- subset(gdac_selected_cancers_annot,TCGA %in% c("OV"))
sig1_counts <- gdac_selected_cancers.chiappinelli_tcga_mat[,colnames(gdac_selected_cancers.chiappinelli_tcga_mat) %in% rownames(gdac_selected_cancers_annot_ov)]
sig2_counts <- gdac_selected_cancers.canadas_tcga_mat[,colnames(gdac_selected_cancers.canadas_tcga_mat) %in% rownames(gdac_selected_cancers_annot_ov)]
bootstrap_corr_medians_ov <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
hist(bootstrap_corr_medians_ov)
mean_corr_ov.chiappinelli_canadas <- mean(bootstrap_corr_medians_ov)
ci_corr_ov.chiappinelli_canadas <- quantile(bootstrap_corr_medians_ov,c(0.025,0.975))
# test the correlation of the two signatures for the two signatures
ov_cor.chiappinelli_canadas <- lapply(seq_len(length.out = ncol(sig1_counts)),function(x) {c(median(sig1_counts[,x]),median(sig2_counts[,x]))})
ov_cor.chiappinelli_canadas <- do.call(rbind,ov_cor.chiappinelli_canadas)
plot(ov_cor.chiappinelli_canadas,pch=20,
     xlab=c("median value of chiappinelli signature"),
     ylab=c("median value of canadas signature"))
cor_ov_cor.chiappinelli_canadas <- cor(ov_cor.chiappinelli_canadas)
print(cor_ov_cor.chiappinelli_canadas > max(ci_corr_ov.chiappinelli_canadas) | cor_ov_cor.chiappinelli_canadas < max(ci_corr_ov.chiappinelli_canadas))

# get BRCA samples
gdac_selected_cancers_annot_brca <- subset(gdac_selected_cancers_annot,TCGA %in% c("BRCA"))
sig1_counts <- gdac_selected_cancers.chiappinelli_tcga_mat[,colnames(gdac_selected_cancers.chiappinelli_tcga_mat) %in% rownames(gdac_selected_cancers_annot_brca)]
sig2_counts <- gdac_selected_cancers.canadas_tcga_mat[,colnames(gdac_selected_cancers.canadas_tcga_mat) %in% rownames(gdac_selected_cancers_annot_brca)]
bootstrap_corr_medians_brca <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
hist(bootstrap_corr_medians_brca)
mean_corr_brca.chiappinelli_canadas <- mean(bootstrap_corr_medians_brca)
ci_corr_brca.chiappinelli_canadas <- quantile(bootstrap_corr_medians_brca,c(0.025,0.975))
# test the correlation of the two signatures for the two signatures
brca_cor.chiappinelli_canadas <- lapply(seq_len(length.out = ncol(sig1_counts)),function(x) {c(median(sig1_counts[,x]),median(sig2_counts[,x]))})
brca_cor.chiappinelli_canadas <- do.call(rbind,brca_cor.chiappinelli_canadas)
plot(brca_cor.chiappinelli_canadas,pch=20,
     xlab=c("median value of chiappinelli signature"),
     ylab=c("median value of canadas signature"))
cor_brca_cor.chiappinelli_canadas <- cor(brca_cor.chiappinelli_canadas)
print(cor_brca_cor.chiappinelli_canadas > max(ci_corr_brca.chiappinelli_canadas) | cor_brca_cor.chiappinelli_canadas < max(ci_corr_brca.chiappinelli_canadas))

# get COAD samples
gdac_selected_cancers_annot_coad <- subset(gdac_selected_cancers_annot,TCGA %in% c("COAD"))
sig1_counts <- gdac_selected_cancers.chiappinelli_tcga_mat[,colnames(gdac_selected_cancers.chiappinelli_tcga_mat) %in% rownames(gdac_selected_cancers_annot_coad)]
sig2_counts <- gdac_selected_cancers.canadas_tcga_mat[,colnames(gdac_selected_cancers.canadas_tcga_mat) %in% rownames(gdac_selected_cancers_annot_coad)]
bootstrap_corr_medians_coad <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
hist(bootstrap_corr_medians_coad)
mean_corr_coad.chiappinelli_canadas <- mean(bootstrap_corr_medians_coad)
ci_corr_coad.chiappinelli_canadas <- quantile(bootstrap_corr_medians_coad,c(0.025,0.975))
# test the correlation of the two signatures for the two signatures
coad_cor.chiappinelli_canadas <- lapply(seq_len(length.out = ncol(sig1_counts)),function(x) {c(median(sig1_counts[,x]),median(sig2_counts[,x]))})
coad_cor.chiappinelli_canadas <- do.call(rbind,coad_cor.chiappinelli_canadas)
plot(coad_cor.chiappinelli_canadas,pch=20,
     xlab=c("median value of chiappinelli signature"),
     ylab=c("median value of canadas signature"))
cor_coad_cor.chiappinelli_canadas <- cor(coad_cor.chiappinelli_canadas)
print(cor_coad_cor.chiappinelli_canadas > max(ci_corr_coad.chiappinelli_canadas) | cor_coad_cor.chiappinelli_canadas < max(ci_corr_coad.chiappinelli_canadas))

# get LUSC samples
gdac_selected_cancers_annot_lusc <- subset(gdac_selected_cancers_annot,TCGA %in% c("LUSC"))
sig1_counts <- gdac_selected_cancers.chiappinelli_tcga_mat[,colnames(gdac_selected_cancers.chiappinelli_tcga_mat) %in% rownames(gdac_selected_cancers_annot_lusc)]
sig2_counts <- gdac_selected_cancers.canadas_tcga_mat[,colnames(gdac_selected_cancers.canadas_tcga_mat) %in% rownames(gdac_selected_cancers_annot_lusc)]
bootstrap_corr_medians_lusc <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
hist(bootstrap_corr_medians_lusc)
mean_corr_lusc.chiappinelli_canadas <- mean(bootstrap_corr_medians_lusc)
ci_corr_lusc.chiappinelli_canadas <- quantile(bootstrap_corr_medians_lusc,c(0.025,0.975))
# test the correlation of the two signatures for the two signatures
lusc_cor.chiappinelli_canadas <- lapply(seq_len(length.out = ncol(sig1_counts)),function(x) {c(median(sig1_counts[,x]),median(sig2_counts[,x]))})
lusc_cor.chiappinelli_canadas <- do.call(rbind,lusc_cor.chiappinelli_canadas)
plot(lusc_cor.chiappinelli_canadas,pch=20,
     xlab=c("median value of chiappinelli signature"),
     ylab=c("median value of canadas signature"))
cor_lusc_cor.chiappinelli_canadas <- cor(lusc_cor.chiappinelli_canadas)
print(cor_lusc_cor.chiappinelli_canadas > max(ci_corr_lusc.chiappinelli_canadas) | cor_lusc_cor.chiappinelli_canadas < max(ci_corr_lusc.chiappinelli_canadas))

```


```{r}
# chiappinelli vs sheng/lafleur

# get OV samples
gdac_selected_cancers_annot_ov <- subset(gdac_selected_cancers_annot,TCGA %in% c("OV"))
sig1_counts <- gdac_selected_cancers.chiappinelli_tcga_mat[,colnames(gdac_selected_cancers.chiappinelli_tcga_mat) %in% rownames(gdac_selected_cancers_annot_ov)]
sig2_counts <- gdac_selected_cancers.sheng_lafleur_tcga_mat[,colnames(gdac_selected_cancers.sheng_lafleur_tcga_mat) %in% rownames(gdac_selected_cancers_annot_ov)]
bootstrap_corr_medians_ov <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
hist(bootstrap_corr_medians_ov.chiappinelli_sheng_lafleur)
mean_corr_ov.chiappinelli_sheng_lafleur <- mean(bootstrap_corr_medians_ov)
ci_corr_ov.chiappinelli_sheng_lafleur <- quantile(bootstrap_corr_medians_ov,c(0.025,0.975))
# test the correlation of the two signatures for the two signatures
ov_cor.chiappinelli_sheng_lafleur <- lapply(seq_len(length.out = ncol(sig1_counts)),function(x) {c(median(sig1_counts[,x]),median(sig2_counts[,x]))})
ov_cor.chiappinelli_sheng_lafleur <- do.call(rbind,ov_cor.chiappinelli_sheng_lafleur)
plot(ov_cor.chiappinelli_sheng_lafleur,pch=20,
     xlab=c("median value of chiappinelli signature"),
     ylab=c("median value of sheng_lafleur signature"))
cor_ov_cor.chiappinelli_sheng_lafleur <- cor(ov_cor.chiappinelli_sheng_lafleur)
print(cor_ov_cor.chiappinelli_sheng_lafleur > max(ci_corr_ov.chiappinelli_sheng_lafleur) | cor_ov_cor.chiappinelli_sheng_lafleur < max(ci_corr_ov.chiappinelli_sheng_lafleur))

# get BRCA samples
gdac_selected_cancers_annot_brca <- subset(gdac_selected_cancers_annot,TCGA %in% c("BRCA"))
sig1_counts <- gdac_selected_cancers.chiappinelli_tcga_mat[,colnames(gdac_selected_cancers.chiappinelli_tcga_mat) %in% rownames(gdac_selected_cancers_annot_brca)]
sig2_counts <- gdac_selected_cancers.sheng_lafleur_tcga_mat[,colnames(gdac_selected_cancers.sheng_lafleur_tcga_mat) %in% rownames(gdac_selected_cancers_annot_brca)]
bootstrap_corr_medians_brca <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
hist(bootstrap_corr_medians_brca.chiappinelli_sheng_lafleur)
mean_corr_brca.chiappinelli_sheng_lafleur <- mean(bootstrap_corr_medians_brca)
ci_corr_brca.chiappinelli_sheng_lafleur <- quantile(bootstrap_corr_medians_brca,c(0.025,0.975))
# test the correlation of the two signatures for the two signatures
brca_cor.chiappinelli_sheng_lafleur <- lapply(seq_len(length.out = ncol(sig1_counts)),function(x) {c(median(sig1_counts[,x]),median(sig2_counts[,x]))})
brca_cor.chiappinelli_sheng_lafleur <- do.call(rbind,brca_cor.chiappinelli_sheng_lafleur)
plot(brca_cor.chiappinelli_sheng_lafleur,pch=20,
     xlab=c("median value of chiappinelli signature"),
     ylab=c("median value of sheng_lafleur signature"))
cor_brca_cor.chiappinelli_sheng_lafleur <- cor(brca_cor.chiappinelli_sheng_lafleur)
print(cor_brca_cor.chiappinelli_sheng_lafleur > max(ci_corr_brca.chiappinelli_sheng_lafleur) | cor_brca_cor.chiappinelli_sheng_lafleur < max(ci_corr_brca.chiappinelli_sheng_lafleur))

# get COAD samples
gdac_selected_cancers_annot_coad <- subset(gdac_selected_cancers_annot,TCGA %in% c("COAD"))
sig1_counts <- gdac_selected_cancers.chiappinelli_tcga_mat[,colnames(gdac_selected_cancers.chiappinelli_tcga_mat) %in% rownames(gdac_selected_cancers_annot_coad)]
sig2_counts <- gdac_selected_cancers.sheng_lafleur_tcga_mat[,colnames(gdac_selected_cancers.sheng_lafleur_tcga_mat) %in% rownames(gdac_selected_cancers_annot_coad)]
bootstrap_corr_medians_coad <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
hist(bootstrap_corr_medians_coad.chiappinelli_sheng_lafleur)
mean_corr_coad.chiappinelli_sheng_lafleur <- mean(bootstrap_corr_medians_coad)
ci_corr_coad.chiappinelli_sheng_lafleur <- quantile(bootstrap_corr_medians_coad,c(0.025,0.975))
# test the correlation of the two signatures for the two signatures
coad_cor.chiappinelli_sheng_lafleur <- lapply(seq_len(length.out = ncol(sig1_counts)),function(x) {c(median(sig1_counts[,x]),median(sig2_counts[,x]))})
coad_cor.chiappinelli_sheng_lafleur <- do.call(rbind,coad_cor.chiappinelli_sheng_lafleur)
plot(coad_cor.chiappinelli_sheng_lafleur,pch=20,
     xlab=c("median value of chiappinelli signature"),
     ylab=c("median value of sheng_lafleur signature"))
cor_coad_cor.chiappinelli_sheng_lafleur <- cor(coad_cor.chiappinelli_sheng_lafleur)
print(cor_coad_cor.chiappinelli_sheng_lafleur > max(ci_corr_coad.chiappinelli_sheng_lafleur) | cor_coad_cor.chiappinelli_sheng_lafleur < max(ci_corr_coad.chiappinelli_sheng_lafleur))

# get LUSC samples
gdac_selected_cancers_annot_lusc <- subset(gdac_selected_cancers_annot,TCGA %in% c("LUSC"))
sig1_counts <- gdac_selected_cancers.chiappinelli_tcga_mat[,colnames(gdac_selected_cancers.chiappinelli_tcga_mat) %in% rownames(gdac_selected_cancers_annot_lusc)]
sig2_counts <- gdac_selected_cancers.sheng_lafleur_tcga_mat[,colnames(gdac_selected_cancers.sheng_lafleur_tcga_mat) %in% rownames(gdac_selected_cancers_annot_lusc)]
bootstrap_corr_medians_lusc <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
hist(bootstrap_corr_medians_lusc.chiappinelli_sheng_lafleur)
mean_corr_lusc.chiappinelli_sheng_lafleur <- mean(bootstrap_corr_medians_lusc)
ci_corr_lusc.chiappinelli_sheng_lafleur <- quantile(bootstrap_corr_medians_lusc,c(0.025,0.975))
# test the correlation of the two signatures for the two signatures
lusc_cor.chiappinelli_sheng_lafleur <- lapply(seq_len(length.out = ncol(sig1_counts)),function(x) {c(median(sig1_counts[,x]),median(sig2_counts[,x]))})
lusc_cor.chiappinelli_sheng_lafleur <- do.call(rbind,lusc_cor.chiappinelli_sheng_lafleur)
plot(lusc_cor.chiappinelli_sheng_lafleur,pch=20,
     xlab=c("median value of chiappinelli signature"),
     ylab=c("median value of sheng_lafleur signature"))
cor_lusc_cor.chiappinelli_sheng_lafleur <- cor(lusc_cor.chiappinelli_sheng_lafleur)
print(cor_lusc_cor.chiappinelli_sheng_lafleur > max(ci_corr_lusc.chiappinelli_sheng_lafleur) | cor_lusc_cor.chiappinelli_sheng_lafleur < max(ci_corr_lusc.chiappinelli_sheng_lafleur))


```

```{r}
# canadas vs sheng/lafleur
# get OV samples
gdac_selected_cancers_annot_ov <- subset(gdac_selected_cancers_annot,TCGA %in% c("OV"))
sig1_counts <- gdac_selected_cancers.canadas_tcga_mat[,colnames(gdac_selected_cancers.canadas_tcga_mat) %in% rownames(gdac_selected_cancers_annot_ov)]
sig2_counts <- gdac_selected_cancers.sheng_lafleur_tcga_mat[,colnames(gdac_selected_cancers.sheng_lafleur_tcga_mat) %in% rownames(gdac_selected_cancers_annot_ov)]
bootstrap_corr_medians_ov <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
hist(bootstrap_corr_medians_ov.canadas_sheng_lafleur)
mean_corr_ov.canadas_sheng_lafleur <- mean(bootstrap_corr_medians_ov)
ci_corr_ov.canadas_sheng_lafleur <- quantile(bootstrap_corr_medians_ov,c(0.025,0.975))
# test the correlation of the two signatures for the two signatures
ov_cor.canadas_sheng_lafleur <- lapply(seq_len(length.out = ncol(sig1_counts)),function(x) {c(median(sig1_counts[,x]),median(sig2_counts[,x]))})
ov_cor.canadas_sheng_lafleur <- do.call(rbind,ov_cor.canadas_sheng_lafleur)
plot(ov_cor.canadas_sheng_lafleur,pch=20,
     xlab=c("median value of canadas signature"),
     ylab=c("median value of sheng_lafleur signature"))
cor_ov_cor.canadas_sheng_lafleur <- cor(ov_cor.canadas_sheng_lafleur)
print(cor_ov_cor.canadas_sheng_lafleur > max(ci_corr_ov.canadas_sheng_lafleur) | cor_ov_cor.canadas_sheng_lafleur < max(ci_corr_ov.canadas_sheng_lafleur))

# get BRCA samples
gdac_selected_cancers_annot_brca <- subset(gdac_selected_cancers_annot,TCGA %in% c("BRCA"))
sig1_counts <- gdac_selected_cancers.canadas_tcga_mat[,colnames(gdac_selected_cancers.canadas_tcga_mat) %in% rownames(gdac_selected_cancers_annot_brca)]
sig2_counts <- gdac_selected_cancers.sheng_lafleur_tcga_mat[,colnames(gdac_selected_cancers.sheng_lafleur_tcga_mat) %in% rownames(gdac_selected_cancers_annot_brca)]
bootstrap_corr_medians_brca <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
hist(bootstrap_corr_medians_brca.canadas_sheng_lafleur)
mean_corr_brca.canadas_sheng_lafleur <- mean(bootstrap_corr_medians_brca)
ci_corr_brca.canadas_sheng_lafleur <- quantile(bootstrap_corr_medians_brca,c(0.025,0.975))
# test the correlation of the two signatures for the two signatures
brca_cor.canadas_sheng_lafleur <- lapply(seq_len(length.out = ncol(sig1_counts)),function(x) {c(median(sig1_counts[,x]),median(sig2_counts[,x]))})
brca_cor.canadas_sheng_lafleur <- do.call(rbind,brca_cor.canadas_sheng_lafleur)
plot(brca_cor.canadas_sheng_lafleur,pch=20,
     xlab=c("median value of canadas signature"),
     ylab=c("median value of sheng_lafleur signature"))
cor_brca_cor.canadas_sheng_lafleur <- cor(brca_cor.canadas_sheng_lafleur)
print(cor_brca_cor.canadas_sheng_lafleur > max(ci_corr_brca.canadas_sheng_lafleur) | cor_brca_cor.canadas_sheng_lafleur < max(ci_corr_brca.canadas_sheng_lafleur))

# get COAD samples
gdac_selected_cancers_annot_coad <- subset(gdac_selected_cancers_annot,TCGA %in% c("COAD"))
sig1_counts <- gdac_selected_cancers.canadas_tcga_mat[,colnames(gdac_selected_cancers.canadas_tcga_mat) %in% rownames(gdac_selected_cancers_annot_coad)]
sig2_counts <- gdac_selected_cancers.sheng_lafleur_tcga_mat[,colnames(gdac_selected_cancers.sheng_lafleur_tcga_mat) %in% rownames(gdac_selected_cancers_annot_coad)]
bootstrap_corr_medians_coad <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
hist(bootstrap_corr_medians_coad.canadas_sheng_lafleur)
mean_corr_coad.canadas_sheng_lafleur <- mean(bootstrap_corr_medians_coad)
ci_corr_coad.canadas_sheng_lafleur <- quantile(bootstrap_corr_medians_coad,c(0.025,0.975))
# test the correlation of the two signatures for the two signatures
coad_cor.canadas_sheng_lafleur <- lapply(seq_len(length.out = ncol(sig1_counts)),function(x) {c(median(sig1_counts[,x]),median(sig2_counts[,x]))})
coad_cor.canadas_sheng_lafleur <- do.call(rbind,coad_cor.canadas_sheng_lafleur)
plot(coad_cor.canadas_sheng_lafleur,pch=20,
     xlab=c("median value of canadas signature"),
     ylab=c("median value of sheng_lafleur signature"))
cor_coad_cor.canadas_sheng_lafleur <- cor(coad_cor.canadas_sheng_lafleur)
print(cor_coad_cor.canadas_sheng_lafleur > max(ci_corr_coad.canadas_sheng_lafleur) | cor_coad_cor.canadas_sheng_lafleur < max(ci_corr_coad.canadas_sheng_lafleur))

# get LUSC samples
gdac_selected_cancers_annot_lusc <- subset(gdac_selected_cancers_annot,TCGA %in% c("LUSC"))
sig1_counts <- gdac_selected_cancers.canadas_tcga_mat[,colnames(gdac_selected_cancers.canadas_tcga_mat) %in% rownames(gdac_selected_cancers_annot_lusc)]
sig2_counts <- gdac_selected_cancers.sheng_lafleur_tcga_mat[,colnames(gdac_selected_cancers.sheng_lafleur_tcga_mat) %in% rownames(gdac_selected_cancers_annot_lusc)]
bootstrap_corr_medians_lusc <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
hist(bootstrap_corr_medians_lusc.canadas_sheng_lafleur)
mean_corr_lusc.canadas_sheng_lafleur <- mean(bootstrap_corr_medians_lusc)
ci_corr_lusc.canadas_sheng_lafleur <- quantile(bootstrap_corr_medians_lusc,c(0.025,0.975))
# test the correlation of the two signatures for the two signatures
lusc_cor.canadas_sheng_lafleur <- lapply(seq_len(length.out = ncol(sig1_counts)),function(x) {c(median(sig1_counts[,x]),median(sig2_counts[,x]))})
lusc_cor.canadas_sheng_lafleur <- do.call(rbind,lusc_cor.canadas_sheng_lafleur)
plot(lusc_cor.canadas_sheng_lafleur,pch=20,
     xlab=c("median value of canadas signature"),
     ylab=c("median value of sheng_lafleur signature"))
cor_lusc_cor.canadas_sheng_lafleur <- cor(lusc_cor.canadas_sheng_lafleur)
print(cor_lusc_cor.canadas_sheng_lafleur > max(ci_corr_lusc.canadas_sheng_lafleur) | cor_lusc_cor.canadas_sheng_lafleur < max(ci_corr_lusc.canadas_sheng_lafleur))
```



## For the four clusters from TCGA-wide analysis

Just show the highest-expressed and lowest expressed for now
```{r}
# highest vs lowest
# get OV samples
gdac_selected_cancers_annot_ov <- subset(gdac_selected_cancers_annot,TCGA %in% c("OV"))
sig1_counts <- highest_unified_counts[,colnames(highest_unified_counts) %in% rownames(gdac_selected_cancers_annot_ov)]
sig2_counts <- lowest_unified_counts[,colnames(lowest_unified_counts) %in% rownames(gdac_selected_cancers_annot_ov)]
bootstrap_corr_medians_ov.highest_vs_lowest <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
mean_corr_ov.highest_vs_lowest <- mean(bootstrap_corr_medians_ov.highest_vs_lowest)
ci_corr_ov.highest_vs_lowest <- quantile(bootstrap_corr_medians_ov.highest_vs_lowest,c(0.025,0.975))

# get BRCA samples
gdac_selected_cancers_annot_brca <- subset(gdac_selected_cancers_annot,TCGA %in% c("BRCA"))
sig1_counts <- gdac_selected_cancers.canadas_tcga_mat[,colnames(gdac_selected_cancers.canadas_tcga_mat) %in% rownames(gdac_selected_cancers_annot_brca)]
sig2_counts <- gdac_selected_cancers.sheng_lafleur_tcga_mat[,colnames(gdac_selected_cancers.sheng_lafleur_tcga_mat) %in% rownames(gdac_selected_cancers_annot_brca)]
bootstrap_corr_medians_brca.highest_vs_lowest <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
mean_corr_brca.highest_vs_lowest <- mean(bootstrap_corr_medians_brca.highest_vs_lowest)
ci_corr_brca.highest_vs_lowest <- quantile(bootstrap_corr_medians_brca.highest_vs_lowest,c(0.025,0.975))

# get COAD samples
gdac_selected_cancers_annot_coad. <- subset(gdac_selected_cancers_annot,TCGA %in% c("COAD"))
sig1_counts <- gdac_selected_cancers.canadas_tcga_mat[,colnames(gdac_selected_cancers.canadas_tcga_mat) %in% rownames(gdac_selected_cancers_annot_coad)]
sig2_counts <- gdac_selected_cancers.sheng_lafleur_tcga_mat[,colnames(gdac_selected_cancers.sheng_lafleur_tcga_mat) %in% rownames(gdac_selected_cancers_annot_coad)]
bootstrap_corr_medians_coad.highest_vs_lowest <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
mean_corr_coad.highest_vs_lowest <- mean(bootstrap_corr_medians_coad.highest_vs_lowest)
ci_corr_coad.highest_vs_lowest <- quantile(bootstrap_corr_medians_coad.highest_vs_lowest,c(0.025,0.975))

# get LUSC samples
gdac_selected_cancers_annot_lusc <- subset(gdac_selected_cancers_annot,TCGA %in% c("LUSC"))
sig1_counts <- gdac_selected_cancers.canadas_tcga_mat[,colnames(gdac_selected_cancers.canadas_tcga_mat) %in% rownames(gdac_selected_cancers_annot_lusc)]
sig2_counts <- gdac_selected_cancers.sheng_lafleur_tcga_mat[,colnames(gdac_selected_cancers.sheng_lafleur_tcga_mat) %in% rownames(gdac_selected_cancers_annot_lusc)]
bootstrap_corr_medians_lusc.highest_vs_lowest <- sapply(X = seq_len(R),
                                 function(x) {
                                   estimate_corr_median_signatures(sig1_counts,sig2_counts)
                                 })
mean_corr_lusc.highest_vs_lowest <- mean(bootstrap_corr_medians_lusc.highest_vs_lowest)
ci_corr_lusc.highest_vs_lowest <- quantile(bootstrap_corr_medians_lusc.highest_vs_lowest,c(0.025,0.975))


# plots
hist(bootstrap_corr_medians_ov.highest_vs_lowest)
hist(bootstrap_corr_medians_brca.highest_vs_lowest)
hist(bootstrap_corr_medians_coad.highest_vs_lowest)
hist(bootstrap_corr_medians_lusc.highest_vs_lowest)
```

```{r}
hist(bootstrap_corr_medians_ov.highest_vs_lowest)
hist(bootstrap_corr_medians_brca.highest_vs_lowest)
hist(bootstrap_corr_medians_coad.highest_vs_lowest)
hist(bootstrap_corr_medians_lusc.highest_vs_lowest)

```

# Correlation with outputs from TIMER

Finally, we would like to correlated the estimated enrichments of immune subsets (based on TIMER analysis) with the three paper's signatures

```{r import_timer_Fmat}

# load FMATs
brca_fmat <- read.table(file = "timer/results/brca/brca_Fmat.txt",header = TRUE,sep = "\t",row.names = 1)
ov_fmat <- read.table(file = "timer/results/ov/ov_Fmat.txt",header = TRUE,sep = "\t",row.names = 1)
lusc_fmat <- read.table(file = "timer/results/lusc/lusc_Fmat.txt",header = TRUE,sep = "\t",row.names = 1)
coad_fmat <- read.table(file = "timer/results/coad/coad_Fmat.txt",header = TRUE,sep = "\t",row.names = 1)

# get median expression of the signature in each entry
get_median_signature <- function(counts) {
  x_median <- apply(X = counts,
                    MARGIN = 2,FUN = median)
  return(x_median)
}

# match Fmat rownames with tcga column names

match_fmat_tcga_names <- function(fmat_names,tcga_names) {
  # as.character
  fmat_names <- as.character(unlist(fmat_names))
  tcga_names <- as.character(unlist(tcga_names))
  # fmat_names has dashes instead of dots
  fmat_names_sub <- gsub(pattern = "-",replacement = ".",x = fmat_names,fixed = TRUE)
  match_index <- sapply(X = fmat_names_sub,
                        FUN = function(x) {
                          x_ind <- grep(pattern = x,x = tcga_names,fixed = TRUE)
                          x_ind
                        })
  # name the indices
  names(match_index) <- fmat_names_sub
  # return
  return(match_index)
}

```

Get medians of the signature across the cancers (use the log10 of the counts)
```{r}
chiappinelli_ifn_signature_median <- get_median_signature(counts = log10(gdac_selected_cancers.chiappinelli_tcga_mat))
canadas_ifn_signature_median <- get_median_signature(counts = log10(gdac_selected_cancers.canadas_tcga_mat))
canadas_3putr_ifn_signature_median <- get_median_signature(counts = log10(gdac_selected_cancers.canadas_tcga_mat_3putr)) # 3putr
sheng_lafleur_ifn_signature_median <- get_median_signature(counts = log10(gdac_selected_cancers.sheng_lafleur_tcga_mat))
```

Go by cancer
```{r}
# BRCA
temp_i <- match_fmat_tcga_names(fmat_names = rownames(brca_fmat),
                                tcga_names = names(chiappinelli_ifn_signature_median))
chiappinelli_brca_timer_signature <- chiappinelli_ifn_signature_median[temp_i]
temp_i <- match_fmat_tcga_names(fmat_names = rownames(brca_fmat),
                                tcga_names = names(canadas_ifn_signature_median))
canadas_brca_timer_signature <- canadas_ifn_signature_median[temp_i]
temp_i <- match_fmat_tcga_names(fmat_names = rownames(brca_fmat),
                                tcga_names = names(canadas_3putr_ifn_signature_median))
canadas_3putr_brca_timer_signature <- canadas_3putr_ifn_signature_median[temp_i]
temp_i <- match_fmat_tcga_names(fmat_names = rownames(brca_fmat),
                                tcga_names = names(sheng_lafleur_ifn_signature_median))
sheng_lafleur_brca_timer_signature <- sheng_lafleur_ifn_signature_median[temp_i]

# colors for fmat
brca_fmat_colors <- rainbow(n = ncol(brca_fmat))
ylim_all <- max(canadas_3putr_brca_timer_signature,
                canadas_brca_timer_signature,
                chiappinelli_brca_timer_signature,
                sheng_lafleur_brca_timer_signature)
ylim_min_all <- min(canadas_3putr_brca_timer_signature,
                canadas_brca_timer_signature,
                chiappinelli_brca_timer_signature,
                sheng_lafleur_brca_timer_signature)

ylim_all <- 4.0
ylim_min_all <- 0.0
xlim_all <- 3.0
xlim_min_all <- 0.0

# plot correlations
par(mfrow = c(1,3))
plot(x = brca_fmat[,1],
     y = chiappinelli_brca_timer_signature,
     xlim=c(xlim_min_all,xlim_all),
     ylim=c(ylim_min_all,ylim_all),
     xlab=c("TIMER value"),
     ylab=c("Median IFN-viral signature value (log10(counts))"),
     type = "n",
     main = paste0("Chiappinelli"))
chiappinelli_brca_timer_signature_cor <- lapply(X = seq_along(brca_fmat),
      function(x) {
        points(y = chiappinelli_brca_timer_signature,
             x = brca_fmat[,x],
             pch=20,
             col = brca_fmat_colors[x])
        timer_positive_subset_ind <- brca_fmat[,x] > 0
        cor(x = chiappinelli_brca_timer_signature[timer_positive_subset_ind],
            y = brca_fmat[timer_positive_subset_ind,x])
      })
names(chiappinelli_brca_timer_signature_cor) <- names(brca_fmat)
legend(x = "bottomright",
       legend = colnames(brca_fmat),
       col = brca_fmat_colors,
       pch=20,
       cex = 0.75)

# plot correlations
plot(x = brca_fmat[,1],
     y = canadas_brca_timer_signature,
     xlim=c(xlim_min_all,xlim_all),
     ylim=c(ylim_min_all,ylim_all),
     xlab=c("TIMER value"),
     ylab=c("Median IFN-viral signature value (log10(counts))"),
     type = "n",
     main = paste0("Canadas"))
canadas_brca_timer_signature_cor <- lapply(X = seq_along(brca_fmat),
      function(x) {
        points(y = canadas_brca_timer_signature,
             x = brca_fmat[,x],
             pch=20,
             col = brca_fmat_colors[x])
        timer_positive_subset_ind <- brca_fmat[,x] > 0
        cor(x = canadas_brca_timer_signature[timer_positive_subset_ind],
            y = brca_fmat[timer_positive_subset_ind,x])
      })
names(canadas_brca_timer_signature_cor) <- names(brca_fmat)
legend(x = "bottomright",
       legend = colnames(brca_fmat),
       col = brca_fmat_colors,
       pch=20,
       cex = 0.75)

# # plot correlations for Canadas 3' UTR ERV-containing genes (sense and antisense)
# plot(x = brca_fmat[,1],
#      y = canadas_3putr_brca_timer_signature,
#      xlim=c(0,max(brca_fmat)),
#      ylim=c(0,ylim_all),
#      xlab=c("TIMER value"),
#      ylab=c("Median IFN-viral signature value (log10(counts))"),
#      type = "n",
#      main = paste0("Canadas (3' UTR ERV-containing genes) signature-TIMER correlation, BRCA"))
# canadas_3putr_brca_timer_signature_cor <- lapply(X = seq_along(brca_fmat),
#       function(x) {
#         points(y = canadas_3putr_brca_timer_signature,
#              x = brca_fmat[,x],
#              pch=20,
#              col = brca_fmat_colors[x])
#         timer_positive_subset_ind <- brca_fmat[,x] > 0
#         cor(x = canadas_3putr_brca_timer_signature[timer_positive_subset_ind],
#             y = brca_fmat[timer_positive_subset_ind,x])
#       })
# names(canadas_3putr_brca_timer_signature_cor) <- names(brca_fmat)
# legend(x = "bottomright",
#        legend = colnames(brca_fmat),
#        col = brca_fmat_colors,
#        pch=20,
#        cex = 0.75)

# plot correlations
plot(x = brca_fmat[,1],
     y = sheng_lafleur_brca_timer_signature,
     xlim=c(xlim_min_all,xlim_all),
     ylim=c(ylim_min_all,ylim_all),
     xlab=c("TIMER value"),
     ylab=c("Median IFN-viral signature value (log10(counts))"),
     type = "n",
     main = paste0("Sheng and LaFleur"))
sheng_lafleur_brca_timer_signature_cor <- lapply(X = seq_along(brca_fmat),
      function(x) {
        points(y = sheng_lafleur_brca_timer_signature,
             x = brca_fmat[,x],
             pch=20,
             col = brca_fmat_colors[x])
        timer_positive_subset_ind <- brca_fmat[,x] > 0
        cor(x = sheng_lafleur_brca_timer_signature[timer_positive_subset_ind],
            y = brca_fmat[timer_positive_subset_ind,x])
      })
names(sheng_lafleur_brca_timer_signature_cor) <- names(brca_fmat)
legend(x = "bottomright",
       legend = colnames(brca_fmat),
       col = brca_fmat_colors,
       pch=20,
       cex = 0.75)

# mark the tumors with no CD8 infiltrates
gdac_selected_cancers_annot_brca$CD8 <- brca_fmat$T_cell.CD8
gdac_selected_cancers_annot_brca$CD4 <- brca_fmat$T_cell.CD4
gdac_selected_cancers_annot_brca$B_cell <- brca_fmat$B_cell
gdac_selected_cancers_annot_brca$Neutrophil <- brca_fmat$Neutrophil
gdac_selected_cancers_annot_brca$Macrophage <- brca_fmat$Macrophage
gdac_selected_cancers_annot_brca$DC <- brca_fmat$DC



```

```{r}
# OV
temp_i <- match_fmat_tcga_names(fmat_names = rownames(ov_fmat),
                                tcga_names = names(chiappinelli_ifn_signature_median))
chiappinelli_ov_timer_signature <- chiappinelli_ifn_signature_median[temp_i]
temp_i <- match_fmat_tcga_names(fmat_names = rownames(ov_fmat),
                                tcga_names = names(canadas_ifn_signature_median))
canadas_ov_timer_signature <- canadas_ifn_signature_median[temp_i]
temp_i <- match_fmat_tcga_names(fmat_names = rownames(ov_fmat),
                                tcga_names = names(canadas_3putr_ifn_signature_median))
canadas_3putr_ov_timer_signature <- canadas_3putr_ifn_signature_median[temp_i]
temp_i <- match_fmat_tcga_names(fmat_names = rownames(ov_fmat),
                                tcga_names = names(sheng_lafleur_ifn_signature_median))
sheng_lafleur_ov_timer_signature <- sheng_lafleur_ifn_signature_median[temp_i]

# colors for fmat
ov_fmat_colors <- rainbow(n = ncol(ov_fmat))

# plot correlations
par(mfrow = c(1,3))
plot(x = ov_fmat[,1],
     y = chiappinelli_ov_timer_signature,
     xlim=c(xlim_min_all,xlim_all),
     ylim=c(ylim_min_all,ylim_all),
     xlab=c("TIMER value"),
     ylab=c("Median IFN-viral signature value (log10(counts))"),
     type = "n",
     main = paste0("Chiappinelli"))
chiappinelli_ov_timer_signature_cor <- lapply(X = seq_along(ov_fmat),
      function(x) {
        points(y = chiappinelli_ov_timer_signature,
             x = ov_fmat[,x],
             pch=20,
             col = ov_fmat_colors[x])
        timer_positive_subset_ind <- ov_fmat[,x] > 0
        cor(x = chiappinelli_ov_timer_signature[timer_positive_subset_ind],
            y = ov_fmat[timer_positive_subset_ind,x])
      })
names(chiappinelli_ov_timer_signature_cor) <- names(ov_fmat)
legend(x = "bottomright",
       legend = colnames(ov_fmat),
       col = ov_fmat_colors,
       pch=20,
       cex = 0.75)

# plot correlations
plot(x = ov_fmat[,1],
     y = canadas_ov_timer_signature,
     xlim=c(xlim_min_all,xlim_all),
     ylim=c(ylim_min_all,ylim_all),
     xlab=c("TIMER value"),
     ylab=c("Median IFN-viral signature value (log10(counts))"),
     type = "n",
     main = paste0("Canadas"))
canadas_ov_timer_signature_cor <- lapply(X = seq_along(ov_fmat),
      function(x) {
        points(y = canadas_ov_timer_signature,
             x = ov_fmat[,x],
             pch=20,
             col = ov_fmat_colors[x])
        timer_positive_subset_ind <- ov_fmat[,x] > 0
        cor(x = canadas_ov_timer_signature[timer_positive_subset_ind],
            y = ov_fmat[timer_positive_subset_ind,x])
      })
names(canadas_ov_timer_signature_cor) <- names(ov_fmat)
legend(x = "bottomright",
       legend = colnames(ov_fmat),
       col = ov_fmat_colors,
       pch=20,
       cex = 0.75)

# # plot correlations for Canadas 3' UTR ERV-containing genes (sense and antisense)
# plot(x = ov_fmat[,1],
#      y = canadas_3putr_ov_timer_signature,
#      xlim=c(0,max(ov_fmat)),
#      ylim=c(0,ylim_all),
#      xlab=c("TIMER value"),
#      ylab=c("Median IFN-viral signature value (log10(counts))"),
#      type = "n",
#      main = paste0("Canadas (3' UTR ERV-containing genes) signature-TIMER correlation, ov"))
# canadas_3putr_ov_timer_signature_cor <- lapply(X = seq_along(ov_fmat),
#       function(x) {
#         points(y = canadas_3putr_ov_timer_signature,
#              x = ov_fmat[,x],
#              pch=20,
#              col = ov_fmat_colors[x])
#         timer_positive_subset_ind <- ov_fmat[,x] > 0
#         cor(x = canadas_3putr_ov_timer_signature[timer_positive_subset_ind],
#             y = ov_fmat[timer_positive_subset_ind,x])
#       })
# names(canadas_3putr_ov_timer_signature_cor) <- names(ov_fmat)
# legend(x = "bottomright",
#        legend = colnames(ov_fmat),
#        col = ov_fmat_colors,
#        pch=20,
#        cex = 0.75)

# plot correlations
plot(x = ov_fmat[,1],
     y = sheng_lafleur_ov_timer_signature,
     xlim=c(xlim_min_all,xlim_all),
     ylim=c(ylim_min_all,ylim_all),
     xlab=c("TIMER value"),
     ylab=c("Median IFN-viral signature value (log10(counts))"),
     type = "n",
     main = paste0("Sheng and LaFleur"))
sheng_lafleur_ov_timer_signature_cor <- lapply(X = seq_along(ov_fmat),
      function(x) {
        points(y = sheng_lafleur_ov_timer_signature,
             x = ov_fmat[,x],
             pch=20,
             col = ov_fmat_colors[x])
        timer_positive_subset_ind <- ov_fmat[,x] > 0
        cor(x = sheng_lafleur_ov_timer_signature[timer_positive_subset_ind],
            y = ov_fmat[timer_positive_subset_ind,x])
      })
names(sheng_lafleur_ov_timer_signature_cor) <- names(ov_fmat)
legend(x = "bottomright",
       legend = colnames(ov_fmat),
       col = ov_fmat_colors,
       pch=20,
       cex = 0.75)

# mark the tumors with no CD8 infiltrates
gdac_selected_cancers_annot_ov$CD8 <- ov_fmat$T_cell.CD8
gdac_selected_cancers_annot_ov$CD4 <- ov_fmat$T_cell.CD4
gdac_selected_cancers_annot_ov$B_cell <- ov_fmat$B_cell
gdac_selected_cancers_annot_ov$Neutrophil <- ov_fmat$Neutrophil
gdac_selected_cancers_annot_ov$Macrophage <- ov_fmat$Macrophage
gdac_selected_cancers_annot_ov$DC <- ov_fmat$DC

```

```{r}

# lusc
temp_i <- match_fmat_tcga_names(fmat_names = rownames(lusc_fmat),
                                tcga_names = names(chiappinelli_ifn_signature_median))
chiappinelli_lusc_timer_signature <- chiappinelli_ifn_signature_median[temp_i]
temp_i <- match_fmat_tcga_names(fmat_names = rownames(lusc_fmat),
                                tcga_names = names(canadas_ifn_signature_median))
canadas_lusc_timer_signature <- canadas_ifn_signature_median[temp_i]
temp_i <- match_fmat_tcga_names(fmat_names = rownames(lusc_fmat),
                                tcga_names = names(canadas_3putr_ifn_signature_median))
canadas_3putr_lusc_timer_signature <- canadas_3putr_ifn_signature_median[temp_i]
temp_i <- match_fmat_tcga_names(fmat_names = rownames(lusc_fmat),
                                tcga_names = names(sheng_lafleur_ifn_signature_median))
sheng_lafleur_lusc_timer_signature <- sheng_lafleur_ifn_signature_median[temp_i]

# colors for fmat
lusc_fmat_colors <- rainbow(n = ncol(lusc_fmat))

# plot correlations
par(mfrow = c(1,3))
plot(x = lusc_fmat[,1],
     y = chiappinelli_lusc_timer_signature,
     xlim=c(xlim_min_all,xlim_all),
     ylim=c(ylim_min_all,ylim_all),
     xlab=c("TIMER value"),
     ylab=c("Median IFN-viral signature value (log10(counts))"),
     type = "n",
     main = paste0("Chiappinelli"))
chiappinelli_lusc_timer_signature_cor <- lapply(X = seq_along(lusc_fmat),
      function(x) {
        points(y = chiappinelli_lusc_timer_signature,
             x = lusc_fmat[,x],
             pch=20,
             col = lusc_fmat_colors[x])
        timer_positive_subset_ind <- lusc_fmat[,x] > 0
        cor(x = chiappinelli_lusc_timer_signature[timer_positive_subset_ind],
            y = lusc_fmat[timer_positive_subset_ind,x])
      })
names(chiappinelli_lusc_timer_signature_cor) <- names(lusc_fmat)
legend(x = "bottomright",
       legend = colnames(lusc_fmat),
       col = lusc_fmat_colors,
       pch=20,
       cex = 0.75)

# plot correlations
plot(x = lusc_fmat[,1],
     y = canadas_lusc_timer_signature,
     xlim=c(xlim_min_all,xlim_all),
     ylim=c(ylim_min_all,ylim_all),
     xlab=c("TIMER value"),
     ylab=c("Median IFN-viral signature value (log10(counts))"),
     type = "n",
     main = paste0("Canadas"))
canadas_lusc_timer_signature_cor <- lapply(X = seq_along(lusc_fmat),
      function(x) {
        points(y = canadas_lusc_timer_signature,
             x = lusc_fmat[,x],
             pch=20,
             col = lusc_fmat_colors[x])
        timer_positive_subset_ind <- lusc_fmat[,x] > 0
        cor(x = canadas_lusc_timer_signature[timer_positive_subset_ind],
            y = lusc_fmat[timer_positive_subset_ind,x])
      })
names(canadas_lusc_timer_signature_cor) <- names(lusc_fmat)
legend(x = "bottomright",
       legend = colnames(lusc_fmat),
       col = lusc_fmat_colors,
       pch=20,
       cex = 0.75)

# # plot correlations for Canadas 3' UTR ERV-containing genes (sense and antisense)
# plot(x = lusc_fmat[,1],
#      y = canadas_3putr_lusc_timer_signature,
#      xlim=c(0,max(lusc_fmat)),
#      ylim=c(0,ylim_all),
#      xlab=c("TIMER value"),
#      ylab=c("Median IFN-viral signature value (log10(counts))"),
#      type = "n",
#      main = paste0("Canadas (3' UTR ERV-containing genes) signature-TIMER correlation, lusc"))
# canadas_3putr_lusc_timer_signature_cor <- lapply(X = seq_along(lusc_fmat),
#       function(x) {
#         points(y = canadas_3putr_lusc_timer_signature,
#              x = lusc_fmat[,x],
#              pch=20,
#              col = lusc_fmat_colors[x])
#         timer_positive_subset_ind <- lusc_fmat[,x] > 0
#         cor(x = canadas_3putr_lusc_timer_signature[timer_positive_subset_ind],
#             y = lusc_fmat[timer_positive_subset_ind,x])
#       })
# names(canadas_3putr_lusc_timer_signature_cor) <- names(lusc_fmat)
# legend(x = "bottomright",
#        legend = colnames(lusc_fmat),
#        col = lusc_fmat_colors,
#        pch=20,
#        cex = 0.75)

# plot correlations
plot(x = lusc_fmat[,1],
     y = sheng_lafleur_lusc_timer_signature,
     xlim=c(xlim_min_all,xlim_all),
     ylim=c(ylim_min_all,ylim_all),
     xlab=c("TIMER value"),
     ylab=c("Median IFN-viral signature value (log10(counts))"),
     type = "n",
     main = paste0("Sheng and LaFleur"))
sheng_lafleur_lusc_timer_signature_cor <- lapply(X = seq_along(lusc_fmat),
      function(x) {
        points(y = sheng_lafleur_lusc_timer_signature,
             x = lusc_fmat[,x],
             pch=20,
             col = lusc_fmat_colors[x])
        timer_positive_subset_ind <- lusc_fmat[,x] > 0
        cor(x = sheng_lafleur_lusc_timer_signature[timer_positive_subset_ind],
            y = lusc_fmat[timer_positive_subset_ind,x])
      })
names(sheng_lafleur_lusc_timer_signature_cor) <- names(lusc_fmat)
legend(x = "bottomright",
       legend = colnames(lusc_fmat),
       col = lusc_fmat_colors,
       pch=20,
       cex = 0.75)

# mark the tumors with no CD8 infiltrates
gdac_selected_cancers_annot_lusc$CD8 <- lusc_fmat$T_cell.CD8
gdac_selected_cancers_annot_lusc$CD4 <- lusc_fmat$T_cell.CD4
gdac_selected_cancers_annot_lusc$B_cell <- lusc_fmat$B_cell
gdac_selected_cancers_annot_lusc$Neutrophil <- lusc_fmat$Neutrophil
gdac_selected_cancers_annot_lusc$Macrophage <- lusc_fmat$Macrophage
gdac_selected_cancers_annot_lusc$DC <- lusc_fmat$DC


```

```{r}

# coad
temp_i <- match_fmat_tcga_names(fmat_names = rownames(coad_fmat),
                                tcga_names = names(chiappinelli_ifn_signature_median))
chiappinelli_coad_timer_signature <- chiappinelli_ifn_signature_median[temp_i]
temp_i <- match_fmat_tcga_names(fmat_names = rownames(coad_fmat),
                                tcga_names = names(canadas_ifn_signature_median))
canadas_coad_timer_signature <- canadas_ifn_signature_median[temp_i]
temp_i <- match_fmat_tcga_names(fmat_names = rownames(coad_fmat),
                                tcga_names = names(canadas_3putr_ifn_signature_median))
canadas_3putr_coad_timer_signature <- canadas_3putr_ifn_signature_median[temp_i]
temp_i <- match_fmat_tcga_names(fmat_names = rownames(coad_fmat),
                                tcga_names = names(sheng_lafleur_ifn_signature_median))
sheng_lafleur_coad_timer_signature <- sheng_lafleur_ifn_signature_median[temp_i]

# colors for fmat
coad_fmat_colors <- rainbow(n = ncol(coad_fmat))

# plot correlations
par(mfrow = c(1,3))
plot(x = coad_fmat[,1],
     y = chiappinelli_coad_timer_signature,
     xlim=c(xlim_min_all,xlim_all),
     ylim=c(ylim_min_all,ylim_all),
     xlab=c("TIMER value"),
     ylab=c("Median IFN-viral signature value (log10(counts))"),
     type = "n",
     main = paste0("Chiappinelli"))
chiappinelli_coad_timer_signature_cor <- lapply(X = seq_along(coad_fmat),
      function(x) {
        points(y = chiappinelli_coad_timer_signature,
             x = coad_fmat[,x],
             pch=20,
             col = coad_fmat_colors[x])
        timer_positive_subset_ind <- coad_fmat[,x] > 0
        cor(x = chiappinelli_coad_timer_signature[timer_positive_subset_ind],
            y = coad_fmat[timer_positive_subset_ind,x])
      })
names(chiappinelli_coad_timer_signature_cor) <- names(coad_fmat)
legend(x = "bottomright",
       legend = colnames(coad_fmat),
       col = coad_fmat_colors,
       pch=20,
       cex = 0.75)

# plot correlations
plot(x = coad_fmat[,1],
     y = canadas_coad_timer_signature,
     xlim=c(xlim_min_all,xlim_all),
     ylim=c(ylim_min_all,ylim_all),
     xlab=c("TIMER value"),
     ylab=c("Median IFN-viral signature value (log10(counts))"),
     type = "n",
     main = paste0("Canadas"))
canadas_coad_timer_signature_cor <- lapply(X = seq_along(coad_fmat),
      function(x) {
        points(y = canadas_coad_timer_signature,
             x = coad_fmat[,x],
             pch=20,
             col = coad_fmat_colors[x])
        timer_positive_subset_ind <- coad_fmat[,x] > 0
        cor(x = canadas_coad_timer_signature[timer_positive_subset_ind],
            y = coad_fmat[timer_positive_subset_ind,x])
      })
names(canadas_coad_timer_signature_cor) <- names(coad_fmat)
legend(x = "bottomright",
       legend = colnames(coad_fmat),
       col = coad_fmat_colors,
       pch=20,
       cex = 0.75)

# # plot correlations for Canadas 3' UTR ERV-containing genes (sense and antisense)
# plot(x = coad_fmat[,1],
#      y = canadas_3putr_coad_timer_signature,
#      xlim=c(0,max(coad_fmat)),
#      ylim=c(0,ylim_all),
#      xlab=c("TIMER value"),
#      ylab=c("Median IFN-viral signature value (log10(counts))"),
#      type = "n",
#      main = paste0("Canadas (3' UTR ERV-containing genes) signature-TIMER correlation, coad"))
# canadas_3putr_coad_timer_signature_cor <- lapply(X = seq_along(coad_fmat),
#       function(x) {
#         points(y = canadas_3putr_coad_timer_signature,
#              x = coad_fmat[,x],
#              pch=20,
#              col = coad_fmat_colors[x])
#         timer_positive_subset_ind <- coad_fmat[,x] > 0
#         cor(x = canadas_3putr_coad_timer_signature[timer_positive_subset_ind],
#             y = coad_fmat[timer_positive_subset_ind,x])
#       })
# names(canadas_3putr_coad_timer_signature_cor) <- names(coad_fmat)
# legend(x = "bottomright",
#        legend = colnames(coad_fmat),
#        col = coad_fmat_colors,
#        pch=20,
#        cex = 0.75)

# plot correlations
plot(x = coad_fmat[,1],
     y = sheng_lafleur_coad_timer_signature,
     xlim=c(xlim_min_all,xlim_all),
     ylim=c(ylim_min_all,ylim_all),
     xlab=c("TIMER value"),
     ylab=c("Median IFN-viral signature value (log10(counts))"),
     type = "n",
     main = paste0("Sheng LaFleur"))
sheng_lafleur_coad_timer_signature_cor <- lapply(X = seq_along(coad_fmat),
      function(x) {
        points(y = sheng_lafleur_coad_timer_signature,
             x = coad_fmat[,x],
             pch=20,
             col = coad_fmat_colors[x])
        timer_positive_subset_ind <- coad_fmat[,x] > 0
        cor(x = sheng_lafleur_coad_timer_signature[timer_positive_subset_ind],
            y = coad_fmat[timer_positive_subset_ind,x])
      })
names(sheng_lafleur_coad_timer_signature_cor) <- names(coad_fmat)
legend(x = "bottomright",
       legend = colnames(coad_fmat),
       col = coad_fmat_colors,
       pch=20,
       cex = 0.75)

# mark the tumors with no CD8 infiltrates
gdac_selected_cancers_annot_coad$CD8 <- coad_fmat$T_cell.CD8
gdac_selected_cancers_annot_coad$CD4 <- coad_fmat$T_cell.CD4
gdac_selected_cancers_annot_coad$B_cell <- coad_fmat$B_cell
gdac_selected_cancers_annot_coad$Neutrophil <- coad_fmat$Neutrophil
gdac_selected_cancers_annot_coad$Macrophage <- coad_fmat$Macrophage
gdac_selected_cancers_annot_coad$DC <- coad_fmat$DC


```

Heatmap of correlations
```{r full_correlations}
sheng_lafleur_cor_timer_ifnviral <- do.call(rbind,list(sheng_lafleur_brca_timer_signature_cor,
                                                       sheng_lafleur_ov_timer_signature_cor,
                                                       sheng_lafleur_lusc_timer_signature_cor,
                                                       sheng_lafleur_coad_timer_signature_cor))
canadas_3putr_lafleur_cor_timer_ifnviral <- do.call(rbind,list(canadas_3putr_brca_timer_signature_cor,
                                                       canadas_3putr_ov_timer_signature_cor,
                                                       canadas_3putr_lusc_timer_signature_cor,
                                                       canadas_3putr_brca_timer_signature_cor))
canadas_lafleur_cor_timer_ifnviral <- do.call(rbind,list(canadas_brca_timer_signature_cor,
                                                       canadas_ov_timer_signature_cor,
                                                       canadas_lusc_timer_signature_cor,
                                                       canadas_brca_timer_signature_cor))
chiappinelli_lafleur_cor_timer_ifnviral <- do.call(rbind,list(chiappinelli_brca_timer_signature_cor,
                                                       chiappinelli_ov_timer_signature_cor,
                                                       chiappinelli_lusc_timer_signature_cor,
                                                       chiappinelli_coad_timer_signature_cor))
sig_timer_cancer <- rbind(sheng_lafleur_cor_timer_ifnviral,
                                       canadas_3putr_lafleur_cor_timer_ifnviral,
                                       canadas_lafleur_cor_timer_ifnviral,
                                       chiappinelli_lafleur_cor_timer_ifnviral)
sig_timer_cancer_rowannot <- data.frame(signature = c(rep("sheng/lafleur",nrow(sheng_lafleur_cor_timer_ifnviral)),
                                                      rep("canadas_3'UTR",nrow(canadas_3putr_lafleur_cor_timer_ifnviral)),
                                                      rep("canadas",nrow(canadas_lafleur_cor_timer_ifnviral)),
                                                      rep("chiappinelli",nrow(chiappinelli_lafleur_cor_timer_ifnviral)))
                                        )
sig_timer_cancer_rowannot$cancer <- c(rep(c("BRCA","OV","LUSC","COAD"),4))
sig_timer_cancer_mat <- apply(sig_timer_cancer,2,as.numeric)

rownames(sig_timer_cancer_mat) <- rownames(sig_timer_cancer_rowannot)

pheatmap(mat = t(sig_timer_cancer_mat),
         annotation_col = sig_timer_cancer_rowannot,
         show_colnames = FALSE,
         fontsize = 6.0,
         main = c("Correlation between IFN-viral signatures\nand TIMER scores for immune subsets"))
```

```
gdac_selected_cancers_annot_brca$CD8 <- brca_fmat$T_cell.CD8
gdac_selected_cancers_annot_ov$CD8 <- ov_fmat$T_cell.CD8
gdac_selected_cancers_annot_lusc$CD8 <- lusc_fmat$T_cell.CD8
gdac_selected_cancers_annot_coad$CD8 <- coad_fmat$T_cell.CD8

```

# ERV gene expression and tumor infiltration

```{r}
print(rownames(brca_gdac)[grep(pattern = "ERV",x = rownames(brca_gdac))])
print(rownames(ov_gdac)[grep(pattern = "ERV",x = rownames(ov_gdac))])
print(rownames(lusc_gdac)[grep(pattern = "ERV",x = rownames(lusc_gdac))])
print(rownames(coad_gdac)[grep(pattern = "ERV",x = rownames(coad_gdac))])
# what is the correlation between the expression of this ERV gene and immune infiltration?

```

The ERV gene "ERVFRDE1" (endogenous retrovirus group FRD member 1, envelope) is present within the processed TCGA data; any additional expression quantification of other ERV's from the TCGA dataset (in particular "TotalRNAseqv2" data) will require raw reads for which I need dbGAP approval (as raw reads are controlled access, since they contain actual patient sequence data). As a rough attempt to study correlations between ERV gene expression from TCGA and immune infiltration, we will correlate the two. 

We use ERVFRDE1 as a *very* rough proxy of ERV expression, since it is part of an inactivated provirus on chromosome 7 of the human genome (it is thought to be important for trophoblast biology). 

```{r}
brca_gdac_ervfrde1 <- brca_gdac[grep(pattern = "ERV",x = rownames(brca_gdac)),]
coad_gdac_ervfrde1 <- coad_gdac[grep(pattern = "ERV",x = rownames(coad_gdac)),]
lusc_gdac_ervfrde1 <- lusc_gdac[grep(pattern = "ERV",x = rownames(lusc_gdac)),]
ov_gdac_ervfrde1 <- ov_gdac[grep(pattern = "ERV",x = rownames(ov_gdac)),]

# brca
plot(log10(as.numeric(brca_gdac_ervfrde1)),
     brca_fmat[,1],
     col=as.factor(brca_fmat[,1] > 0),
     main = paste("TIMER-inferred Immune subset versus ERVFRDE1 expression in BRCA"),
     type = "n",
     xlim=c(0,5),
     ylim=c(0,3),
     pch=20)
sapply(X = seq_len(ncol(brca_fmat)),
       FUN = function(x) {
         points(log10(as.numeric(brca_gdac_ervfrde1)),
              brca_fmat[,x],
              col=brca_fmat_colors[x],
              pch=20)
              #col=as.factor(brca_fmat[,x] > 0),
              # pch=20)
       })
legend(x = "topright",
       legend = colnames(coad_fmat),
       col = coad_fmat_colors,
       pch=20,
       cex = 0.75)

# coad
plot(log10(as.numeric(coad_gdac_ervfrde1)),
     coad_fmat[,1],
     col=as.factor(coad_fmat[,1] > 0),
     main = paste("TIMER-inferred Immune subset versus ERVFRDE1 expression in COAD"),
     type = "n",
     xlim=c(0,5),
     ylim=c(0,3),
     pch=20)
sapply(X = seq_len(ncol(coad_fmat)),
       FUN = function(x) {
         points(log10(as.numeric(coad_gdac_ervfrde1)),
              coad_fmat[,x],
              col=coad_fmat_colors[x],
              pch=20)
       })
legend(x = "topright",
       legend = colnames(coad_fmat),
       col = coad_fmat_colors,
       pch=20,
       cex = 0.75)

# lusc
plot(log10(as.numeric(lusc_gdac_ervfrde1)),
     lusc_fmat[,1],
     col=as.factor(lusc_fmat[,1] > 0),
     main = paste("TIMER-inferred Immune subset versus ERVFRDE1 expression in LUSC"),
     type = "n",
     xlim=c(0,5),
     ylim=c(0,3),
     pch=20)
sapply(X = seq_len(ncol(lusc_fmat)),
       FUN = function(x) {
         points(log10(as.numeric(lusc_gdac_ervfrde1)),
              lusc_fmat[,x],
              col=lusc_fmat_colors[x],
              pch=20)
       })
legend(x = "topright",
       legend = colnames(coad_fmat),
       col = coad_fmat_colors,
       pch=20,
       cex = 0.75)

# ov
plot(log10(as.numeric(ov_gdac_ervfrde1)),
     ov_fmat[,1],
     col=as.factor(ov_fmat[,1] > 0),
     main = paste("TIMER-inferred Immune subset versus ERVFRDE1 expression in OV"),
     type = "n",
     xlim=c(0,5),
     ylim=c(0,3),
     pch=20)
sapply(X = seq_len(ncol(ov_fmat)),
       FUN = function(x) {
         points(log10(as.numeric(ov_gdac_ervfrde1)),
              ov_fmat[,x],
              col=coad_fmat_colors[x],
              pch=20)
       })
legend(x = "topright",
       legend = colnames(coad_fmat),
       col = coad_fmat_colors,
       pch=20,
       cex = 0.75)


```

ERVFRDE1 production and estimated immune cell enrichment (for all subsets studied) in the TCGA tumor samples are weakly correlated. Without raw RNA-seq data for TCGA, we cannot definitely state whether directly-measured ERV gene expression is correlated with immune infiltration. 

Additionally, the signatures that we obtained do not correlate very well with the individual signatures
```{r}
# BRCA
plot(as.numeric(chiappinelli_brca_timer_signature),
     log10(as.numeric(brca_gdac_ervfrde1)),
     col=c("red"),
     main = paste("brca median signature value versus ERVFRDE1 expression in brca"),
     xlim=c(0,4),
     ylim=c(0,4),
     xlab = c("Median log10(IFN-viral signature+1)"),
     ylab = c("log10(ERVFRDE1)"),
     pch=20)
points(as.numeric(canadas_brca_timer_signature),
     log10(as.numeric(brca_gdac_ervfrde1)),
     col=c("blue"),
     pch=20)
points(as.numeric(canadas_3putr_brca_timer_signature),
     log10(as.numeric(brca_gdac_ervfrde1)),
     col=c("orange"),
     pch=20)
points(as.numeric(sheng_lafleur_brca_timer_signature),
     log10(as.numeric(brca_gdac_ervfrde1)),
     col=c("grey"),
     pch=20)
legend("topleft",
       legend = c("Chiappinelli",
                  "Canadas",
                  "Canadas 3' UTR ERV-containing genes",
                  "Sheng and LaFleur"),
       col = c("red","blue","orange","grey"),cex = 0.75,
       pch = 20)

# OV
plot(as.numeric(chiappinelli_ov_timer_signature),
     log10(as.numeric(ov_gdac_ervfrde1)),
     col=c("red"),
     main = paste("ov median signature value versus ERVFRDE1 expression in ov"),
     xlim=c(0,4),
     ylim=c(0,4),
     xlab = c("Median log10(IFN-viral signature+1)"),
     ylab = c("log10(ERVFRDE1)"),
     pch=20)
points(as.numeric(canadas_ov_timer_signature),
     log10(as.numeric(ov_gdac_ervfrde1)),
     col=c("blue"),
     pch=20)
points(as.numeric(canadas_3putr_ov_timer_signature),
     log10(as.numeric(ov_gdac_ervfrde1)),
     col=c("orange"),
     pch=20)
points(as.numeric(sheng_lafleur_ov_timer_signature),
     log10(as.numeric(ov_gdac_ervfrde1)),
     col=c("grey"),
     pch=20)
legend("topleft",
       legend = c("Chiappinelli",
                  "Canadas",
                  "Canadas 3' UTR ERV-containing genes",
                  "Sheng and LaFleur"),
       col = c("red","blue","orange","grey"),cex = 0.75,
       pch = 20)

# LUSC
plot(as.numeric(chiappinelli_lusc_timer_signature),
     log10(as.numeric(lusc_gdac_ervfrde1)),
     col=c("red"),
     main = paste("lusc median signature value versus ERVFRDE1 expression in lusc"),
     xlim=c(0,4),
     ylim=c(0,4),
     xlab = c("Median log10(IFN-viral signature+1)"),
     ylab = c("log10(ERVFRDE1)"),
     pch=20)
points(as.numeric(canadas_lusc_timer_signature),
     log10(as.numeric(lusc_gdac_ervfrde1)),
     col=c("blue"),
     pch=20)
points(as.numeric(canadas_3putr_lusc_timer_signature),
     log10(as.numeric(lusc_gdac_ervfrde1)),
     col=c("orange"),
     pch=20)
points(as.numeric(sheng_lafleur_lusc_timer_signature),
     log10(as.numeric(lusc_gdac_ervfrde1)),
     col=c("grey"),
     pch=20)
legend("topleft",
       legend = c("Chiappinelli",
                  "Canadas",
                  "Canadas 3' UTR ERV-containing genes",
                  "Sheng and LaFleur"),
       col = c("red","blue","orange","grey"),cex = 0.75,
       pch = 20)

# COAD
plot(as.numeric(chiappinelli_coad_timer_signature),
     log10(as.numeric(coad_gdac_ervfrde1)),
     col=c("red"),
     main = paste("coad median signature value versus ERVFRDE1 expression in coad"),
     xlim=c(0,4),
     ylim=c(0,4),
     xlab = c("Median log10(IFN-viral signature+1)"),
     ylab = c("log10(ERVFRDE1)"),
     pch=20)
points(as.numeric(canadas_coad_timer_signature),
     log10(as.numeric(coad_gdac_ervfrde1)),
     col=c("blue"),
     pch=20)
points(as.numeric(canadas_3putr_coad_timer_signature),
     log10(as.numeric(coad_gdac_ervfrde1)),
     col=c("orange"),
     pch=20)
points(as.numeric(sheng_lafleur_coad_timer_signature),
     log10(as.numeric(coad_gdac_ervfrde1)),
     col=c("grey"),
     pch=20)
legend("topleft",
       legend = c("Chiappinelli",
                  "Canadas",
                  "Canadas 3' UTR ERV-containing genes",
                  "Sheng and LaFleur"),
       col = c("red","blue","orange","grey"),cex = 0.75,
       pch = 20)


```


## Urothelial cancer ERVs vs immune infiltration

I also downloaded a recent dataset from Snyder et al PLoS 2017 (https://journals.plos.org/plosmedicine/article?id=10.1371/journal.pmed.1002309#sec008) of urothelial cancer patients that were on anti-CTLA-4 therapy. Solovyok et al Cell Reports 2018 (https://www.cell.com/cell-reports/pdf/S2211-1247(18)30384-X.pdf) analyzed this dataset to examine correlations between ERV gene expression and patient outcomes, but we will mine this dataset to examine ERV expression corerlations with immune infiltration (as the RNA-seq samples are of patient biopsies). 

<do this after you paste the other figures>

# Conclusions

1. The three signatures from the paper all seem to include different components of tumor innate immunity and the antiviral response. However, the median expression values of the signatures are reasonably well-correlated with each other.
  a. Why would ERV activation lead to such different signatures? Different experimental and biological conditions mean that different transcriptional "programs" are activated 
  b. What happens if we consider only up-regulated components of each signature?
2. When we pool the signatures, we find four groups that stand out which could constitute a "unified" signature: a highly-expressed subset consisting of housekeeping genes and components of antigen presentation, a more sparsely-expressed subset in tumors, and two more "medium-expressed" subset of genes. One of the two "medium-subsets" is enriched for components of the interferon response pathway.
3. Correlations between IFN-viral signatures appear to differ more by cancer type, with weaker correlations in OV. Moreover, the Canadas signatures show the strongest median values
